// @vercel/sdk/esm/mcp-server/tools/accessGroupsUpdateAccessGroupProject.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { accessGroupsUpdateAccessGroupProject } from "../../funcs/accessGroupsUpdateAccessGroupProject.js";
import { UpdateAccessGroupProjectRequest$inboundSchema } from "../../models/updateaccessgroupprojectop.js";
import { formatResult } from "../tools.js";
const args = {
    request: UpdateAccessGroupProjectRequest$inboundSchema,
};
export const tool$accessGroupsUpdateAccessGroupProject = {
    name: "access-groups-update-access-group-project",
    description: `Update an access group project

Allows update of an access group project`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await accessGroupsUpdateAccessGroupProject(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
