// @vercel/sdk/esm/mcp-server/tools/accessGroupsUpdateAccessGroup.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { accessGroupsUpdateAccessGroup } from "../../funcs/accessGroupsUpdateAccessGroup.js";
import { UpdateAccessGroupRequest$inboundSchema } from "../../models/updateaccessgroupop.js";
import { formatResult } from "../tools.js";
const args = {
    request: UpdateAccessGroupRequest$inboundSchema,
};
export const tool$accessGroupsUpdateAccessGroup = {
    name: "access-groups-update-access-group",
    description: `Update an access group

Allows to update an access group metadata`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await accessGroupsUpdateAccessGroup(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
