// @vercel/sdk/esm/mcp-server/tools/artifactsStatus.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { artifactsStatus } from "../../funcs/artifactsStatus.js";
import { StatusRequest$inboundSchema } from "../../models/statusop.js";
import { formatResult } from "../tools.js";
const args = {
    request: StatusRequest$inboundSchema,
};
export const tool$artifactsStatus = {
    name: "artifacts-status",
    description: `Get status of Remote Caching for this principal

Check the status of Remote Caching for this principal. Returns a JSON-encoded status indicating if Remote Caching is enabled, disabled, or disabled due to usage limits.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await artifactsStatus(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
