// @vercel/sdk/esm/mcp-server/tools/authenticationGetAuthToken.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { authenticationGetAuthToken } from "../../funcs/authenticationGetAuthToken.js";
import { GetAuthTokenRequest$inboundSchema } from "../../models/getauthtokenop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetAuthTokenRequest$inboundSchema,
};
export const tool$authenticationGetAuthToken = {
    name: "authentication-get-auth-token",
    description: `Get Auth Token Metadata

Retrieve metadata about an authentication token belonging to the currently authenticated User.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await authenticationGetAuthToken(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
