// @vercel/sdk/esm/mcp-server/tools/checksUpdateCheck.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { checksUpdateCheck } from "../../funcs/checksUpdateCheck.js";
import { UpdateCheckRequest$inboundSchema } from "../../models/updatecheckop.js";
import { formatResult } from "../tools.js";
const args = {
    request: UpdateCheckRequest$inboundSchema,
};
export const tool$checksUpdateCheck = {
    name: "checks-update-check",
    description: `Update a check

Update an existing check. This endpoint must be called with an OAuth2 or it will produce a 400 error.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await checksUpdateCheck(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
