// @vercel/sdk/esm/mcp-server/tools/deploymentsCancelDeployment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { deploymentsCancelDeployment } from "../../funcs/deploymentsCancelDeployment.js";
import { CancelDeploymentRequest$inboundSchema } from "../../models/canceldeploymentop.js";
import { formatResult } from "../tools.js";
const args = {
    request: CancelDeploymentRequest$inboundSchema,
};
export const tool$deploymentsCancelDeployment = {
    name: "deployments-cancel-deployment",
    description: `Cancel a deployment

This endpoint allows you to cancel a deployment which is currently building, by supplying its \`id\` in the URL.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await deploymentsCancelDeployment(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
