// @vercel/sdk/esm/mcp-server/tools/dnsGetRecords.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { dnsGetRecords } from "../../funcs/dnsGetRecords.js";
import { GetRecordsRequest$inboundSchema } from "../../models/getrecordsop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetRecordsRequest$inboundSchema,
};
export const tool$dnsGetRecords = {
    name: "dns-get-records",
    description: `List existing DNS records

Retrieves a list of DNS records created for a domain name. By default it returns 20 records if no limit is provided. The rest can be retrieved using the pagination options.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await dnsGetRecords(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
