// @vercel/sdk/esm/mcp-server/tools/domainsDeleteDomain.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { domainsDeleteDomain } from "../../funcs/domainsDeleteDomain.js";
import { DeleteDomainRequest$inboundSchema } from "../../models/deletedomainop.js";
import { formatResult } from "../tools.js";
const args = {
    request: DeleteDomainRequest$inboundSchema,
};
export const tool$domainsDeleteDomain = {
    name: "domains-delete-domain",
    description: `Remove a domain by name

Delete a previously registered domain name from Vercel. Deleting a domain will automatically remove any associated aliases.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await domainsDeleteDomain(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
