// @vercel/sdk/esm/mcp-server/tools/edgeConfigPatchEdgeConfigItems.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigPatchEdgeConfigItems } from "../../funcs/edgeConfigPatchEdgeConfigItems.js";
import { PatchEdgeConfigItemsRequest$inboundSchema } from "../../models/patchedgeconfigitemsop.js";
import { formatResult } from "../tools.js";
const args = {
    request: PatchEdgeConfigItemsRequest$inboundSchema,
};
export const tool$edgeConfigPatchEdgeConfigItems = {
    name: "edge-config-patch-edge-config-items",
    description: `Update Edge Config items in batch

Update multiple Edge Config Items in batch.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigPatchEdgeConfigItems(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
