// @vercel/sdk/esm/mcp-server/tools/edgeConfigDeleteEdgeConfigTokens.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigDeleteEdgeConfigTokens } from "../../funcs/edgeConfigDeleteEdgeConfigTokens.js";
import { DeleteEdgeConfigTokensRequest$inboundSchema } from "../../models/deleteedgeconfigtokensop.js";
import { formatResult } from "../tools.js";
const args = {
    request: DeleteEdgeConfigTokensRequest$inboundSchema,
};
export const tool$edgeConfigDeleteEdgeConfigTokens = {
    name: "edge-config-delete-edge-config-tokens",
    description: `Delete one or more Edge Config tokens

Deletes one or more tokens of an existing Edge Config.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigDeleteEdgeConfigTokens(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return formatResult(void 0, apiCall);
    },
};
