// @vercel/sdk/esm/mcp-server/tools/edgeConfigPatchEdgeConfigSchema.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigPatchEdgeConfigSchema } from "../../funcs/edgeConfigPatchEdgeConfigSchema.js";
import { PatchEdgeConfigSchemaRequest$inboundSchema } from "../../models/patchedgeconfigschemaop.js";
import { formatResult } from "../tools.js";
const args = {
    request: PatchEdgeConfigSchemaRequest$inboundSchema,
};
export const tool$edgeConfigPatchEdgeConfigSchema = {
    name: "edge-config-patch-edge-config-schema",
    description: `Update Edge Config schema

Update an Edge Config's schema.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigPatchEdgeConfigSchema(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
