// @vercel/sdk/esm/mcp-server/tools/edgeConfigGetEdgeConfigBackups.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigGetEdgeConfigBackups } from "../../funcs/edgeConfigGetEdgeConfigBackups.js";
import { GetEdgeConfigBackupsRequest$inboundSchema } from "../../models/getedgeconfigbackupsop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetEdgeConfigBackupsRequest$inboundSchema,
};
export const tool$edgeConfigGetEdgeConfigBackups = {
    name: "edge-config-get-edge-config-backups",
    description: `Get Edge Config backups

Returns backups of an Edge Config.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigGetEdgeConfigBackups(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
