// @vercel/sdk/esm/mcp-server/tools/marketplaceCreateInstallationIntegrationEdgeConfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { marketplaceCreateInstallationIntegrationEdgeConfig } from "../../funcs/marketplaceCreateInstallationIntegrationEdgeConfig.js";
import { HeadV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationEdgeConfigRequest$inboundSchema, } from "../../models/headv1installationsintegrationconfigurationidresourcesresourceidexperimentationedgeconfigop.js";
import { formatResult } from "../tools.js";
const args = {
    request: HeadV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationEdgeConfigRequest$inboundSchema,
};
export const tool$marketplaceCreateInstallationIntegrationEdgeConfig = {
    name: "marketplace-create-installation-integration-edge-config",
    description: `Get the data of a user-provided Edge Config

When the user enabled Edge Config syncing, then this endpoint can be used by the partner to fetch the contents of the Edge Config.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await marketplaceCreateInstallationIntegrationEdgeConfig(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
