// @vercel/sdk/esm/mcp-server/tools/marketplaceUpdateInstallationIntegrationEdgeConfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { marketplaceUpdateInstallationIntegrationEdgeConfig } from "../../funcs/marketplaceUpdateInstallationIntegrationEdgeConfig.js";
import { PutV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationEdgeConfigRequest$inboundSchema, } from "../../models/putv1installationsintegrationconfigurationidresourcesresourceidexperimentationedgeconfigop.js";
import { formatResult } from "../tools.js";
const args = {
    request: PutV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationEdgeConfigRequest$inboundSchema,
};
export const tool$marketplaceUpdateInstallationIntegrationEdgeConfig = {
    name: "marketplace-update-installation-integration-edge-config",
    description: `Push data into a user-provided Edge Config

When the user enabled Edge Config syncing, then this endpoint can be used by the partner to push their configuration data into the relevant Edge Config.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await marketplaceUpdateInstallationIntegrationEdgeConfig(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
