// @vercel/sdk/esm/mcp-server/tools/marketplaceSubmitPrepaymentBalances.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { marketplaceSubmitPrepaymentBalances } from "../../funcs/marketplaceSubmitPrepaymentBalances.js";
import { SubmitPrepaymentBalancesRequest$inboundSchema } from "../../models/submitprepaymentbalancesop.js";
import { formatResult } from "../tools.js";
const args = {
    request: SubmitPrepaymentBalancesRequest$inboundSchema,
};
export const tool$marketplaceSubmitPrepaymentBalances = {
    name: "marketplace-submit-prepayment-balances",
    description: `Submit Prepayment Balances

Sends the prepayment balances. The partner should do this at least once a day and ideally once per hour. <br/> Use the \`credentials.access_token\` we provided in the [Upsert Installation](#upsert-installation) body to authorize this request.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await marketplaceSubmitPrepaymentBalances(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return formatResult(void 0, apiCall);
    },
};
