// @vercel/sdk/esm/mcp-server/tools/projectsGetProjectDomains.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { projectsGetProjectDomains } from "../../funcs/projectsGetProjectDomains.js";
import { GetProjectDomainsRequest$inboundSchema } from "../../models/getprojectdomainsop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetProjectDomainsRequest$inboundSchema,
};
export const tool$projectsGetProjectDomains = {
    name: "projects-get-project-domains",
    description: `Retrieve project domains by project by id or name

Retrieve the domains associated with a given project by passing either the project \`id\` or \`name\` in the URL.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await projectsGetProjectDomains(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
