// @vercel/sdk/esm/mcp-server/tools/projectsUpdateProjectProtectionBypass.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { projectsUpdateProjectProtectionBypass } from "../../funcs/projectsUpdateProjectProtectionBypass.js";
import { UpdateProjectProtectionBypassRequest$inboundSchema } from "../../models/updateprojectprotectionbypassop.js";
import { formatResult } from "../tools.js";
const args = {
    request: UpdateProjectProtectionBypassRequest$inboundSchema,
};
export const tool$projectsUpdateProjectProtectionBypass = {
    name: "projects-update-project-protection-bypass",
    description: `Update Protection Bypass for Automation

Update the deployment protection automation bypass for a project`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await projectsUpdateProjectProtectionBypass(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
