// @vercel/sdk/esm/mcp-server/tools/projectsListPromoteAliases.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { projectsListPromoteAliases } from "../../funcs/projectsListPromoteAliases.js";
import { ListPromoteAliasesRequest$inboundSchema } from "../../models/listpromotealiasesop.js";
import { formatResult } from "../tools.js";
const args = {
    request: ListPromoteAliasesRequest$inboundSchema,
};
export const tool$projectsListPromoteAliases = {
    name: "projects-list-promote-aliases",
    description: `Gets a list of aliases with status for the current promote

Get a list of aliases related to the last promote request with their mapping status`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await projectsListPromoteAliases(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
