// @vercel/sdk/esm/mcp-server/tools/projectsCreateProject.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { projectsCreateProject } from "../../funcs/projectsCreateProject.js";
import { CreateProjectRequest$inboundSchema } from "../../models/createprojectop.js";
import { formatResult } from "../tools.js";
const args = {
    request: CreateProjectRequest$inboundSchema,
};
export const tool$projectsCreateProject = {
    name: "projects-create-project",
    description: `Create a new project

Allows to create a new project with the provided configuration. It only requires the project \`name\` but more configuration can be provided to override the defaults.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await projectsCreateProject(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
