// @vercel/sdk/esm/mcp-server/tools/securityGetFirewallConfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { securityGetFirewallConfig } from "../../funcs/securityGetFirewallConfig.js";
import { GetFirewallConfigRequest$inboundSchema } from "../../models/getfirewallconfigop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetFirewallConfigRequest$inboundSchema,
};
export const tool$securityGetFirewallConfig = {
    name: "security-get-firewall-config",
    description: `Read Firewall Configuration

Retrieve the specified firewall configuration for a project. The deployed configVersion will be \`active\``,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await securityGetFirewallConfig(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
