// @vercel/sdk/esm/mcp-server/tools/teamsInviteUserToTeam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { teamsInviteUserToTeam } from "../../funcs/teamsInviteUserToTeam.js";
import { InviteUserToTeamRequest$inboundSchema } from "../../models/inviteusertoteamop.js";
import { formatResult } from "../tools.js";
const args = {
    request: InviteUserToTeamRequest$inboundSchema,
};
export const tool$teamsInviteUserToTeam = {
    name: "teams-invite-user-to-team",
    description: `Invite a user

Invite a user to join the team specified in the URL. The authenticated user needs to be an \`OWNER\` in order to successfully invoke this endpoint. The user can be specified with an email or an ID. If both email and ID are provided, ID will take priority.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await teamsInviteUserToTeam(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
