// @vercel/sdk/esm/mcp-server/tools/teamsPatchTeam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { teamsPatchTeam } from "../../funcs/teamsPatchTeam.js";
import { PatchTeamRequest$inboundSchema } from "../../models/patchteamop.js";
import { formatResult } from "../tools.js";
const args = {
    request: PatchTeamRequest$inboundSchema,
};
export const tool$teamsPatchTeam = {
    name: "teams-patch-team",
    description: `Update a Team

Update the information of a Team specified by the \`teamId\` parameter. The request body should contain the information that will be updated on the Team.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await teamsPatchTeam(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
