// @vercel/sdk/esm/mcp-server/tools/teamsRemoveTeamMember.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { teamsRemoveTeamMember } from "../../funcs/teamsRemoveTeamMember.js";
import { RemoveTeamMemberRequest$inboundSchema } from "../../models/removeteammemberop.js";
import { formatResult } from "../tools.js";
const args = {
    request: RemoveTeamMemberRequest$inboundSchema,
};
export const tool$teamsRemoveTeamMember = {
    name: "teams-remove-team-member",
    description: `Remove a Team Member

Remove a Team Member from the Team, or dismiss a user that requested access, or leave a team.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await teamsRemoveTeamMember(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
