// @vercel/sdk/esm/mcp-server/tools/teamsUpdateTeamMember.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { teamsUpdateTeamMember } from "../../funcs/teamsUpdateTeamMember.js";
import { UpdateTeamMemberRequest$inboundSchema } from "../../models/updateteammemberop.js";
import { formatResult } from "../tools.js";
const args = {
    request: UpdateTeamMemberRequest$inboundSchema,
};
export const tool$teamsUpdateTeamMember = {
    name: "teams-update-team-member",
    description: `Update a Team Member

Update the membership of a Team Member on the Team specified by \`teamId\`, such as changing the _role_ of the member, or confirming a request to join the Team for an unconfirmed member. The authenticated user must be an \`OWNER\` of the Team.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await teamsUpdateTeamMember(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
