// @vercel/sdk/esm/models/deleteedgeconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteEdgeConfigRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const DeleteEdgeConfigRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteEdgeConfigRequest$;
(function (DeleteEdgeConfigRequest$) {
    /** @deprecated use `DeleteEdgeConfigRequest$inboundSchema` instead. */
    DeleteEdgeConfigRequest$.inboundSchema = DeleteEdgeConfigRequest$inboundSchema;
    /** @deprecated use `DeleteEdgeConfigRequest$outboundSchema` instead. */
    DeleteEdgeConfigRequest$.outboundSchema = DeleteEdgeConfigRequest$outboundSchema;
})(DeleteEdgeConfigRequest$ || (DeleteEdgeConfigRequest$ = {}));
export function deleteEdgeConfigRequestToJSON(deleteEdgeConfigRequest) {
    return JSON.stringify(DeleteEdgeConfigRequest$outboundSchema.parse(deleteEdgeConfigRequest));
}
export function deleteEdgeConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteEdgeConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteEdgeConfigRequest' from JSON`);
}
