// @vercel/sdk/esm/models/getedgeconfigschemaop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetEdgeConfigSchemaRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetEdgeConfigSchemaRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetEdgeConfigSchemaRequest$;
(function (GetEdgeConfigSchemaRequest$) {
    /** @deprecated use `GetEdgeConfigSchemaRequest$inboundSchema` instead. */
    GetEdgeConfigSchemaRequest$.inboundSchema = GetEdgeConfigSchemaRequest$inboundSchema;
    /** @deprecated use `GetEdgeConfigSchemaRequest$outboundSchema` instead. */
    GetEdgeConfigSchemaRequest$.outboundSchema = GetEdgeConfigSchemaRequest$outboundSchema;
})(GetEdgeConfigSchemaRequest$ || (GetEdgeConfigSchemaRequest$ = {}));
export function getEdgeConfigSchemaRequestToJSON(getEdgeConfigSchemaRequest) {
    return JSON.stringify(GetEdgeConfigSchemaRequest$outboundSchema.parse(getEdgeConfigSchemaRequest));
}
export function getEdgeConfigSchemaRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigSchemaRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigSchemaRequest' from JSON`);
}
/** @internal */
export const GetEdgeConfigSchemaResponseBody$inboundSchema = z.object({});
/** @internal */
export const GetEdgeConfigSchemaResponseBody$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetEdgeConfigSchemaResponseBody$;
(function (GetEdgeConfigSchemaResponseBody$) {
    /** @deprecated use `GetEdgeConfigSchemaResponseBody$inboundSchema` instead. */
    GetEdgeConfigSchemaResponseBody$.inboundSchema = GetEdgeConfigSchemaResponseBody$inboundSchema;
    /** @deprecated use `GetEdgeConfigSchemaResponseBody$outboundSchema` instead. */
    GetEdgeConfigSchemaResponseBody$.outboundSchema = GetEdgeConfigSchemaResponseBody$outboundSchema;
})(GetEdgeConfigSchemaResponseBody$ || (GetEdgeConfigSchemaResponseBody$ = {}));
export function getEdgeConfigSchemaResponseBodyToJSON(getEdgeConfigSchemaResponseBody) {
    return JSON.stringify(GetEdgeConfigSchemaResponseBody$outboundSchema.parse(getEdgeConfigSchemaResponseBody));
}
export function getEdgeConfigSchemaResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigSchemaResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigSchemaResponseBody' from JSON`);
}
