// @vercel/sdk/esm/models/listdeploymentfilesop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const ListDeploymentFilesRequest$inboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const ListDeploymentFilesRequest$outboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListDeploymentFilesRequest$;
(function (ListDeploymentFilesRequest$) {
    /** @deprecated use `ListDeploymentFilesRequest$inboundSchema` instead. */
    ListDeploymentFilesRequest$.inboundSchema = ListDeploymentFilesRequest$inboundSchema;
    /** @deprecated use `ListDeploymentFilesRequest$outboundSchema` instead. */
    ListDeploymentFilesRequest$.outboundSchema = ListDeploymentFilesRequest$outboundSchema;
})(ListDeploymentFilesRequest$ || (ListDeploymentFilesRequest$ = {}));
export function listDeploymentFilesRequestToJSON(listDeploymentFilesRequest) {
    return JSON.stringify(ListDeploymentFilesRequest$outboundSchema.parse(listDeploymentFilesRequest));
}
export function listDeploymentFilesRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentFilesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentFilesRequest' from JSON`);
}
