// @vercel/sdk/esm/models/removeprojectdomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const RemoveProjectDomainRequestBody$inboundSchema = z.object({
    removeRedirects: z.boolean().optional(),
});
/** @internal */
export const RemoveProjectDomainRequestBody$outboundSchema = z.object({
    removeRedirects: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveProjectDomainRequestBody$;
(function (RemoveProjectDomainRequestBody$) {
    /** @deprecated use `RemoveProjectDomainRequestBody$inboundSchema` instead. */
    RemoveProjectDomainRequestBody$.inboundSchema = RemoveProjectDomainRequestBody$inboundSchema;
    /** @deprecated use `RemoveProjectDomainRequestBody$outboundSchema` instead. */
    RemoveProjectDomainRequestBody$.outboundSchema = RemoveProjectDomainRequestBody$outboundSchema;
})(RemoveProjectDomainRequestBody$ || (RemoveProjectDomainRequestBody$ = {}));
export function removeProjectDomainRequestBodyToJSON(removeProjectDomainRequestBody) {
    return JSON.stringify(RemoveProjectDomainRequestBody$outboundSchema.parse(removeProjectDomainRequestBody));
}
export function removeProjectDomainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectDomainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectDomainRequestBody' from JSON`);
}
/** @internal */
export const RemoveProjectDomainRequest$inboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => RemoveProjectDomainRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const RemoveProjectDomainRequest$outboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => RemoveProjectDomainRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveProjectDomainRequest$;
(function (RemoveProjectDomainRequest$) {
    /** @deprecated use `RemoveProjectDomainRequest$inboundSchema` instead. */
    RemoveProjectDomainRequest$.inboundSchema = RemoveProjectDomainRequest$inboundSchema;
    /** @deprecated use `RemoveProjectDomainRequest$outboundSchema` instead. */
    RemoveProjectDomainRequest$.outboundSchema = RemoveProjectDomainRequest$outboundSchema;
})(RemoveProjectDomainRequest$ || (RemoveProjectDomainRequest$ = {}));
export function removeProjectDomainRequestToJSON(removeProjectDomainRequest) {
    return JSON.stringify(RemoveProjectDomainRequest$outboundSchema.parse(removeProjectDomainRequest));
}
export function removeProjectDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectDomainRequest' from JSON`);
}
/** @internal */
export const RemoveProjectDomainResponseBody$inboundSchema = z.object({});
/** @internal */
export const RemoveProjectDomainResponseBody$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveProjectDomainResponseBody$;
(function (RemoveProjectDomainResponseBody$) {
    /** @deprecated use `RemoveProjectDomainResponseBody$inboundSchema` instead. */
    RemoveProjectDomainResponseBody$.inboundSchema = RemoveProjectDomainResponseBody$inboundSchema;
    /** @deprecated use `RemoveProjectDomainResponseBody$outboundSchema` instead. */
    RemoveProjectDomainResponseBody$.outboundSchema = RemoveProjectDomainResponseBody$outboundSchema;
})(RemoveProjectDomainResponseBody$ || (RemoveProjectDomainResponseBody$ = {}));
export function removeProjectDomainResponseBodyToJSON(removeProjectDomainResponseBody) {
    return JSON.stringify(RemoveProjectDomainResponseBody$outboundSchema.parse(removeProjectDomainResponseBody));
}
export function removeProjectDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectDomainResponseBody' from JSON`);
}
