// @vercel/sdk/esm/mcp-server/tools/deploymentsGetDeploymentEvents.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { deploymentsGetDeploymentEvents } from "../../funcs/deploymentsGetDeploymentEvents.js";
import { GetDeploymentEventsRequest$inboundSchema } from "../../models/getdeploymenteventsop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetDeploymentEventsRequest$inboundSchema,
};
export const tool$deploymentsGetDeploymentEvents = {
    name: "deployments-get-deployment-events",
    description: `Get deployment events

Get the build logs of a deployment by deployment ID and build ID. It can work as an infinite stream of logs or as a JSON endpoint depending on the input parameters.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await deploymentsGetDeploymentEvents(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
