// @vercel/sdk/esm/mcp-server/tools/edgeConfigGetEdgeConfigToken.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigGetEdgeConfigToken } from "../../funcs/edgeConfigGetEdgeConfigToken.js";
import { GetEdgeConfigTokenRequest$inboundSchema } from "../../models/getedgeconfigtokenop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetEdgeConfigTokenRequest$inboundSchema,
};
export const tool$edgeConfigGetEdgeConfigToken = {
    name: "edge-config-get-edge-config-token",
    description: `Get Edge Config token meta data

Return meta data about an Edge Config token.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigGetEdgeConfigToken(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
