// @vercel/sdk/esm/mcp-server/tools/environmentGetV9ProjectsIdOrNameCustomEnvironments.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { environmentGetV9ProjectsIdOrNameCustomEnvironments } from "../../funcs/environmentGetV9ProjectsIdOrNameCustomEnvironments.js";
import { GetV9ProjectsIdOrNameCustomEnvironmentsRequest$inboundSchema } from "../../models/getv9projectsidornamecustomenvironmentsop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetV9ProjectsIdOrNameCustomEnvironmentsRequest$inboundSchema,
};
export const tool$environmentGetV9ProjectsIdOrNameCustomEnvironments = {
    name: "environment-get-v9-projects-id-or-name-custom-environments",
    description: `Retrieve custom environments

Retrieve custom environments for the project. Must not be named 'Production' or 'Preview'.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await environmentGetV9ProjectsIdOrNameCustomEnvironments(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
