// @vercel/sdk/esm/mcp-server/tools/projectsEditProjectEnv.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { projectsEditProjectEnv } from "../../funcs/projectsEditProjectEnv.js";
import { EditProjectEnvRequest$inboundSchema } from "../../models/editprojectenvop.js";
import { formatResult } from "../tools.js";
const args = {
    request: EditProjectEnvRequest$inboundSchema,
};
export const tool$projectsEditProjectEnv = {
    name: "projects-edit-project-env",
    description: `Edit an environment variable

Edit a specific environment variable for a given project by passing the environment variable identifier and either passing the project \`id\` or \`name\` in the URL.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await projectsEditProjectEnv(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
