// @vercel/sdk/esm/mcp-server/tools/rollingReleaseUpdateRollingReleaseConfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { rollingReleaseUpdateRollingReleaseConfig } from "../../funcs/rollingReleaseUpdateRollingReleaseConfig.js";
import { UpdateRollingReleaseConfigRequest$inboundSchema } from "../../models/updaterollingreleaseconfigop.js";
import { formatResult } from "../tools.js";
const args = {
    request: UpdateRollingReleaseConfigRequest$inboundSchema,
};
export const tool$rollingReleaseUpdateRollingReleaseConfig = {
    name: "rolling-release-update-rolling-release-config",
    description: `Update the rolling release settings for the project

Update (or disable) Rolling Releases for a project. Changing the config never alters a rollout that's already in-flight. It only affects the next production deployment. This also applies to disabling Rolling Releases. If you want to also stop the current rollout, call this endpoint to disable the feature, and then call either the /complete or /abort endpoint. Note: Enabling Rolling Releases automatically enables skew protection on the project with the default value if it wasn't configured already.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await rollingReleaseUpdateRollingReleaseConfig(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
