// @vercel/sdk/esm/mcp-server/tools/teamsCreateTeam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { teamsCreateTeam } from "../../funcs/teamsCreateTeam.js";
import { CreateTeamRequestBody$inboundSchema } from "../../models/createteamop.js";
import { formatResult } from "../tools.js";
const args = {
    request: CreateTeamRequestBody$inboundSchema,
};
export const tool$teamsCreateTeam = {
    name: "teams-create-team",
    description: `Create a Team

Create a new Team under your account. You need to send a POST request with the desired Team slug, and optionally the Team name.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await teamsCreateTeam(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
