// @vercel/sdk/esm/mcp-server/tools/teamsGetTeamAccessRequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { teamsGetTeamAccessRequest } from "../../funcs/teamsGetTeamAccessRequest.js";
import { GetTeamAccessRequestRequest$inboundSchema } from "../../models/getteamaccessrequestop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetTeamAccessRequestRequest$inboundSchema,
};
export const tool$teamsGetTeamAccessRequest = {
    name: "teams-get-team-access-request",
    description: `Get access request status

Check the status of a join request. It'll respond with a 404 if the request has been declined. If no \`userId\` path segment was provided, this endpoint will instead return the status of the authenticated user.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await teamsGetTeamAccessRequest(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
