// @vercel/sdk/esm/models/checkdomainstatusop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const CheckDomainStatusRequest$inboundSchema = z.object({
    name: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const CheckDomainStatusRequest$outboundSchema = z.object({
    name: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckDomainStatusRequest$;
(function (CheckDomainStatusRequest$) {
    /** @deprecated use `CheckDomainStatusRequest$inboundSchema` instead. */
    CheckDomainStatusRequest$.inboundSchema = CheckDomainStatusRequest$inboundSchema;
    /** @deprecated use `CheckDomainStatusRequest$outboundSchema` instead. */
    CheckDomainStatusRequest$.outboundSchema = CheckDomainStatusRequest$outboundSchema;
})(CheckDomainStatusRequest$ || (CheckDomainStatusRequest$ = {}));
export function checkDomainStatusRequestToJSON(checkDomainStatusRequest) {
    return JSON.stringify(CheckDomainStatusRequest$outboundSchema.parse(checkDomainStatusRequest));
}
export function checkDomainStatusRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckDomainStatusRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckDomainStatusRequest' from JSON`);
}
/** @internal */
export const CheckDomainStatusResponseBody$inboundSchema = z.object({
    available: z.boolean(),
});
/** @internal */
export const CheckDomainStatusResponseBody$outboundSchema = z.object({
    available: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckDomainStatusResponseBody$;
(function (CheckDomainStatusResponseBody$) {
    /** @deprecated use `CheckDomainStatusResponseBody$inboundSchema` instead. */
    CheckDomainStatusResponseBody$.inboundSchema = CheckDomainStatusResponseBody$inboundSchema;
    /** @deprecated use `CheckDomainStatusResponseBody$outboundSchema` instead. */
    CheckDomainStatusResponseBody$.outboundSchema = CheckDomainStatusResponseBody$outboundSchema;
})(CheckDomainStatusResponseBody$ || (CheckDomainStatusResponseBody$ = {}));
export function checkDomainStatusResponseBodyToJSON(checkDomainStatusResponseBody) {
    return JSON.stringify(CheckDomainStatusResponseBody$outboundSchema.parse(checkDomainStatusResponseBody));
}
export function checkDomainStatusResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckDomainStatusResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckDomainStatusResponseBody' from JSON`);
}
