// @vercel/sdk/esm/models/deleteintegrationresourceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteIntegrationResourceRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
});
/** @internal */
export const DeleteIntegrationResourceRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteIntegrationResourceRequest$;
(function (DeleteIntegrationResourceRequest$) {
    /** @deprecated use `DeleteIntegrationResourceRequest$inboundSchema` instead. */
    DeleteIntegrationResourceRequest$.inboundSchema = DeleteIntegrationResourceRequest$inboundSchema;
    /** @deprecated use `DeleteIntegrationResourceRequest$outboundSchema` instead. */
    DeleteIntegrationResourceRequest$.outboundSchema = DeleteIntegrationResourceRequest$outboundSchema;
})(DeleteIntegrationResourceRequest$ || (DeleteIntegrationResourceRequest$ = {}));
export function deleteIntegrationResourceRequestToJSON(deleteIntegrationResourceRequest) {
    return JSON.stringify(DeleteIntegrationResourceRequest$outboundSchema.parse(deleteIntegrationResourceRequest));
}
export function deleteIntegrationResourceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteIntegrationResourceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteIntegrationResourceRequest' from JSON`);
}
