// @vercel/sdk/esm/models/getdeploymentfilecontentsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetDeploymentFileContentsRequest$inboundSchema = z.object({
    id: z.string(),
    fileId: z.string(),
    path: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetDeploymentFileContentsRequest$outboundSchema = z.object({
    id: z.string(),
    fileId: z.string(),
    path: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDeploymentFileContentsRequest$;
(function (GetDeploymentFileContentsRequest$) {
    /** @deprecated use `GetDeploymentFileContentsRequest$inboundSchema` instead. */
    GetDeploymentFileContentsRequest$.inboundSchema = GetDeploymentFileContentsRequest$inboundSchema;
    /** @deprecated use `GetDeploymentFileContentsRequest$outboundSchema` instead. */
    GetDeploymentFileContentsRequest$.outboundSchema = GetDeploymentFileContentsRequest$outboundSchema;
})(GetDeploymentFileContentsRequest$ || (GetDeploymentFileContentsRequest$ = {}));
export function getDeploymentFileContentsRequestToJSON(getDeploymentFileContentsRequest) {
    return JSON.stringify(GetDeploymentFileContentsRequest$outboundSchema.parse(getDeploymentFileContentsRequest));
}
export function getDeploymentFileContentsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDeploymentFileContentsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDeploymentFileContentsRequest' from JSON`);
}
