// @vercel/sdk/esm/models/getprojectdomainsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/**
 * Filters only production domains when set to `true`.
 */
export const Production = {
    True: "true",
    False: "false",
};
/**
 * Filters on the target of the domain. Can be either \"production\", \"preview\"
 */
export const QueryParamTarget = {
    Production: "production",
    Preview: "preview",
};
/**
 * Excludes redirect project domains when \"false\". Includes redirect project domains when \"true\" (default).
 */
export const Redirects = {
    True: "true",
    False: "false",
};
/**
 * Filters domains based on their verification status.
 */
export const Verified = {
    True: "true",
    False: "false",
};
/**
 * Domains sort order by createdAt
 */
export const Order = {
    Asc: "ASC",
    Desc: "DESC",
};
/** @internal */
export const Production$inboundSchema = z
    .nativeEnum(Production);
/** @internal */
export const Production$outboundSchema = Production$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Production$;
(function (Production$) {
    /** @deprecated use `Production$inboundSchema` instead. */
    Production$.inboundSchema = Production$inboundSchema;
    /** @deprecated use `Production$outboundSchema` instead. */
    Production$.outboundSchema = Production$outboundSchema;
})(Production$ || (Production$ = {}));
/** @internal */
export const QueryParamTarget$inboundSchema = z.nativeEnum(QueryParamTarget);
/** @internal */
export const QueryParamTarget$outboundSchema = QueryParamTarget$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var QueryParamTarget$;
(function (QueryParamTarget$) {
    /** @deprecated use `QueryParamTarget$inboundSchema` instead. */
    QueryParamTarget$.inboundSchema = QueryParamTarget$inboundSchema;
    /** @deprecated use `QueryParamTarget$outboundSchema` instead. */
    QueryParamTarget$.outboundSchema = QueryParamTarget$outboundSchema;
})(QueryParamTarget$ || (QueryParamTarget$ = {}));
/** @internal */
export const Redirects$inboundSchema = z
    .nativeEnum(Redirects);
/** @internal */
export const Redirects$outboundSchema = Redirects$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Redirects$;
(function (Redirects$) {
    /** @deprecated use `Redirects$inboundSchema` instead. */
    Redirects$.inboundSchema = Redirects$inboundSchema;
    /** @deprecated use `Redirects$outboundSchema` instead. */
    Redirects$.outboundSchema = Redirects$outboundSchema;
})(Redirects$ || (Redirects$ = {}));
/** @internal */
export const Verified$inboundSchema = z
    .nativeEnum(Verified);
/** @internal */
export const Verified$outboundSchema = Verified$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Verified$;
(function (Verified$) {
    /** @deprecated use `Verified$inboundSchema` instead. */
    Verified$.inboundSchema = Verified$inboundSchema;
    /** @deprecated use `Verified$outboundSchema` instead. */
    Verified$.outboundSchema = Verified$outboundSchema;
})(Verified$ || (Verified$ = {}));
/** @internal */
export const Order$inboundSchema = z.nativeEnum(Order);
/** @internal */
export const Order$outboundSchema = Order$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Order$;
(function (Order$) {
    /** @deprecated use `Order$inboundSchema` instead. */
    Order$.inboundSchema = Order$inboundSchema;
    /** @deprecated use `Order$outboundSchema` instead. */
    Order$.outboundSchema = Order$outboundSchema;
})(Order$ || (Order$ = {}));
/** @internal */
export const GetProjectDomainsRequest$inboundSchema = z.object({
    idOrName: z.string(),
    production: Production$inboundSchema.default("false"),
    target: QueryParamTarget$inboundSchema.optional(),
    customEnvironmentId: z.string().optional(),
    gitBranch: z.string().optional(),
    redirects: Redirects$inboundSchema.default("true"),
    redirect: z.string().optional(),
    verified: Verified$inboundSchema.optional(),
    limit: z.number().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    order: Order$inboundSchema.default("DESC"),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetProjectDomainsRequest$outboundSchema = z.object({
    idOrName: z.string(),
    production: Production$outboundSchema.default("false"),
    target: QueryParamTarget$outboundSchema.optional(),
    customEnvironmentId: z.string().optional(),
    gitBranch: z.string().optional(),
    redirects: Redirects$outboundSchema.default("true"),
    redirect: z.string().optional(),
    verified: Verified$outboundSchema.optional(),
    limit: z.number().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    order: Order$outboundSchema.default("DESC"),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectDomainsRequest$;
(function (GetProjectDomainsRequest$) {
    /** @deprecated use `GetProjectDomainsRequest$inboundSchema` instead. */
    GetProjectDomainsRequest$.inboundSchema = GetProjectDomainsRequest$inboundSchema;
    /** @deprecated use `GetProjectDomainsRequest$outboundSchema` instead. */
    GetProjectDomainsRequest$.outboundSchema = GetProjectDomainsRequest$outboundSchema;
})(GetProjectDomainsRequest$ || (GetProjectDomainsRequest$ = {}));
export function getProjectDomainsRequestToJSON(getProjectDomainsRequest) {
    return JSON.stringify(GetProjectDomainsRequest$outboundSchema.parse(getProjectDomainsRequest));
}
export function getProjectDomainsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainsRequest' from JSON`);
}
/** @internal */
export const GetProjectDomainsResponseBodyVerification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const GetProjectDomainsResponseBodyVerification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectDomainsResponseBodyVerification$;
(function (GetProjectDomainsResponseBodyVerification$) {
    /** @deprecated use `GetProjectDomainsResponseBodyVerification$inboundSchema` instead. */
    GetProjectDomainsResponseBodyVerification$.inboundSchema = GetProjectDomainsResponseBodyVerification$inboundSchema;
    /** @deprecated use `GetProjectDomainsResponseBodyVerification$outboundSchema` instead. */
    GetProjectDomainsResponseBodyVerification$.outboundSchema = GetProjectDomainsResponseBodyVerification$outboundSchema;
})(GetProjectDomainsResponseBodyVerification$ || (GetProjectDomainsResponseBodyVerification$ = {}));
export function getProjectDomainsResponseBodyVerificationToJSON(getProjectDomainsResponseBodyVerification) {
    return JSON.stringify(GetProjectDomainsResponseBodyVerification$outboundSchema.parse(getProjectDomainsResponseBodyVerification));
}
export function getProjectDomainsResponseBodyVerificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainsResponseBodyVerification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainsResponseBodyVerification' from JSON`);
}
/** @internal */
export const GetProjectDomainsResponseBodyDomains$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => GetProjectDomainsResponseBodyVerification$inboundSchema)).optional(),
});
/** @internal */
export const GetProjectDomainsResponseBodyDomains$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => GetProjectDomainsResponseBodyVerification$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectDomainsResponseBodyDomains$;
(function (GetProjectDomainsResponseBodyDomains$) {
    /** @deprecated use `GetProjectDomainsResponseBodyDomains$inboundSchema` instead. */
    GetProjectDomainsResponseBodyDomains$.inboundSchema = GetProjectDomainsResponseBodyDomains$inboundSchema;
    /** @deprecated use `GetProjectDomainsResponseBodyDomains$outboundSchema` instead. */
    GetProjectDomainsResponseBodyDomains$.outboundSchema = GetProjectDomainsResponseBodyDomains$outboundSchema;
})(GetProjectDomainsResponseBodyDomains$ || (GetProjectDomainsResponseBodyDomains$ = {}));
export function getProjectDomainsResponseBodyDomainsToJSON(getProjectDomainsResponseBodyDomains) {
    return JSON.stringify(GetProjectDomainsResponseBodyDomains$outboundSchema.parse(getProjectDomainsResponseBodyDomains));
}
export function getProjectDomainsResponseBodyDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainsResponseBodyDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainsResponseBodyDomains' from JSON`);
}
/** @internal */
export const GetProjectDomainsResponseBody2$inboundSchema = z.object({
    domains: z.array(z.lazy(() => GetProjectDomainsResponseBodyDomains$inboundSchema)),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const GetProjectDomainsResponseBody2$outboundSchema = z.object({
    domains: z.array(z.lazy(() => GetProjectDomainsResponseBodyDomains$outboundSchema)),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectDomainsResponseBody2$;
(function (GetProjectDomainsResponseBody2$) {
    /** @deprecated use `GetProjectDomainsResponseBody2$inboundSchema` instead. */
    GetProjectDomainsResponseBody2$.inboundSchema = GetProjectDomainsResponseBody2$inboundSchema;
    /** @deprecated use `GetProjectDomainsResponseBody2$outboundSchema` instead. */
    GetProjectDomainsResponseBody2$.outboundSchema = GetProjectDomainsResponseBody2$outboundSchema;
})(GetProjectDomainsResponseBody2$ || (GetProjectDomainsResponseBody2$ = {}));
export function getProjectDomainsResponseBody2ToJSON(getProjectDomainsResponseBody2) {
    return JSON.stringify(GetProjectDomainsResponseBody2$outboundSchema.parse(getProjectDomainsResponseBody2));
}
export function getProjectDomainsResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainsResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainsResponseBody2' from JSON`);
}
/** @internal */
export const ResponseBodyVerification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const ResponseBodyVerification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ResponseBodyVerification$;
(function (ResponseBodyVerification$) {
    /** @deprecated use `ResponseBodyVerification$inboundSchema` instead. */
    ResponseBodyVerification$.inboundSchema = ResponseBodyVerification$inboundSchema;
    /** @deprecated use `ResponseBodyVerification$outboundSchema` instead. */
    ResponseBodyVerification$.outboundSchema = ResponseBodyVerification$outboundSchema;
})(ResponseBodyVerification$ || (ResponseBodyVerification$ = {}));
export function responseBodyVerificationToJSON(responseBodyVerification) {
    return JSON.stringify(ResponseBodyVerification$outboundSchema.parse(responseBodyVerification));
}
export function responseBodyVerificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseBodyVerification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseBodyVerification' from JSON`);
}
/** @internal */
export const ResponseBodyDomains$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => ResponseBodyVerification$inboundSchema))
        .optional(),
});
/** @internal */
export const ResponseBodyDomains$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => ResponseBodyVerification$outboundSchema))
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ResponseBodyDomains$;
(function (ResponseBodyDomains$) {
    /** @deprecated use `ResponseBodyDomains$inboundSchema` instead. */
    ResponseBodyDomains$.inboundSchema = ResponseBodyDomains$inboundSchema;
    /** @deprecated use `ResponseBodyDomains$outboundSchema` instead. */
    ResponseBodyDomains$.outboundSchema = ResponseBodyDomains$outboundSchema;
})(ResponseBodyDomains$ || (ResponseBodyDomains$ = {}));
export function responseBodyDomainsToJSON(responseBodyDomains) {
    return JSON.stringify(ResponseBodyDomains$outboundSchema.parse(responseBodyDomains));
}
export function responseBodyDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseBodyDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseBodyDomains' from JSON`);
}
/** @internal */
export const GetProjectDomainsResponseBodyPagination$inboundSchema = z.object({
    count: z.number(),
    next: z.nullable(z.number()),
    prev: z.nullable(z.number()),
});
/** @internal */
export const GetProjectDomainsResponseBodyPagination$outboundSchema = z.object({
    count: z.number(),
    next: z.nullable(z.number()),
    prev: z.nullable(z.number()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectDomainsResponseBodyPagination$;
(function (GetProjectDomainsResponseBodyPagination$) {
    /** @deprecated use `GetProjectDomainsResponseBodyPagination$inboundSchema` instead. */
    GetProjectDomainsResponseBodyPagination$.inboundSchema = GetProjectDomainsResponseBodyPagination$inboundSchema;
    /** @deprecated use `GetProjectDomainsResponseBodyPagination$outboundSchema` instead. */
    GetProjectDomainsResponseBodyPagination$.outboundSchema = GetProjectDomainsResponseBodyPagination$outboundSchema;
})(GetProjectDomainsResponseBodyPagination$ || (GetProjectDomainsResponseBodyPagination$ = {}));
export function getProjectDomainsResponseBodyPaginationToJSON(getProjectDomainsResponseBodyPagination) {
    return JSON.stringify(GetProjectDomainsResponseBodyPagination$outboundSchema.parse(getProjectDomainsResponseBodyPagination));
}
export function getProjectDomainsResponseBodyPaginationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainsResponseBodyPagination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainsResponseBodyPagination' from JSON`);
}
/** @internal */
export const GetProjectDomainsResponseBody1$inboundSchema = z.object({
    domains: z.array(z.lazy(() => ResponseBodyDomains$inboundSchema)),
    pagination: z.lazy(() => GetProjectDomainsResponseBodyPagination$inboundSchema),
});
/** @internal */
export const GetProjectDomainsResponseBody1$outboundSchema = z.object({
    domains: z.array(z.lazy(() => ResponseBodyDomains$outboundSchema)),
    pagination: z.lazy(() => GetProjectDomainsResponseBodyPagination$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectDomainsResponseBody1$;
(function (GetProjectDomainsResponseBody1$) {
    /** @deprecated use `GetProjectDomainsResponseBody1$inboundSchema` instead. */
    GetProjectDomainsResponseBody1$.inboundSchema = GetProjectDomainsResponseBody1$inboundSchema;
    /** @deprecated use `GetProjectDomainsResponseBody1$outboundSchema` instead. */
    GetProjectDomainsResponseBody1$.outboundSchema = GetProjectDomainsResponseBody1$outboundSchema;
})(GetProjectDomainsResponseBody1$ || (GetProjectDomainsResponseBody1$ = {}));
export function getProjectDomainsResponseBody1ToJSON(getProjectDomainsResponseBody1) {
    return JSON.stringify(GetProjectDomainsResponseBody1$outboundSchema.parse(getProjectDomainsResponseBody1));
}
export function getProjectDomainsResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainsResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainsResponseBody1' from JSON`);
}
/** @internal */
export const GetProjectDomainsResponseBody$inboundSchema = z.union([
    z.lazy(() => GetProjectDomainsResponseBody1$inboundSchema),
    z.lazy(() => GetProjectDomainsResponseBody2$inboundSchema),
]);
/** @internal */
export const GetProjectDomainsResponseBody$outboundSchema = z.union([
    z.lazy(() => GetProjectDomainsResponseBody1$outboundSchema),
    z.lazy(() => GetProjectDomainsResponseBody2$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectDomainsResponseBody$;
(function (GetProjectDomainsResponseBody$) {
    /** @deprecated use `GetProjectDomainsResponseBody$inboundSchema` instead. */
    GetProjectDomainsResponseBody$.inboundSchema = GetProjectDomainsResponseBody$inboundSchema;
    /** @deprecated use `GetProjectDomainsResponseBody$outboundSchema` instead. */
    GetProjectDomainsResponseBody$.outboundSchema = GetProjectDomainsResponseBody$outboundSchema;
})(GetProjectDomainsResponseBody$ || (GetProjectDomainsResponseBody$ = {}));
export function getProjectDomainsResponseBodyToJSON(getProjectDomainsResponseBody) {
    return JSON.stringify(GetProjectDomainsResponseBody$outboundSchema.parse(getProjectDomainsResponseBody));
}
export function getProjectDomainsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainsResponseBody' from JSON`);
}
