// @vercel/sdk/esm/models/putfirewallconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const PutFirewallConfigAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsGenAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsXssAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsSfAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction = {
    Deny: "deny",
    Log: "log",
};
/**
 * [Parameter](https://vercel.com/docs/security/vercel-waf/rule-configuration#parameters) from the incoming traffic.
 */
export const PutFirewallConfigType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Region: "region",
    Protocol: "protocol",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
};
export const Op = {
    Re: "re",
    Eq: "eq",
    Neq: "neq",
    Ex: "ex",
    Nex: "nex",
    Inc: "inc",
    Ninc: "ninc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
};
export const PutFirewallConfigSecurityRequestRequestBodyRulesActionAction = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const Algo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const Action1 = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    RateLimit: "rate_limit",
};
export const PutFirewallConfigSecurityRequestRequestBodyIpsAction = {
    Deny: "deny",
    Challenge: "challenge",
    Log: "log",
    Bypass: "bypass",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Protocol: "protocol",
    Region: "region",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    ServerAction: "server_action",
};
export const PutFirewallConfigOp = {
    Re: "re",
    Eq: "eq",
    Ex: "ex",
    Inc: "inc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
    Nex: "nex",
    Ninc: "ninc",
    Neq: "neq",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const PutFirewallConfigAlgo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    RateLimit: "rate_limit",
};
export const PutFirewallConfigSecurityResponse200Action = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
/** @internal */
export const PutFirewallConfigAction$inboundSchema = z.nativeEnum(PutFirewallConfigAction);
/** @internal */
export const PutFirewallConfigAction$outboundSchema = PutFirewallConfigAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigAction$;
(function (PutFirewallConfigAction$) {
    /** @deprecated use `PutFirewallConfigAction$inboundSchema` instead. */
    PutFirewallConfigAction$.inboundSchema = PutFirewallConfigAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigAction$outboundSchema` instead. */
    PutFirewallConfigAction$.outboundSchema = PutFirewallConfigAction$outboundSchema;
})(PutFirewallConfigAction$ || (PutFirewallConfigAction$ = {}));
/** @internal */
export const Sd$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigAction$inboundSchema,
});
/** @internal */
export const Sd$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Sd$;
(function (Sd$) {
    /** @deprecated use `Sd$inboundSchema` instead. */
    Sd$.inboundSchema = Sd$inboundSchema;
    /** @deprecated use `Sd$outboundSchema` instead. */
    Sd$.outboundSchema = Sd$outboundSchema;
})(Sd$ || (Sd$ = {}));
export function sdToJSON(sd) {
    return JSON.stringify(Sd$outboundSchema.parse(sd));
}
export function sdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Sd$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Sd' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityAction);
/** @internal */
export const PutFirewallConfigSecurityAction$outboundSchema = PutFirewallConfigSecurityAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityAction$;
(function (PutFirewallConfigSecurityAction$) {
    /** @deprecated use `PutFirewallConfigSecurityAction$inboundSchema` instead. */
    PutFirewallConfigSecurityAction$.inboundSchema = PutFirewallConfigSecurityAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityAction$outboundSchema` instead. */
    PutFirewallConfigSecurityAction$.outboundSchema = PutFirewallConfigSecurityAction$outboundSchema;
})(PutFirewallConfigSecurityAction$ || (PutFirewallConfigSecurityAction$ = {}));
/** @internal */
export const Ma$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityAction$inboundSchema,
});
/** @internal */
export const Ma$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Ma$;
(function (Ma$) {
    /** @deprecated use `Ma$inboundSchema` instead. */
    Ma$.inboundSchema = Ma$inboundSchema;
    /** @deprecated use `Ma$outboundSchema` instead. */
    Ma$.outboundSchema = Ma$outboundSchema;
})(Ma$ || (Ma$ = {}));
export function maToJSON(ma) {
    return JSON.stringify(Ma$outboundSchema.parse(ma));
}
export function maFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Ma$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Ma' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestAction);
/** @internal */
export const PutFirewallConfigSecurityRequestAction$outboundSchema = PutFirewallConfigSecurityRequestAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestAction$;
(function (PutFirewallConfigSecurityRequestAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestAction$.inboundSchema = PutFirewallConfigSecurityRequestAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestAction$.outboundSchema = PutFirewallConfigSecurityRequestAction$outboundSchema;
})(PutFirewallConfigSecurityRequestAction$ || (PutFirewallConfigSecurityRequestAction$ = {}));
/** @internal */
export const Lfi$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestAction$inboundSchema,
});
/** @internal */
export const Lfi$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Lfi$;
(function (Lfi$) {
    /** @deprecated use `Lfi$inboundSchema` instead. */
    Lfi$.inboundSchema = Lfi$inboundSchema;
    /** @deprecated use `Lfi$outboundSchema` instead. */
    Lfi$.outboundSchema = Lfi$outboundSchema;
})(Lfi$ || (Lfi$ = {}));
export function lfiToJSON(lfi) {
    return JSON.stringify(Lfi$outboundSchema.parse(lfi));
}
export function lfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Lfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Lfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema = z
    .nativeEnum(PutFirewallConfigSecurityRequestRequestBodyAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyAction$ || (PutFirewallConfigSecurityRequestRequestBodyAction$ = {}));
/** @internal */
export const Rfi$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema,
});
/** @internal */
export const Rfi$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Rfi$;
(function (Rfi$) {
    /** @deprecated use `Rfi$inboundSchema` instead. */
    Rfi$.inboundSchema = Rfi$inboundSchema;
    /** @deprecated use `Rfi$outboundSchema` instead. */
    Rfi$.outboundSchema = Rfi$outboundSchema;
})(Rfi$ || (Rfi$ = {}));
export function rfiToJSON(rfi) {
    return JSON.stringify(Rfi$outboundSchema.parse(rfi));
}
export function rfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Rfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyCrsAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyCrsAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyCrsAction$ || (PutFirewallConfigSecurityRequestRequestBodyCrsAction$ = {}));
/** @internal */
export const Rce$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema,
});
/** @internal */
export const Rce$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Rce$;
(function (Rce$) {
    /** @deprecated use `Rce$inboundSchema` instead. */
    Rce$.inboundSchema = Rce$inboundSchema;
    /** @deprecated use `Rce$outboundSchema` instead. */
    Rce$.outboundSchema = Rce$outboundSchema;
})(Rce$ || (Rce$ = {}));
export function rceToJSON(rce) {
    return JSON.stringify(Rce$outboundSchema.parse(rce));
}
export function rceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Rce$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rce' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$ || (PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$ = {}));
/** @internal */
export const Php$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema,
});
/** @internal */
export const Php$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Php$;
(function (Php$) {
    /** @deprecated use `Php$inboundSchema` instead. */
    Php$.inboundSchema = Php$inboundSchema;
    /** @deprecated use `Php$outboundSchema` instead. */
    Php$.outboundSchema = Php$outboundSchema;
})(Php$ || (Php$ = {}));
export function phpToJSON(php) {
    return JSON.stringify(Php$outboundSchema.parse(php));
}
export function phpFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Php$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Php' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsGenAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$ || (PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$ = {}));
/** @internal */
export const Gen$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema,
});
/** @internal */
export const Gen$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Gen$;
(function (Gen$) {
    /** @deprecated use `Gen$inboundSchema` instead. */
    Gen$.inboundSchema = Gen$inboundSchema;
    /** @deprecated use `Gen$outboundSchema` instead. */
    Gen$.outboundSchema = Gen$outboundSchema;
})(Gen$ || (Gen$ = {}));
export function genToJSON(gen) {
    return JSON.stringify(Gen$outboundSchema.parse(gen));
}
export function genFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Gen$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Gen' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsXssAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$ || (PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$ = {}));
/** @internal */
export const Xss$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema,
});
/** @internal */
export const Xss$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Xss$;
(function (Xss$) {
    /** @deprecated use `Xss$inboundSchema` instead. */
    Xss$.inboundSchema = Xss$inboundSchema;
    /** @deprecated use `Xss$outboundSchema` instead. */
    Xss$.outboundSchema = Xss$outboundSchema;
})(Xss$ || (Xss$ = {}));
export function xssToJSON(xss) {
    return JSON.stringify(Xss$outboundSchema.parse(xss));
}
export function xssFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Xss$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Xss' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$ || (PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$ = {}));
/** @internal */
export const Sqli$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema,
});
/** @internal */
export const Sqli$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Sqli$;
(function (Sqli$) {
    /** @deprecated use `Sqli$inboundSchema` instead. */
    Sqli$.inboundSchema = Sqli$inboundSchema;
    /** @deprecated use `Sqli$outboundSchema` instead. */
    Sqli$.outboundSchema = Sqli$outboundSchema;
})(Sqli$ || (Sqli$ = {}));
export function sqliToJSON(sqli) {
    return JSON.stringify(Sqli$outboundSchema.parse(sqli));
}
export function sqliFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Sqli$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Sqli' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsSfAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$ || (PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$ = {}));
/** @internal */
export const Sf$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema,
});
/** @internal */
export const Sf$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Sf$;
(function (Sf$) {
    /** @deprecated use `Sf$inboundSchema` instead. */
    Sf$.inboundSchema = Sf$inboundSchema;
    /** @deprecated use `Sf$outboundSchema` instead. */
    Sf$.outboundSchema = Sf$outboundSchema;
})(Sf$ || (Sf$ = {}));
export function sfToJSON(sf) {
    return JSON.stringify(Sf$outboundSchema.parse(sf));
}
export function sfFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Sf$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Sf' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$ || (PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$ = {}));
/** @internal */
export const Java$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema,
});
/** @internal */
export const Java$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Java$;
(function (Java$) {
    /** @deprecated use `Java$inboundSchema` instead. */
    Java$.inboundSchema = Java$inboundSchema;
    /** @deprecated use `Java$outboundSchema` instead. */
    Java$.outboundSchema = Java$outboundSchema;
})(Java$ || (Java$ = {}));
export function javaToJSON(java) {
    return JSON.stringify(Java$outboundSchema.parse(java));
}
export function javaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Java$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Java' from JSON`);
}
/** @internal */
export const Crs$inboundSchema = z
    .object({
    sd: z.lazy(() => Sd$inboundSchema).optional(),
    ma: z.lazy(() => Ma$inboundSchema).optional(),
    lfi: z.lazy(() => Lfi$inboundSchema).optional(),
    rfi: z.lazy(() => Rfi$inboundSchema).optional(),
    rce: z.lazy(() => Rce$inboundSchema).optional(),
    php: z.lazy(() => Php$inboundSchema).optional(),
    gen: z.lazy(() => Gen$inboundSchema).optional(),
    xss: z.lazy(() => Xss$inboundSchema).optional(),
    sqli: z.lazy(() => Sqli$inboundSchema).optional(),
    sf: z.lazy(() => Sf$inboundSchema).optional(),
    java: z.lazy(() => Java$inboundSchema).optional(),
});
/** @internal */
export const Crs$outboundSchema = z
    .object({
    sd: z.lazy(() => Sd$outboundSchema).optional(),
    ma: z.lazy(() => Ma$outboundSchema).optional(),
    lfi: z.lazy(() => Lfi$outboundSchema).optional(),
    rfi: z.lazy(() => Rfi$outboundSchema).optional(),
    rce: z.lazy(() => Rce$outboundSchema).optional(),
    php: z.lazy(() => Php$outboundSchema).optional(),
    gen: z.lazy(() => Gen$outboundSchema).optional(),
    xss: z.lazy(() => Xss$outboundSchema).optional(),
    sqli: z.lazy(() => Sqli$outboundSchema).optional(),
    sf: z.lazy(() => Sf$outboundSchema).optional(),
    java: z.lazy(() => Java$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Crs$;
(function (Crs$) {
    /** @deprecated use `Crs$inboundSchema` instead. */
    Crs$.inboundSchema = Crs$inboundSchema;
    /** @deprecated use `Crs$outboundSchema` instead. */
    Crs$.outboundSchema = Crs$outboundSchema;
})(Crs$ || (Crs$ = {}));
export function crsToJSON(crs) {
    return JSON.stringify(Crs$outboundSchema.parse(crs));
}
export function crsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Crs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Crs' from JSON`);
}
/** @internal */
export const PutFirewallConfigType$inboundSchema = z.nativeEnum(PutFirewallConfigType);
/** @internal */
export const PutFirewallConfigType$outboundSchema = PutFirewallConfigType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigType$;
(function (PutFirewallConfigType$) {
    /** @deprecated use `PutFirewallConfigType$inboundSchema` instead. */
    PutFirewallConfigType$.inboundSchema = PutFirewallConfigType$inboundSchema;
    /** @deprecated use `PutFirewallConfigType$outboundSchema` instead. */
    PutFirewallConfigType$.outboundSchema = PutFirewallConfigType$outboundSchema;
})(PutFirewallConfigType$ || (PutFirewallConfigType$ = {}));
/** @internal */
export const Op$inboundSchema = z.nativeEnum(Op);
/** @internal */
export const Op$outboundSchema = Op$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Op$;
(function (Op$) {
    /** @deprecated use `Op$inboundSchema` instead. */
    Op$.inboundSchema = Op$inboundSchema;
    /** @deprecated use `Op$outboundSchema` instead. */
    Op$.outboundSchema = Op$outboundSchema;
})(Op$ || (Op$ = {}));
/** @internal */
export const Value$inboundSchema = z
    .union([z.string(), z.array(z.string()), z.number()]);
/** @internal */
export const Value$outboundSchema = z.union([z.string(), z.array(z.string()), z.number()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Value$;
(function (Value$) {
    /** @deprecated use `Value$inboundSchema` instead. */
    Value$.inboundSchema = Value$inboundSchema;
    /** @deprecated use `Value$outboundSchema` instead. */
    Value$.outboundSchema = Value$outboundSchema;
})(Value$ || (Value$ = {}));
export function valueToJSON(value) {
    return JSON.stringify(Value$outboundSchema.parse(value));
}
export function valueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Value' from JSON`);
}
/** @internal */
export const Conditions$inboundSchema = z.object({
    type: PutFirewallConfigType$inboundSchema,
    op: Op$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
/** @internal */
export const Conditions$outboundSchema = z.object({
    type: PutFirewallConfigType$outboundSchema,
    op: Op$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Conditions$;
(function (Conditions$) {
    /** @deprecated use `Conditions$inboundSchema` instead. */
    Conditions$.inboundSchema = Conditions$inboundSchema;
    /** @deprecated use `Conditions$outboundSchema` instead. */
    Conditions$.outboundSchema = Conditions$outboundSchema;
})(Conditions$ || (Conditions$ = {}));
export function conditionsToJSON(conditions) {
    return JSON.stringify(Conditions$outboundSchema.parse(conditions));
}
export function conditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Conditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Conditions' from JSON`);
}
/** @internal */
export const ConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => Conditions$inboundSchema)),
});
/** @internal */
export const ConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => Conditions$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ConditionGroup$;
(function (ConditionGroup$) {
    /** @deprecated use `ConditionGroup$inboundSchema` instead. */
    ConditionGroup$.inboundSchema = ConditionGroup$inboundSchema;
    /** @deprecated use `ConditionGroup$outboundSchema` instead. */
    ConditionGroup$.outboundSchema = ConditionGroup$outboundSchema;
})(ConditionGroup$ || (ConditionGroup$ = {}));
export function conditionGroupToJSON(conditionGroup) {
    return JSON.stringify(ConditionGroup$outboundSchema.parse(conditionGroup));
}
export function conditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ConditionGroup' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyRulesActionAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$ || (PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$ = {}));
/** @internal */
export const Algo$inboundSchema = z.nativeEnum(Algo);
/** @internal */
export const Algo$outboundSchema = Algo$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Algo$;
(function (Algo$) {
    /** @deprecated use `Algo$inboundSchema` instead. */
    Algo$.inboundSchema = Algo$inboundSchema;
    /** @deprecated use `Algo$outboundSchema` instead. */
    Algo$.outboundSchema = Algo$outboundSchema;
})(Algo$ || (Algo$ = {}));
/** @internal */
export const Action1$inboundSchema = z
    .nativeEnum(Action1);
/** @internal */
export const Action1$outboundSchema = Action1$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Action1$;
(function (Action1$) {
    /** @deprecated use `Action1$inboundSchema` instead. */
    Action1$.inboundSchema = Action1$inboundSchema;
    /** @deprecated use `Action1$outboundSchema` instead. */
    Action1$.outboundSchema = Action1$outboundSchema;
})(Action1$ || (Action1$ = {}));
/** @internal */
export const RateLimitAction$inboundSchema = z.union([Action1$inboundSchema, z.any()]);
/** @internal */
export const RateLimitAction$outboundSchema = z.union([Action1$outboundSchema, z.any()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RateLimitAction$;
(function (RateLimitAction$) {
    /** @deprecated use `RateLimitAction$inboundSchema` instead. */
    RateLimitAction$.inboundSchema = RateLimitAction$inboundSchema;
    /** @deprecated use `RateLimitAction$outboundSchema` instead. */
    RateLimitAction$.outboundSchema = RateLimitAction$outboundSchema;
})(RateLimitAction$ || (RateLimitAction$ = {}));
export function rateLimitActionToJSON(rateLimitAction) {
    return JSON.stringify(RateLimitAction$outboundSchema.parse(rateLimitAction));
}
export function rateLimitActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RateLimitAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RateLimitAction' from JSON`);
}
/** @internal */
export const RateLimit1$inboundSchema = z.object({
    algo: Algo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([Action1$inboundSchema, z.any()])).optional(),
});
/** @internal */
export const RateLimit1$outboundSchema = z.object({
    algo: Algo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([Action1$outboundSchema, z.any()])).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RateLimit1$;
(function (RateLimit1$) {
    /** @deprecated use `RateLimit1$inboundSchema` instead. */
    RateLimit1$.inboundSchema = RateLimit1$inboundSchema;
    /** @deprecated use `RateLimit1$outboundSchema` instead. */
    RateLimit1$.outboundSchema = RateLimit1$outboundSchema;
})(RateLimit1$ || (RateLimit1$ = {}));
export function rateLimit1ToJSON(rateLimit1) {
    return JSON.stringify(RateLimit1$outboundSchema.parse(rateLimit1));
}
export function rateLimit1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RateLimit1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RateLimit1' from JSON`);
}
/** @internal */
export const RateLimit$inboundSchema = z.union([z.lazy(() => RateLimit1$inboundSchema), z.any()]);
/** @internal */
export const RateLimit$outboundSchema = z.union([z.lazy(() => RateLimit1$outboundSchema), z.any()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RateLimit$;
(function (RateLimit$) {
    /** @deprecated use `RateLimit$inboundSchema` instead. */
    RateLimit$.inboundSchema = RateLimit$inboundSchema;
    /** @deprecated use `RateLimit$outboundSchema` instead. */
    RateLimit$.outboundSchema = RateLimit$outboundSchema;
})(RateLimit$ || (RateLimit$ = {}));
export function rateLimitToJSON(rateLimit) {
    return JSON.stringify(RateLimit$outboundSchema.parse(rateLimit));
}
export function rateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RateLimit' from JSON`);
}
/** @internal */
export const Redirect1$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const Redirect1$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Redirect1$;
(function (Redirect1$) {
    /** @deprecated use `Redirect1$inboundSchema` instead. */
    Redirect1$.inboundSchema = Redirect1$inboundSchema;
    /** @deprecated use `Redirect1$outboundSchema` instead. */
    Redirect1$.outboundSchema = Redirect1$outboundSchema;
})(Redirect1$ || (Redirect1$ = {}));
export function redirect1ToJSON(redirect1) {
    return JSON.stringify(Redirect1$outboundSchema.parse(redirect1));
}
export function redirect1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Redirect1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Redirect1' from JSON`);
}
/** @internal */
export const Redirect$inboundSchema = z.union([z.lazy(() => Redirect1$inboundSchema), z.any()]);
/** @internal */
export const Redirect$outboundSchema = z.union([z.lazy(() => Redirect1$outboundSchema), z.any()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Redirect$;
(function (Redirect$) {
    /** @deprecated use `Redirect$inboundSchema` instead. */
    Redirect$.inboundSchema = Redirect$inboundSchema;
    /** @deprecated use `Redirect$outboundSchema` instead. */
    Redirect$.outboundSchema = Redirect$outboundSchema;
})(Redirect$ || (Redirect$ = {}));
export function redirectToJSON(redirect) {
    return JSON.stringify(Redirect$outboundSchema.parse(redirect));
}
export function redirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Redirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Redirect' from JSON`);
}
/** @internal */
export const Mitigate$inboundSchema = z.object({
    action: PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema,
    rateLimit: z.nullable(z.union([z.lazy(() => RateLimit1$inboundSchema), z.any()])).optional(),
    redirect: z.nullable(z.union([z.lazy(() => Redirect1$inboundSchema), z.any()])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const Mitigate$outboundSchema = z.object({
    action: PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$outboundSchema,
    rateLimit: z.nullable(z.union([z.lazy(() => RateLimit1$outboundSchema), z.any()])).optional(),
    redirect: z.nullable(z.union([z.lazy(() => Redirect1$outboundSchema), z.any()])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Mitigate$;
(function (Mitigate$) {
    /** @deprecated use `Mitigate$inboundSchema` instead. */
    Mitigate$.inboundSchema = Mitigate$inboundSchema;
    /** @deprecated use `Mitigate$outboundSchema` instead. */
    Mitigate$.outboundSchema = Mitigate$outboundSchema;
})(Mitigate$ || (Mitigate$ = {}));
export function mitigateToJSON(mitigate) {
    return JSON.stringify(Mitigate$outboundSchema.parse(mitigate));
}
export function mitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Mitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Mitigate' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema = z.object({
    mitigate: z.lazy(() => Mitigate$inboundSchema).optional(),
});
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema = z.object({
    mitigate: z.lazy(() => Mitigate$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyRulesAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyRulesAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyRulesAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyRulesAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyRulesAction$ || (PutFirewallConfigSecurityRequestRequestBodyRulesAction$ = {}));
export function putFirewallConfigSecurityRequestRequestBodyRulesActionToJSON(putFirewallConfigSecurityRequestRequestBodyRulesAction) {
    return JSON.stringify(PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema.parse(putFirewallConfigSecurityRequestRequestBodyRulesAction));
}
export function putFirewallConfigSecurityRequestRequestBodyRulesActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSecurityRequestRequestBodyRulesAction' from JSON`);
}
/** @internal */
export const PutFirewallConfigRules$inboundSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => ConditionGroup$inboundSchema)),
    action: z.lazy(() => PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema),
});
/** @internal */
export const PutFirewallConfigRules$outboundSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => ConditionGroup$outboundSchema)),
    action: z.lazy(() => PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigRules$;
(function (PutFirewallConfigRules$) {
    /** @deprecated use `PutFirewallConfigRules$inboundSchema` instead. */
    PutFirewallConfigRules$.inboundSchema = PutFirewallConfigRules$inboundSchema;
    /** @deprecated use `PutFirewallConfigRules$outboundSchema` instead. */
    PutFirewallConfigRules$.outboundSchema = PutFirewallConfigRules$outboundSchema;
})(PutFirewallConfigRules$ || (PutFirewallConfigRules$ = {}));
export function putFirewallConfigRulesToJSON(putFirewallConfigRules) {
    return JSON.stringify(PutFirewallConfigRules$outboundSchema.parse(putFirewallConfigRules));
}
export function putFirewallConfigRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRules' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyIpsAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyIpsAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRequestRequestBodyIpsAction$;
(function (PutFirewallConfigSecurityRequestRequestBodyIpsAction$) {
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyIpsAction$.inboundSchema = PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRequestRequestBodyIpsAction$outboundSchema` instead. */
    PutFirewallConfigSecurityRequestRequestBodyIpsAction$.outboundSchema = PutFirewallConfigSecurityRequestRequestBodyIpsAction$outboundSchema;
})(PutFirewallConfigSecurityRequestRequestBodyIpsAction$ || (PutFirewallConfigSecurityRequestRequestBodyIpsAction$ = {}));
/** @internal */
export const Ips$inboundSchema = z
    .object({
    id: z.string().optional(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema,
});
/** @internal */
export const Ips$outboundSchema = z
    .object({
    id: z.string().optional(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityRequestRequestBodyIpsAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Ips$;
(function (Ips$) {
    /** @deprecated use `Ips$inboundSchema` instead. */
    Ips$.inboundSchema = Ips$inboundSchema;
    /** @deprecated use `Ips$outboundSchema` instead. */
    Ips$.outboundSchema = Ips$outboundSchema;
})(Ips$ || (Ips$ = {}));
export function ipsToJSON(ips) {
    return JSON.stringify(Ips$outboundSchema.parse(ips));
}
export function ipsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Ips$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Ips' from JSON`);
}
/** @internal */
export const PutFirewallConfigRequestBody$inboundSchema = z.object({
    firewallEnabled: z.boolean(),
    managedRules: z.record(z.any()).optional(),
    crs: z.lazy(() => Crs$inboundSchema).optional(),
    rules: z.array(z.lazy(() => PutFirewallConfigRules$inboundSchema)).optional(),
    ips: z.array(z.lazy(() => Ips$inboundSchema)).optional(),
    botIdEnabled: z.boolean().optional(),
});
/** @internal */
export const PutFirewallConfigRequestBody$outboundSchema = z.object({
    firewallEnabled: z.boolean(),
    managedRules: z.record(z.any()).optional(),
    crs: z.lazy(() => Crs$outboundSchema).optional(),
    rules: z.array(z.lazy(() => PutFirewallConfigRules$outboundSchema))
        .optional(),
    ips: z.array(z.lazy(() => Ips$outboundSchema)).optional(),
    botIdEnabled: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigRequestBody$;
(function (PutFirewallConfigRequestBody$) {
    /** @deprecated use `PutFirewallConfigRequestBody$inboundSchema` instead. */
    PutFirewallConfigRequestBody$.inboundSchema = PutFirewallConfigRequestBody$inboundSchema;
    /** @deprecated use `PutFirewallConfigRequestBody$outboundSchema` instead. */
    PutFirewallConfigRequestBody$.outboundSchema = PutFirewallConfigRequestBody$outboundSchema;
})(PutFirewallConfigRequestBody$ || (PutFirewallConfigRequestBody$ = {}));
export function putFirewallConfigRequestBodyToJSON(putFirewallConfigRequestBody) {
    return JSON.stringify(PutFirewallConfigRequestBody$outboundSchema.parse(putFirewallConfigRequestBody));
}
export function putFirewallConfigRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRequestBody' from JSON`);
}
/** @internal */
export const PutFirewallConfigRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => PutFirewallConfigRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const PutFirewallConfigRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => PutFirewallConfigRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigRequest$;
(function (PutFirewallConfigRequest$) {
    /** @deprecated use `PutFirewallConfigRequest$inboundSchema` instead. */
    PutFirewallConfigRequest$.inboundSchema = PutFirewallConfigRequest$inboundSchema;
    /** @deprecated use `PutFirewallConfigRequest$outboundSchema` instead. */
    PutFirewallConfigRequest$.outboundSchema = PutFirewallConfigRequest$outboundSchema;
})(PutFirewallConfigRequest$ || (PutFirewallConfigRequest$ = {}));
export function putFirewallConfigRequestToJSON(putFirewallConfigRequest) {
    return JSON.stringify(PutFirewallConfigRequest$outboundSchema.parse(putFirewallConfigRequest));
}
export function putFirewallConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRequest' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$ = {}));
/** @internal */
export const PutFirewallConfigSd$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigSd$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSd$;
(function (PutFirewallConfigSd$) {
    /** @deprecated use `PutFirewallConfigSd$inboundSchema` instead. */
    PutFirewallConfigSd$.inboundSchema = PutFirewallConfigSd$inboundSchema;
    /** @deprecated use `PutFirewallConfigSd$outboundSchema` instead. */
    PutFirewallConfigSd$.outboundSchema = PutFirewallConfigSd$outboundSchema;
})(PutFirewallConfigSd$ || (PutFirewallConfigSd$ = {}));
export function putFirewallConfigSdToJSON(putFirewallConfigSd) {
    return JSON.stringify(PutFirewallConfigSd$outboundSchema.parse(putFirewallConfigSd));
}
export function putFirewallConfigSdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSd$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSd' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$ = {}));
/** @internal */
export const PutFirewallConfigMa$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigMa$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsMaAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigMa$;
(function (PutFirewallConfigMa$) {
    /** @deprecated use `PutFirewallConfigMa$inboundSchema` instead. */
    PutFirewallConfigMa$.inboundSchema = PutFirewallConfigMa$inboundSchema;
    /** @deprecated use `PutFirewallConfigMa$outboundSchema` instead. */
    PutFirewallConfigMa$.outboundSchema = PutFirewallConfigMa$outboundSchema;
})(PutFirewallConfigMa$ || (PutFirewallConfigMa$ = {}));
export function putFirewallConfigMaToJSON(putFirewallConfigMa) {
    return JSON.stringify(PutFirewallConfigMa$outboundSchema.parse(putFirewallConfigMa));
}
export function putFirewallConfigMaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigMa$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigMa' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONAction$ = {}));
/** @internal */
export const PutFirewallConfigLfi$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigLfi$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigLfi$;
(function (PutFirewallConfigLfi$) {
    /** @deprecated use `PutFirewallConfigLfi$inboundSchema` instead. */
    PutFirewallConfigLfi$.inboundSchema = PutFirewallConfigLfi$inboundSchema;
    /** @deprecated use `PutFirewallConfigLfi$outboundSchema` instead. */
    PutFirewallConfigLfi$.outboundSchema = PutFirewallConfigLfi$outboundSchema;
})(PutFirewallConfigLfi$ || (PutFirewallConfigLfi$ = {}));
export function putFirewallConfigLfiToJSON(putFirewallConfigLfi) {
    return JSON.stringify(PutFirewallConfigLfi$outboundSchema.parse(putFirewallConfigLfi));
}
export function putFirewallConfigLfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigLfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigLfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$ = {}));
/** @internal */
export const PutFirewallConfigRfi$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigRfi$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigRfi$;
(function (PutFirewallConfigRfi$) {
    /** @deprecated use `PutFirewallConfigRfi$inboundSchema` instead. */
    PutFirewallConfigRfi$.inboundSchema = PutFirewallConfigRfi$inboundSchema;
    /** @deprecated use `PutFirewallConfigRfi$outboundSchema` instead. */
    PutFirewallConfigRfi$.outboundSchema = PutFirewallConfigRfi$outboundSchema;
})(PutFirewallConfigRfi$ || (PutFirewallConfigRfi$ = {}));
export function putFirewallConfigRfiToJSON(putFirewallConfigRfi) {
    return JSON.stringify(PutFirewallConfigRfi$outboundSchema.parse(putFirewallConfigRfi));
}
export function putFirewallConfigRfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$ = {}));
/** @internal */
export const PutFirewallConfigRce$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigRce$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigRce$;
(function (PutFirewallConfigRce$) {
    /** @deprecated use `PutFirewallConfigRce$inboundSchema` instead. */
    PutFirewallConfigRce$.inboundSchema = PutFirewallConfigRce$inboundSchema;
    /** @deprecated use `PutFirewallConfigRce$outboundSchema` instead. */
    PutFirewallConfigRce$.outboundSchema = PutFirewallConfigRce$outboundSchema;
})(PutFirewallConfigRce$ || (PutFirewallConfigRce$ = {}));
export function putFirewallConfigRceToJSON(putFirewallConfigRce) {
    return JSON.stringify(PutFirewallConfigRce$outboundSchema.parse(putFirewallConfigRce));
}
export function putFirewallConfigRceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRce$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRce' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$ = {}));
/** @internal */
export const PutFirewallConfigPhp$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigPhp$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigPhp$;
(function (PutFirewallConfigPhp$) {
    /** @deprecated use `PutFirewallConfigPhp$inboundSchema` instead. */
    PutFirewallConfigPhp$.inboundSchema = PutFirewallConfigPhp$inboundSchema;
    /** @deprecated use `PutFirewallConfigPhp$outboundSchema` instead. */
    PutFirewallConfigPhp$.outboundSchema = PutFirewallConfigPhp$outboundSchema;
})(PutFirewallConfigPhp$ || (PutFirewallConfigPhp$ = {}));
export function putFirewallConfigPhpToJSON(putFirewallConfigPhp) {
    return JSON.stringify(PutFirewallConfigPhp$outboundSchema.parse(putFirewallConfigPhp));
}
export function putFirewallConfigPhpFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigPhp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigPhp' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$ = {}));
/** @internal */
export const PutFirewallConfigGen$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigGen$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigGen$;
(function (PutFirewallConfigGen$) {
    /** @deprecated use `PutFirewallConfigGen$inboundSchema` instead. */
    PutFirewallConfigGen$.inboundSchema = PutFirewallConfigGen$inboundSchema;
    /** @deprecated use `PutFirewallConfigGen$outboundSchema` instead. */
    PutFirewallConfigGen$.outboundSchema = PutFirewallConfigGen$outboundSchema;
})(PutFirewallConfigGen$ || (PutFirewallConfigGen$ = {}));
export function putFirewallConfigGenToJSON(putFirewallConfigGen) {
    return JSON.stringify(PutFirewallConfigGen$outboundSchema.parse(putFirewallConfigGen));
}
export function putFirewallConfigGenFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigGen$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigGen' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$ = {}));
/** @internal */
export const PutFirewallConfigXss$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigXss$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigXss$;
(function (PutFirewallConfigXss$) {
    /** @deprecated use `PutFirewallConfigXss$inboundSchema` instead. */
    PutFirewallConfigXss$.inboundSchema = PutFirewallConfigXss$inboundSchema;
    /** @deprecated use `PutFirewallConfigXss$outboundSchema` instead. */
    PutFirewallConfigXss$.outboundSchema = PutFirewallConfigXss$outboundSchema;
})(PutFirewallConfigXss$ || (PutFirewallConfigXss$ = {}));
export function putFirewallConfigXssToJSON(putFirewallConfigXss) {
    return JSON.stringify(PutFirewallConfigXss$outboundSchema.parse(putFirewallConfigXss));
}
export function putFirewallConfigXssFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigXss$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigXss' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$ = {}));
/** @internal */
export const PutFirewallConfigSqli$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigSqli$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSqli$;
(function (PutFirewallConfigSqli$) {
    /** @deprecated use `PutFirewallConfigSqli$inboundSchema` instead. */
    PutFirewallConfigSqli$.inboundSchema = PutFirewallConfigSqli$inboundSchema;
    /** @deprecated use `PutFirewallConfigSqli$outboundSchema` instead. */
    PutFirewallConfigSqli$.outboundSchema = PutFirewallConfigSqli$outboundSchema;
})(PutFirewallConfigSqli$ || (PutFirewallConfigSqli$ = {}));
export function putFirewallConfigSqliToJSON(putFirewallConfigSqli) {
    return JSON.stringify(PutFirewallConfigSqli$outboundSchema.parse(putFirewallConfigSqli));
}
export function putFirewallConfigSqliFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSqli$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSqli' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$ = {}));
/** @internal */
export const PutFirewallConfigSf$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigSf$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSf$;
(function (PutFirewallConfigSf$) {
    /** @deprecated use `PutFirewallConfigSf$inboundSchema` instead. */
    PutFirewallConfigSf$.inboundSchema = PutFirewallConfigSf$inboundSchema;
    /** @deprecated use `PutFirewallConfigSf$outboundSchema` instead. */
    PutFirewallConfigSf$.outboundSchema = PutFirewallConfigSf$outboundSchema;
})(PutFirewallConfigSf$ || (PutFirewallConfigSf$ = {}));
export function putFirewallConfigSfToJSON(putFirewallConfigSf) {
    return JSON.stringify(PutFirewallConfigSf$outboundSchema.parse(putFirewallConfigSf));
}
export function putFirewallConfigSfFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSf$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSf' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$ = {}));
/** @internal */
export const PutFirewallConfigJava$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigJava$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigJava$;
(function (PutFirewallConfigJava$) {
    /** @deprecated use `PutFirewallConfigJava$inboundSchema` instead. */
    PutFirewallConfigJava$.inboundSchema = PutFirewallConfigJava$inboundSchema;
    /** @deprecated use `PutFirewallConfigJava$outboundSchema` instead. */
    PutFirewallConfigJava$.outboundSchema = PutFirewallConfigJava$outboundSchema;
})(PutFirewallConfigJava$ || (PutFirewallConfigJava$ = {}));
export function putFirewallConfigJavaToJSON(putFirewallConfigJava) {
    return JSON.stringify(PutFirewallConfigJava$outboundSchema.parse(putFirewallConfigJava));
}
export function putFirewallConfigJavaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigJava$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigJava' from JSON`);
}
/** @internal */
export const PutFirewallConfigCrs$inboundSchema = z.object({
    sd: z.lazy(() => PutFirewallConfigSd$inboundSchema),
    ma: z.lazy(() => PutFirewallConfigMa$inboundSchema),
    lfi: z.lazy(() => PutFirewallConfigLfi$inboundSchema),
    rfi: z.lazy(() => PutFirewallConfigRfi$inboundSchema),
    rce: z.lazy(() => PutFirewallConfigRce$inboundSchema),
    php: z.lazy(() => PutFirewallConfigPhp$inboundSchema),
    gen: z.lazy(() => PutFirewallConfigGen$inboundSchema),
    xss: z.lazy(() => PutFirewallConfigXss$inboundSchema),
    sqli: z.lazy(() => PutFirewallConfigSqli$inboundSchema),
    sf: z.lazy(() => PutFirewallConfigSf$inboundSchema),
    java: z.lazy(() => PutFirewallConfigJava$inboundSchema),
});
/** @internal */
export const PutFirewallConfigCrs$outboundSchema = z.object({
    sd: z.lazy(() => PutFirewallConfigSd$outboundSchema),
    ma: z.lazy(() => PutFirewallConfigMa$outboundSchema),
    lfi: z.lazy(() => PutFirewallConfigLfi$outboundSchema),
    rfi: z.lazy(() => PutFirewallConfigRfi$outboundSchema),
    rce: z.lazy(() => PutFirewallConfigRce$outboundSchema),
    php: z.lazy(() => PutFirewallConfigPhp$outboundSchema),
    gen: z.lazy(() => PutFirewallConfigGen$outboundSchema),
    xss: z.lazy(() => PutFirewallConfigXss$outboundSchema),
    sqli: z.lazy(() => PutFirewallConfigSqli$outboundSchema),
    sf: z.lazy(() => PutFirewallConfigSf$outboundSchema),
    java: z.lazy(() => PutFirewallConfigJava$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigCrs$;
(function (PutFirewallConfigCrs$) {
    /** @deprecated use `PutFirewallConfigCrs$inboundSchema` instead. */
    PutFirewallConfigCrs$.inboundSchema = PutFirewallConfigCrs$inboundSchema;
    /** @deprecated use `PutFirewallConfigCrs$outboundSchema` instead. */
    PutFirewallConfigCrs$.outboundSchema = PutFirewallConfigCrs$outboundSchema;
})(PutFirewallConfigCrs$ || (PutFirewallConfigCrs$ = {}));
export function putFirewallConfigCrsToJSON(putFirewallConfigCrs) {
    return JSON.stringify(PutFirewallConfigCrs$outboundSchema.parse(putFirewallConfigCrs));
}
export function putFirewallConfigCrsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigCrs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigCrs' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityType$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityType);
/** @internal */
export const PutFirewallConfigSecurityType$outboundSchema = PutFirewallConfigSecurityType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityType$;
(function (PutFirewallConfigSecurityType$) {
    /** @deprecated use `PutFirewallConfigSecurityType$inboundSchema` instead. */
    PutFirewallConfigSecurityType$.inboundSchema = PutFirewallConfigSecurityType$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityType$outboundSchema` instead. */
    PutFirewallConfigSecurityType$.outboundSchema = PutFirewallConfigSecurityType$outboundSchema;
})(PutFirewallConfigSecurityType$ || (PutFirewallConfigSecurityType$ = {}));
/** @internal */
export const PutFirewallConfigOp$inboundSchema = z.nativeEnum(PutFirewallConfigOp);
/** @internal */
export const PutFirewallConfigOp$outboundSchema = PutFirewallConfigOp$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigOp$;
(function (PutFirewallConfigOp$) {
    /** @deprecated use `PutFirewallConfigOp$inboundSchema` instead. */
    PutFirewallConfigOp$.inboundSchema = PutFirewallConfigOp$inboundSchema;
    /** @deprecated use `PutFirewallConfigOp$outboundSchema` instead. */
    PutFirewallConfigOp$.outboundSchema = PutFirewallConfigOp$outboundSchema;
})(PutFirewallConfigOp$ || (PutFirewallConfigOp$ = {}));
/** @internal */
export const PutFirewallConfigValue$inboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
/** @internal */
export const PutFirewallConfigValue$outboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigValue$;
(function (PutFirewallConfigValue$) {
    /** @deprecated use `PutFirewallConfigValue$inboundSchema` instead. */
    PutFirewallConfigValue$.inboundSchema = PutFirewallConfigValue$inboundSchema;
    /** @deprecated use `PutFirewallConfigValue$outboundSchema` instead. */
    PutFirewallConfigValue$.outboundSchema = PutFirewallConfigValue$outboundSchema;
})(PutFirewallConfigValue$ || (PutFirewallConfigValue$ = {}));
export function putFirewallConfigValueToJSON(putFirewallConfigValue) {
    return JSON.stringify(PutFirewallConfigValue$outboundSchema.parse(putFirewallConfigValue));
}
export function putFirewallConfigValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigValue' from JSON`);
}
/** @internal */
export const PutFirewallConfigConditions$inboundSchema = z.object({
    type: PutFirewallConfigSecurityType$inboundSchema,
    op: PutFirewallConfigOp$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
/** @internal */
export const PutFirewallConfigConditions$outboundSchema = z.object({
    type: PutFirewallConfigSecurityType$outboundSchema,
    op: PutFirewallConfigOp$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigConditions$;
(function (PutFirewallConfigConditions$) {
    /** @deprecated use `PutFirewallConfigConditions$inboundSchema` instead. */
    PutFirewallConfigConditions$.inboundSchema = PutFirewallConfigConditions$inboundSchema;
    /** @deprecated use `PutFirewallConfigConditions$outboundSchema` instead. */
    PutFirewallConfigConditions$.outboundSchema = PutFirewallConfigConditions$outboundSchema;
})(PutFirewallConfigConditions$ || (PutFirewallConfigConditions$ = {}));
export function putFirewallConfigConditionsToJSON(putFirewallConfigConditions) {
    return JSON.stringify(PutFirewallConfigConditions$outboundSchema.parse(putFirewallConfigConditions));
}
export function putFirewallConfigConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigConditions' from JSON`);
}
/** @internal */
export const PutFirewallConfigConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => PutFirewallConfigConditions$inboundSchema)),
});
/** @internal */
export const PutFirewallConfigConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => PutFirewallConfigConditions$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigConditionGroup$;
(function (PutFirewallConfigConditionGroup$) {
    /** @deprecated use `PutFirewallConfigConditionGroup$inboundSchema` instead. */
    PutFirewallConfigConditionGroup$.inboundSchema = PutFirewallConfigConditionGroup$inboundSchema;
    /** @deprecated use `PutFirewallConfigConditionGroup$outboundSchema` instead. */
    PutFirewallConfigConditionGroup$.outboundSchema = PutFirewallConfigConditionGroup$outboundSchema;
})(PutFirewallConfigConditionGroup$ || (PutFirewallConfigConditionGroup$ = {}));
export function putFirewallConfigConditionGroupToJSON(putFirewallConfigConditionGroup) {
    return JSON.stringify(PutFirewallConfigConditionGroup$outboundSchema.parse(putFirewallConfigConditionGroup));
}
export function putFirewallConfigConditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigConditionGroup' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$ = {}));
/** @internal */
export const PutFirewallConfigAlgo$inboundSchema = z.nativeEnum(PutFirewallConfigAlgo);
/** @internal */
export const PutFirewallConfigAlgo$outboundSchema = PutFirewallConfigAlgo$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigAlgo$;
(function (PutFirewallConfigAlgo$) {
    /** @deprecated use `PutFirewallConfigAlgo$inboundSchema` instead. */
    PutFirewallConfigAlgo$.inboundSchema = PutFirewallConfigAlgo$inboundSchema;
    /** @deprecated use `PutFirewallConfigAlgo$outboundSchema` instead. */
    PutFirewallConfigAlgo$.outboundSchema = PutFirewallConfigAlgo$outboundSchema;
})(PutFirewallConfigAlgo$ || (PutFirewallConfigAlgo$ = {}));
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$ = {}));
/** @internal */
export const PutFirewallConfigRateLimit$inboundSchema = z.object({
    algo: PutFirewallConfigAlgo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$inboundSchema).optional(),
});
/** @internal */
export const PutFirewallConfigRateLimit$outboundSchema = z.object({
    algo: PutFirewallConfigAlgo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesActionAction$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigRateLimit$;
(function (PutFirewallConfigRateLimit$) {
    /** @deprecated use `PutFirewallConfigRateLimit$inboundSchema` instead. */
    PutFirewallConfigRateLimit$.inboundSchema = PutFirewallConfigRateLimit$inboundSchema;
    /** @deprecated use `PutFirewallConfigRateLimit$outboundSchema` instead. */
    PutFirewallConfigRateLimit$.outboundSchema = PutFirewallConfigRateLimit$outboundSchema;
})(PutFirewallConfigRateLimit$ || (PutFirewallConfigRateLimit$ = {}));
export function putFirewallConfigRateLimitToJSON(putFirewallConfigRateLimit) {
    return JSON.stringify(PutFirewallConfigRateLimit$outboundSchema.parse(putFirewallConfigRateLimit));
}
export function putFirewallConfigRateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRateLimit' from JSON`);
}
/** @internal */
export const PutFirewallConfigRedirect$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const PutFirewallConfigRedirect$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigRedirect$;
(function (PutFirewallConfigRedirect$) {
    /** @deprecated use `PutFirewallConfigRedirect$inboundSchema` instead. */
    PutFirewallConfigRedirect$.inboundSchema = PutFirewallConfigRedirect$inboundSchema;
    /** @deprecated use `PutFirewallConfigRedirect$outboundSchema` instead. */
    PutFirewallConfigRedirect$.outboundSchema = PutFirewallConfigRedirect$outboundSchema;
})(PutFirewallConfigRedirect$ || (PutFirewallConfigRedirect$ = {}));
export function putFirewallConfigRedirectToJSON(putFirewallConfigRedirect) {
    return JSON.stringify(PutFirewallConfigRedirect$outboundSchema.parse(putFirewallConfigRedirect));
}
export function putFirewallConfigRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRedirect' from JSON`);
}
/** @internal */
export const PutFirewallConfigMitigate$inboundSchema = z.object({
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$inboundSchema,
    rateLimit: z.nullable(z.lazy(() => PutFirewallConfigRateLimit$inboundSchema))
        .optional(),
    redirect: z.nullable(z.lazy(() => PutFirewallConfigRedirect$inboundSchema))
        .optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const PutFirewallConfigMitigate$outboundSchema = z.object({
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveRulesAction$outboundSchema,
    rateLimit: z.nullable(z.lazy(() => PutFirewallConfigRateLimit$outboundSchema))
        .optional(),
    redirect: z.nullable(z.lazy(() => PutFirewallConfigRedirect$outboundSchema))
        .optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigMitigate$;
(function (PutFirewallConfigMitigate$) {
    /** @deprecated use `PutFirewallConfigMitigate$inboundSchema` instead. */
    PutFirewallConfigMitigate$.inboundSchema = PutFirewallConfigMitigate$inboundSchema;
    /** @deprecated use `PutFirewallConfigMitigate$outboundSchema` instead. */
    PutFirewallConfigMitigate$.outboundSchema = PutFirewallConfigMitigate$outboundSchema;
})(PutFirewallConfigMitigate$ || (PutFirewallConfigMitigate$ = {}));
export function putFirewallConfigMitigateToJSON(putFirewallConfigMitigate) {
    return JSON.stringify(PutFirewallConfigMitigate$outboundSchema.parse(putFirewallConfigMitigate));
}
export function putFirewallConfigMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigMitigate' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponseAction$inboundSchema = z.object({
    mitigate: z.lazy(() => PutFirewallConfigMitigate$inboundSchema).optional(),
});
/** @internal */
export const PutFirewallConfigSecurityResponseAction$outboundSchema = z.object({
    mitigate: z.lazy(() => PutFirewallConfigMitigate$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponseAction$;
(function (PutFirewallConfigSecurityResponseAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponseAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponseAction$.inboundSchema = PutFirewallConfigSecurityResponseAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponseAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponseAction$.outboundSchema = PutFirewallConfigSecurityResponseAction$outboundSchema;
})(PutFirewallConfigSecurityResponseAction$ || (PutFirewallConfigSecurityResponseAction$ = {}));
export function putFirewallConfigSecurityResponseActionToJSON(putFirewallConfigSecurityResponseAction) {
    return JSON.stringify(PutFirewallConfigSecurityResponseAction$outboundSchema.parse(putFirewallConfigSecurityResponseAction));
}
export function putFirewallConfigSecurityResponseActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSecurityResponseAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSecurityResponseAction' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRules$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => PutFirewallConfigConditionGroup$inboundSchema)),
    action: z.lazy(() => PutFirewallConfigSecurityResponseAction$inboundSchema),
});
/** @internal */
export const PutFirewallConfigSecurityRules$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => PutFirewallConfigConditionGroup$outboundSchema)),
    action: z.lazy(() => PutFirewallConfigSecurityResponseAction$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityRules$;
(function (PutFirewallConfigSecurityRules$) {
    /** @deprecated use `PutFirewallConfigSecurityRules$inboundSchema` instead. */
    PutFirewallConfigSecurityRules$.inboundSchema = PutFirewallConfigSecurityRules$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityRules$outboundSchema` instead. */
    PutFirewallConfigSecurityRules$.outboundSchema = PutFirewallConfigSecurityRules$outboundSchema;
})(PutFirewallConfigSecurityRules$ || (PutFirewallConfigSecurityRules$ = {}));
export function putFirewallConfigSecurityRulesToJSON(putFirewallConfigSecurityRules) {
    return JSON.stringify(PutFirewallConfigSecurityRules$outboundSchema.parse(putFirewallConfigSecurityRules));
}
export function putFirewallConfigSecurityRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSecurityRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSecurityRules' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200Action$inboundSchema = z
    .nativeEnum(PutFirewallConfigSecurityResponse200Action);
/** @internal */
export const PutFirewallConfigSecurityResponse200Action$outboundSchema = PutFirewallConfigSecurityResponse200Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200Action$;
(function (PutFirewallConfigSecurityResponse200Action$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200Action$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200Action$.inboundSchema = PutFirewallConfigSecurityResponse200Action$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200Action$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200Action$.outboundSchema = PutFirewallConfigSecurityResponse200Action$outboundSchema;
})(PutFirewallConfigSecurityResponse200Action$ || (PutFirewallConfigSecurityResponse200Action$ = {}));
/** @internal */
export const PutFirewallConfigIps$inboundSchema = z.object({
    id: z.string(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityResponse200Action$inboundSchema,
});
/** @internal */
export const PutFirewallConfigIps$outboundSchema = z.object({
    id: z.string(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityResponse200Action$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigIps$;
(function (PutFirewallConfigIps$) {
    /** @deprecated use `PutFirewallConfigIps$inboundSchema` instead. */
    PutFirewallConfigIps$.inboundSchema = PutFirewallConfigIps$inboundSchema;
    /** @deprecated use `PutFirewallConfigIps$outboundSchema` instead. */
    PutFirewallConfigIps$.outboundSchema = PutFirewallConfigIps$outboundSchema;
})(PutFirewallConfigIps$ || (PutFirewallConfigIps$ = {}));
export function putFirewallConfigIpsToJSON(putFirewallConfigIps) {
    return JSON.stringify(PutFirewallConfigIps$outboundSchema.parse(putFirewallConfigIps));
}
export function putFirewallConfigIpsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigIps$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigIps' from JSON`);
}
/** @internal */
export const PutFirewallConfigChanges$inboundSchema = z.object({});
/** @internal */
export const PutFirewallConfigChanges$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigChanges$;
(function (PutFirewallConfigChanges$) {
    /** @deprecated use `PutFirewallConfigChanges$inboundSchema` instead. */
    PutFirewallConfigChanges$.inboundSchema = PutFirewallConfigChanges$inboundSchema;
    /** @deprecated use `PutFirewallConfigChanges$outboundSchema` instead. */
    PutFirewallConfigChanges$.outboundSchema = PutFirewallConfigChanges$outboundSchema;
})(PutFirewallConfigChanges$ || (PutFirewallConfigChanges$ = {}));
export function putFirewallConfigChangesToJSON(putFirewallConfigChanges) {
    return JSON.stringify(PutFirewallConfigChanges$outboundSchema.parse(putFirewallConfigChanges));
}
export function putFirewallConfigChangesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigChanges$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigChanges' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$ = {}));
/** @internal */
export const PutFirewallConfigBotProtection$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const PutFirewallConfigBotProtection$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigBotProtection$;
(function (PutFirewallConfigBotProtection$) {
    /** @deprecated use `PutFirewallConfigBotProtection$inboundSchema` instead. */
    PutFirewallConfigBotProtection$.inboundSchema = PutFirewallConfigBotProtection$inboundSchema;
    /** @deprecated use `PutFirewallConfigBotProtection$outboundSchema` instead. */
    PutFirewallConfigBotProtection$.outboundSchema = PutFirewallConfigBotProtection$outboundSchema;
})(PutFirewallConfigBotProtection$ || (PutFirewallConfigBotProtection$ = {}));
export function putFirewallConfigBotProtectionToJSON(putFirewallConfigBotProtection) {
    return JSON.stringify(PutFirewallConfigBotProtection$outboundSchema.parse(putFirewallConfigBotProtection));
}
export function putFirewallConfigBotProtectionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigBotProtection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigBotProtection' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$ = {}));
/** @internal */
export const PutFirewallConfigAiBots$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const PutFirewallConfigAiBots$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigAiBots$;
(function (PutFirewallConfigAiBots$) {
    /** @deprecated use `PutFirewallConfigAiBots$inboundSchema` instead. */
    PutFirewallConfigAiBots$.inboundSchema = PutFirewallConfigAiBots$inboundSchema;
    /** @deprecated use `PutFirewallConfigAiBots$outboundSchema` instead. */
    PutFirewallConfigAiBots$.outboundSchema = PutFirewallConfigAiBots$outboundSchema;
})(PutFirewallConfigAiBots$ || (PutFirewallConfigAiBots$ = {}));
export function putFirewallConfigAiBotsToJSON(putFirewallConfigAiBots) {
    return JSON.stringify(PutFirewallConfigAiBots$outboundSchema.parse(putFirewallConfigAiBots));
}
export function putFirewallConfigAiBotsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigAiBots$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigAiBots' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$;
(function (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$) {
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$.inboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema;
    /** @deprecated use `PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$outboundSchema` instead. */
    PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$.outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$outboundSchema;
})(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$ || (PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$ = {}));
/** @internal */
export const PutFirewallConfigOwasp$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const PutFirewallConfigOwasp$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigOwasp$;
(function (PutFirewallConfigOwasp$) {
    /** @deprecated use `PutFirewallConfigOwasp$inboundSchema` instead. */
    PutFirewallConfigOwasp$.inboundSchema = PutFirewallConfigOwasp$inboundSchema;
    /** @deprecated use `PutFirewallConfigOwasp$outboundSchema` instead. */
    PutFirewallConfigOwasp$.outboundSchema = PutFirewallConfigOwasp$outboundSchema;
})(PutFirewallConfigOwasp$ || (PutFirewallConfigOwasp$ = {}));
export function putFirewallConfigOwaspToJSON(putFirewallConfigOwasp) {
    return JSON.stringify(PutFirewallConfigOwasp$outboundSchema.parse(putFirewallConfigOwasp));
}
export function putFirewallConfigOwaspFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigOwasp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigOwasp' from JSON`);
}
/** @internal */
export const PutFirewallConfigManagedRules$inboundSchema = z.object({
    bot_protection: z.lazy(() => PutFirewallConfigBotProtection$inboundSchema)
        .optional(),
    ai_bots: z.lazy(() => PutFirewallConfigAiBots$inboundSchema).optional(),
    owasp: z.lazy(() => PutFirewallConfigOwasp$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "bot_protection": "botProtection",
        "ai_bots": "aiBots",
    });
});
/** @internal */
export const PutFirewallConfigManagedRules$outboundSchema = z.object({
    botProtection: z.lazy(() => PutFirewallConfigBotProtection$outboundSchema)
        .optional(),
    aiBots: z.lazy(() => PutFirewallConfigAiBots$outboundSchema).optional(),
    owasp: z.lazy(() => PutFirewallConfigOwasp$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        botProtection: "bot_protection",
        aiBots: "ai_bots",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigManagedRules$;
(function (PutFirewallConfigManagedRules$) {
    /** @deprecated use `PutFirewallConfigManagedRules$inboundSchema` instead. */
    PutFirewallConfigManagedRules$.inboundSchema = PutFirewallConfigManagedRules$inboundSchema;
    /** @deprecated use `PutFirewallConfigManagedRules$outboundSchema` instead. */
    PutFirewallConfigManagedRules$.outboundSchema = PutFirewallConfigManagedRules$outboundSchema;
})(PutFirewallConfigManagedRules$ || (PutFirewallConfigManagedRules$ = {}));
export function putFirewallConfigManagedRulesToJSON(putFirewallConfigManagedRules) {
    return JSON.stringify(PutFirewallConfigManagedRules$outboundSchema.parse(putFirewallConfigManagedRules));
}
export function putFirewallConfigManagedRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigManagedRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigManagedRules' from JSON`);
}
/** @internal */
export const Active$inboundSchema = z
    .object({
    ownerId: z.string(),
    projectKey: z.string(),
    id: z.string(),
    version: z.number(),
    updatedAt: z.string(),
    firewallEnabled: z.boolean(),
    crs: z.lazy(() => PutFirewallConfigCrs$inboundSchema),
    rules: z.array(z.lazy(() => PutFirewallConfigSecurityRules$inboundSchema)),
    ips: z.array(z.lazy(() => PutFirewallConfigIps$inboundSchema)),
    changes: z.array(z.lazy(() => PutFirewallConfigChanges$inboundSchema)),
    managedRules: z.lazy(() => PutFirewallConfigManagedRules$inboundSchema)
        .optional(),
    botIdEnabled: z.boolean().optional(),
});
/** @internal */
export const Active$outboundSchema = z.object({
    ownerId: z.string(),
    projectKey: z.string(),
    id: z.string(),
    version: z.number(),
    updatedAt: z.string(),
    firewallEnabled: z.boolean(),
    crs: z.lazy(() => PutFirewallConfigCrs$outboundSchema),
    rules: z.array(z.lazy(() => PutFirewallConfigSecurityRules$outboundSchema)),
    ips: z.array(z.lazy(() => PutFirewallConfigIps$outboundSchema)),
    changes: z.array(z.lazy(() => PutFirewallConfigChanges$outboundSchema)),
    managedRules: z.lazy(() => PutFirewallConfigManagedRules$outboundSchema)
        .optional(),
    botIdEnabled: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Active$;
(function (Active$) {
    /** @deprecated use `Active$inboundSchema` instead. */
    Active$.inboundSchema = Active$inboundSchema;
    /** @deprecated use `Active$outboundSchema` instead. */
    Active$.outboundSchema = Active$outboundSchema;
})(Active$ || (Active$ = {}));
export function activeToJSON(active) {
    return JSON.stringify(Active$outboundSchema.parse(active));
}
export function activeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Active$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Active' from JSON`);
}
/** @internal */
export const PutFirewallConfigResponseBody$inboundSchema = z.object({
    active: z.lazy(() => Active$inboundSchema),
});
/** @internal */
export const PutFirewallConfigResponseBody$outboundSchema = z.object({
    active: z.lazy(() => Active$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PutFirewallConfigResponseBody$;
(function (PutFirewallConfigResponseBody$) {
    /** @deprecated use `PutFirewallConfigResponseBody$inboundSchema` instead. */
    PutFirewallConfigResponseBody$.inboundSchema = PutFirewallConfigResponseBody$inboundSchema;
    /** @deprecated use `PutFirewallConfigResponseBody$outboundSchema` instead. */
    PutFirewallConfigResponseBody$.outboundSchema = PutFirewallConfigResponseBody$outboundSchema;
})(PutFirewallConfigResponseBody$ || (PutFirewallConfigResponseBody$ = {}));
export function putFirewallConfigResponseBodyToJSON(putFirewallConfigResponseBody) {
    return JSON.stringify(PutFirewallConfigResponseBody$outboundSchema.parse(putFirewallConfigResponseBody));
}
export function putFirewallConfigResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigResponseBody' from JSON`);
}
