// @vercel/sdk/esm/models/requestpromoteop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const RequestPromoteRequest$inboundSchema = z.object({
    projectId: z.string(),
    deploymentId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const RequestPromoteRequest$outboundSchema = z.object({
    projectId: z.string(),
    deploymentId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestPromoteRequest$;
(function (RequestPromoteRequest$) {
    /** @deprecated use `RequestPromoteRequest$inboundSchema` instead. */
    RequestPromoteRequest$.inboundSchema = RequestPromoteRequest$inboundSchema;
    /** @deprecated use `RequestPromoteRequest$outboundSchema` instead. */
    RequestPromoteRequest$.outboundSchema = RequestPromoteRequest$outboundSchema;
})(RequestPromoteRequest$ || (RequestPromoteRequest$ = {}));
export function requestPromoteRequestToJSON(requestPromoteRequest) {
    return JSON.stringify(RequestPromoteRequest$outboundSchema.parse(requestPromoteRequest));
}
export function requestPromoteRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestPromoteRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestPromoteRequest' from JSON`);
}
