// @vercel/sdk/esm/models/updatefirewallconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateFirewallConfigRequestBodyId = {
    AiBots: "ai_bots",
    BotFilter: "bot_filter",
    BotProtection: "bot_protection",
    Owasp: "owasp",
};
export const UpdateFirewallConfigRequestBodySecurityRequest12Action = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
};
export const UpdateFirewallConfigRequestBodySecurityRequest11Action = {
    ManagedRulesUpdate: "managedRules.update",
};
export const RequestBodyId = {
    AiBots: "ai_bots",
    BotFilter: "bot_filter",
    BotProtection: "bot_protection",
    Owasp: "owasp",
};
export const UpdateFirewallConfigRequestBodySecurityRequest11ValueAction = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
};
export const UpdateFirewallConfigRequestBodySecurityRequest10Action = {
    IpRemove: "ip.remove",
};
export const UpdateFirewallConfigRequestBodySecurityRequest9Action = {
    IpUpdate: "ip.update",
};
export const UpdateFirewallConfigRequestBodySecurityRequest9ValueAction = {
    Deny: "deny",
    Challenge: "challenge",
    Log: "log",
    Bypass: "bypass",
};
export const UpdateFirewallConfigRequestBodySecurityRequest8Action = {
    IpInsert: "ip.insert",
};
export const UpdateFirewallConfigRequestBodySecurityRequest8ValueAction = {
    Deny: "deny",
    Challenge: "challenge",
    Log: "log",
    Bypass: "bypass",
};
export const UpdateFirewallConfigRequestBodySecurityRequest7Action = {
    CrsDisable: "crs.disable",
};
export const UpdateFirewallConfigRequestBodySecurityRequest6Action = {
    CrsUpdate: "crs.update",
};
export const Id = {
    Sd: "sd",
    Ma: "ma",
    Lfi: "lfi",
    Rfi: "rfi",
    Rce: "rce",
    Php: "php",
    Gen: "gen",
    Xss: "xss",
    Sqli: "sqli",
    Sf: "sf",
    Java: "java",
};
export const UpdateFirewallConfigRequestBodySecurityRequest6ValueAction = {
    Deny: "deny",
    Log: "log",
};
export const UpdateFirewallConfigRequestBodySecurityRequest5Action = {
    RulesPriority: "rules.priority",
};
export const UpdateFirewallConfigRequestBodySecurityRequest4Action = {
    RulesRemove: "rules.remove",
};
export const UpdateFirewallConfigRequestBodySecurityRequestAction = {
    RulesUpdate: "rules.update",
};
export const UpdateFirewallConfigRequestBodySecurityType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Region: "region",
    Protocol: "protocol",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    ServerAction: "server_action",
};
export const UpdateFirewallConfigRequestBodyOp = {
    Re: "re",
    Eq: "eq",
    Neq: "neq",
    Ex: "ex",
    Nex: "nex",
    Inc: "inc",
    Ninc: "ninc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
};
export const UpdateFirewallConfigRequestBodySecurityRequest3ValueAction = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const UpdateFirewallConfigRateLimitAlgo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const UpdateFirewallConfigActionSecurity1 = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    RateLimit: "rate_limit",
};
export const UpdateFirewallConfigRequestBodySecurityAction = {
    RulesInsert: "rules.insert",
};
export const UpdateFirewallConfigRequestBodyType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Region: "region",
    Protocol: "protocol",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    ServerAction: "server_action",
};
export const RequestBodyOp = {
    Re: "re",
    Eq: "eq",
    Neq: "neq",
    Ex: "ex",
    Nex: "nex",
    Inc: "inc",
    Ninc: "ninc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
};
export const UpdateFirewallConfigRequestBodySecurityRequest2ValueAction = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const RateLimitAlgo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const UpdateFirewallConfigAction1 = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    RateLimit: "rate_limit",
};
export const UpdateFirewallConfigRequestBodyAction = {
    FirewallEnabled: "firewallEnabled",
};
/** @internal */
export const RequestBody13$inboundSchema = z.object({
    action: z.string(),
    id: z.string().optional(),
    value: z.boolean(),
});
/** @internal */
export const RequestBody13$outboundSchema = z.object({
    action: z.string(),
    id: z.string().optional(),
    value: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBody13$;
(function (RequestBody13$) {
    /** @deprecated use `RequestBody13$inboundSchema` instead. */
    RequestBody13$.inboundSchema = RequestBody13$inboundSchema;
    /** @deprecated use `RequestBody13$outboundSchema` instead. */
    RequestBody13$.outboundSchema = RequestBody13$outboundSchema;
})(RequestBody13$ || (RequestBody13$ = {}));
export function requestBody13ToJSON(requestBody13) {
    return JSON.stringify(RequestBody13$outboundSchema.parse(requestBody13));
}
export function requestBody13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBody13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBody13' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyId$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodyId);
/** @internal */
export const UpdateFirewallConfigRequestBodyId$outboundSchema = UpdateFirewallConfigRequestBodyId$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyId$;
(function (UpdateFirewallConfigRequestBodyId$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyId$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyId$.inboundSchema = UpdateFirewallConfigRequestBodyId$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyId$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyId$.outboundSchema = UpdateFirewallConfigRequestBodyId$outboundSchema;
})(UpdateFirewallConfigRequestBodyId$ || (UpdateFirewallConfigRequestBodyId$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest12Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest12Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest12Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest12Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest12Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest12Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest12Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest12Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest12Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest12Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest12Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest12Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest12Action$ || (UpdateFirewallConfigRequestBodySecurityRequest12Action$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest12Value$inboundSchema = z.object({
    active: z.boolean(),
    action: UpdateFirewallConfigRequestBodySecurityRequest12Action$inboundSchema
        .optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest12Value$outboundSchema = z.object({
    active: z.boolean(),
    action: UpdateFirewallConfigRequestBodySecurityRequest12Action$outboundSchema
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest12Value$;
(function (UpdateFirewallConfigRequestBodySecurityRequest12Value$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest12Value$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest12Value$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest12Value$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest12Value$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest12Value$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest12Value$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest12Value$ || (UpdateFirewallConfigRequestBodySecurityRequest12Value$ = {}));
export function updateFirewallConfigRequestBodySecurityRequest12ValueToJSON(updateFirewallConfigRequestBodySecurityRequest12Value) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequest12Value$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequest12Value));
}
export function updateFirewallConfigRequestBodySecurityRequest12ValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequest12Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequest12Value' from JSON`);
}
/** @internal */
export const RequestBody12$inboundSchema = z.object({
    action: z.string(),
    id: UpdateFirewallConfigRequestBodyId$inboundSchema,
    value: z.record(z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest12Value$inboundSchema)),
});
/** @internal */
export const RequestBody12$outboundSchema = z.object({
    action: z.string(),
    id: UpdateFirewallConfigRequestBodyId$outboundSchema,
    value: z.record(z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest12Value$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBody12$;
(function (RequestBody12$) {
    /** @deprecated use `RequestBody12$inboundSchema` instead. */
    RequestBody12$.inboundSchema = RequestBody12$inboundSchema;
    /** @deprecated use `RequestBody12$outboundSchema` instead. */
    RequestBody12$.outboundSchema = RequestBody12$outboundSchema;
})(RequestBody12$ || (RequestBody12$ = {}));
export function requestBody12ToJSON(requestBody12) {
    return JSON.stringify(RequestBody12$outboundSchema.parse(requestBody12));
}
export function requestBody12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBody12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBody12' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest11Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest11Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest11Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest11Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest11Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest11Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest11Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest11Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest11Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest11Action$ || (UpdateFirewallConfigRequestBodySecurityRequest11Action$ = {}));
/** @internal */
export const RequestBodyId$inboundSchema = z.nativeEnum(RequestBodyId);
/** @internal */
export const RequestBodyId$outboundSchema = RequestBodyId$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyId$;
(function (RequestBodyId$) {
    /** @deprecated use `RequestBodyId$inboundSchema` instead. */
    RequestBodyId$.inboundSchema = RequestBodyId$inboundSchema;
    /** @deprecated use `RequestBodyId$outboundSchema` instead. */
    RequestBodyId$.outboundSchema = RequestBodyId$outboundSchema;
})(RequestBodyId$ || (RequestBodyId$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest11ValueAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$;
(function (UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$ || (UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest11Value$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$inboundSchema
        .optional(),
    active: z.boolean(),
});
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest11Value$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest11ValueAction$outboundSchema
        .optional(),
    active: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest11Value$;
(function (UpdateFirewallConfigRequestBodySecurityRequest11Value$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest11Value$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest11Value$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11Value$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest11Value$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest11Value$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest11Value$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest11Value$ || (UpdateFirewallConfigRequestBodySecurityRequest11Value$ = {}));
export function updateFirewallConfigRequestBodySecurityRequest11ValueToJSON(updateFirewallConfigRequestBodySecurityRequest11Value) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequest11Value$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequest11Value));
}
export function updateFirewallConfigRequestBodySecurityRequest11ValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequest11Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequest11Value' from JSON`);
}
/** @internal */
export const RequestBody11$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest11Action$inboundSchema,
    id: RequestBodyId$inboundSchema,
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest11Value$inboundSchema),
});
/** @internal */
export const RequestBody11$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest11Action$outboundSchema,
    id: RequestBodyId$outboundSchema,
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest11Value$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBody11$;
(function (RequestBody11$) {
    /** @deprecated use `RequestBody11$inboundSchema` instead. */
    RequestBody11$.inboundSchema = RequestBody11$inboundSchema;
    /** @deprecated use `RequestBody11$outboundSchema` instead. */
    RequestBody11$.outboundSchema = RequestBody11$outboundSchema;
})(RequestBody11$ || (RequestBody11$ = {}));
export function requestBody11ToJSON(requestBody11) {
    return JSON.stringify(RequestBody11$outboundSchema.parse(requestBody11));
}
export function requestBody11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBody11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBody11' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest10Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest10Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest10Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest10Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest10Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest10Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest10Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest10Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest10Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest10Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest10Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest10Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest10Action$ || (UpdateFirewallConfigRequestBodySecurityRequest10Action$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBody10$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest10Action$inboundSchema,
    id: z.string(),
    value: z.nullable(z.any()).optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBody10$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest10Action$outboundSchema,
    id: z.string(),
    value: z.nullable(z.any()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody10$;
(function (UpdateFirewallConfigRequestBody10$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody10$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody10$.inboundSchema = UpdateFirewallConfigRequestBody10$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody10$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody10$.outboundSchema = UpdateFirewallConfigRequestBody10$outboundSchema;
})(UpdateFirewallConfigRequestBody10$ || (UpdateFirewallConfigRequestBody10$ = {}));
export function updateFirewallConfigRequestBody10ToJSON(updateFirewallConfigRequestBody10) {
    return JSON.stringify(UpdateFirewallConfigRequestBody10$outboundSchema.parse(updateFirewallConfigRequestBody10));
}
export function updateFirewallConfigRequestBody10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody10' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest9Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest9Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest9Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest9Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest9Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest9Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest9Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest9Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest9Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest9Action$ || (UpdateFirewallConfigRequestBodySecurityRequest9Action$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest9ValueAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$;
(function (UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$ || (UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest9Value$inboundSchema = z.object({
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$inboundSchema,
});
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest9Value$outboundSchema = z.object({
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: UpdateFirewallConfigRequestBodySecurityRequest9ValueAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest9Value$;
(function (UpdateFirewallConfigRequestBodySecurityRequest9Value$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest9Value$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest9Value$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9Value$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest9Value$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest9Value$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest9Value$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest9Value$ || (UpdateFirewallConfigRequestBodySecurityRequest9Value$ = {}));
export function updateFirewallConfigRequestBodySecurityRequest9ValueToJSON(updateFirewallConfigRequestBodySecurityRequest9Value) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequest9Value$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequest9Value));
}
export function updateFirewallConfigRequestBodySecurityRequest9ValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequest9Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequest9Value' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBody9$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest9Action$inboundSchema,
    id: z.string(),
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest9Value$inboundSchema),
});
/** @internal */
export const UpdateFirewallConfigRequestBody9$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest9Action$outboundSchema,
    id: z.string(),
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest9Value$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody9$;
(function (UpdateFirewallConfigRequestBody9$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody9$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody9$.inboundSchema = UpdateFirewallConfigRequestBody9$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody9$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody9$.outboundSchema = UpdateFirewallConfigRequestBody9$outboundSchema;
})(UpdateFirewallConfigRequestBody9$ || (UpdateFirewallConfigRequestBody9$ = {}));
export function updateFirewallConfigRequestBody9ToJSON(updateFirewallConfigRequestBody9) {
    return JSON.stringify(UpdateFirewallConfigRequestBody9$outboundSchema.parse(updateFirewallConfigRequestBody9));
}
export function updateFirewallConfigRequestBody9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody9' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest8Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest8Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest8Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest8Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest8Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest8Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest8Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest8Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest8Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest8Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest8Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest8Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest8Action$ || (UpdateFirewallConfigRequestBodySecurityRequest8Action$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest8ValueAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$;
(function (UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$ || (UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequestValue$inboundSchema = z.object({
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$inboundSchema,
});
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequestValue$outboundSchema = z.object({
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: UpdateFirewallConfigRequestBodySecurityRequest8ValueAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequestValue$;
(function (UpdateFirewallConfigRequestBodySecurityRequestValue$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequestValue$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequestValue$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequestValue$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequestValue$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequestValue$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequestValue$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequestValue$ || (UpdateFirewallConfigRequestBodySecurityRequestValue$ = {}));
export function updateFirewallConfigRequestBodySecurityRequestValueToJSON(updateFirewallConfigRequestBodySecurityRequestValue) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequestValue$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequestValue));
}
export function updateFirewallConfigRequestBodySecurityRequestValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequestValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequestValue' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBody8$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest8Action$inboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequestValue$inboundSchema),
});
/** @internal */
export const UpdateFirewallConfigRequestBody8$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest8Action$outboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequestValue$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody8$;
(function (UpdateFirewallConfigRequestBody8$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody8$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody8$.inboundSchema = UpdateFirewallConfigRequestBody8$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody8$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody8$.outboundSchema = UpdateFirewallConfigRequestBody8$outboundSchema;
})(UpdateFirewallConfigRequestBody8$ || (UpdateFirewallConfigRequestBody8$ = {}));
export function updateFirewallConfigRequestBody8ToJSON(updateFirewallConfigRequestBody8) {
    return JSON.stringify(UpdateFirewallConfigRequestBody8$outboundSchema.parse(updateFirewallConfigRequestBody8));
}
export function updateFirewallConfigRequestBody8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody8' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest7Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest7Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest7Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest7Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest7Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest7Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest7Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest7Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest7Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest7Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest7Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest7Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest7Action$ || (UpdateFirewallConfigRequestBodySecurityRequest7Action$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBody7$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest7Action$inboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.nullable(z.any()).optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBody7$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest7Action$outboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.nullable(z.any()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody7$;
(function (UpdateFirewallConfigRequestBody7$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody7$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody7$.inboundSchema = UpdateFirewallConfigRequestBody7$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody7$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody7$.outboundSchema = UpdateFirewallConfigRequestBody7$outboundSchema;
})(UpdateFirewallConfigRequestBody7$ || (UpdateFirewallConfigRequestBody7$ = {}));
export function updateFirewallConfigRequestBody7ToJSON(updateFirewallConfigRequestBody7) {
    return JSON.stringify(UpdateFirewallConfigRequestBody7$outboundSchema.parse(updateFirewallConfigRequestBody7));
}
export function updateFirewallConfigRequestBody7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody7' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest6Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest6Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest6Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest6Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest6Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest6Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest6Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest6Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest6Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest6Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest6Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest6Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest6Action$ || (UpdateFirewallConfigRequestBodySecurityRequest6Action$ = {}));
/** @internal */
export const Id$inboundSchema = z.nativeEnum(Id);
/** @internal */
export const Id$outboundSchema = Id$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Id$;
(function (Id$) {
    /** @deprecated use `Id$inboundSchema` instead. */
    Id$.inboundSchema = Id$inboundSchema;
    /** @deprecated use `Id$outboundSchema` instead. */
    Id$.outboundSchema = Id$outboundSchema;
})(Id$ || (Id$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest6ValueAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$;
(function (UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$ || (UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityValue$inboundSchema = z.object({
    active: z.boolean(),
    action: UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$inboundSchema,
});
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityValue$outboundSchema = z.object({
    active: z.boolean(),
    action: UpdateFirewallConfigRequestBodySecurityRequest6ValueAction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityValue$;
(function (UpdateFirewallConfigRequestBodySecurityValue$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityValue$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityValue$.inboundSchema = UpdateFirewallConfigRequestBodySecurityValue$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityValue$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityValue$.outboundSchema = UpdateFirewallConfigRequestBodySecurityValue$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityValue$ || (UpdateFirewallConfigRequestBodySecurityValue$ = {}));
export function updateFirewallConfigRequestBodySecurityValueToJSON(updateFirewallConfigRequestBodySecurityValue) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityValue$outboundSchema.parse(updateFirewallConfigRequestBodySecurityValue));
}
export function updateFirewallConfigRequestBodySecurityValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityValue' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBody6$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest6Action$inboundSchema,
    id: Id$inboundSchema,
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityValue$inboundSchema),
});
/** @internal */
export const UpdateFirewallConfigRequestBody6$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest6Action$outboundSchema,
    id: Id$outboundSchema,
    value: z.lazy(() => UpdateFirewallConfigRequestBodySecurityValue$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody6$;
(function (UpdateFirewallConfigRequestBody6$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody6$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody6$.inboundSchema = UpdateFirewallConfigRequestBody6$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody6$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody6$.outboundSchema = UpdateFirewallConfigRequestBody6$outboundSchema;
})(UpdateFirewallConfigRequestBody6$ || (UpdateFirewallConfigRequestBody6$ = {}));
export function updateFirewallConfigRequestBody6ToJSON(updateFirewallConfigRequestBody6) {
    return JSON.stringify(UpdateFirewallConfigRequestBody6$outboundSchema.parse(updateFirewallConfigRequestBody6));
}
export function updateFirewallConfigRequestBody6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody6' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest5Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest5Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest5Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest5Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest5Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest5Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest5Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest5Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest5Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest5Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest5Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest5Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest5Action$ || (UpdateFirewallConfigRequestBodySecurityRequest5Action$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBody5$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest5Action$inboundSchema,
    id: z.string(),
    value: z.number(),
});
/** @internal */
export const UpdateFirewallConfigRequestBody5$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest5Action$outboundSchema,
    id: z.string(),
    value: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody5$;
(function (UpdateFirewallConfigRequestBody5$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody5$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody5$.inboundSchema = UpdateFirewallConfigRequestBody5$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody5$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody5$.outboundSchema = UpdateFirewallConfigRequestBody5$outboundSchema;
})(UpdateFirewallConfigRequestBody5$ || (UpdateFirewallConfigRequestBody5$ = {}));
export function updateFirewallConfigRequestBody5ToJSON(updateFirewallConfigRequestBody5) {
    return JSON.stringify(UpdateFirewallConfigRequestBody5$outboundSchema.parse(updateFirewallConfigRequestBody5));
}
export function updateFirewallConfigRequestBody5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody5' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest4Action$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest4Action);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest4Action$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest4Action$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest4Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest4Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest4Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest4Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest4Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest4Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest4Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest4Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest4Action$ || (UpdateFirewallConfigRequestBodySecurityRequest4Action$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBody4$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest4Action$inboundSchema,
    id: z.string(),
    value: z.nullable(z.any()).optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBody4$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest4Action$outboundSchema,
    id: z.string(),
    value: z.nullable(z.any()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody4$;
(function (UpdateFirewallConfigRequestBody4$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody4$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody4$.inboundSchema = UpdateFirewallConfigRequestBody4$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody4$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody4$.outboundSchema = UpdateFirewallConfigRequestBody4$outboundSchema;
})(UpdateFirewallConfigRequestBody4$ || (UpdateFirewallConfigRequestBody4$ = {}));
export function updateFirewallConfigRequestBody4ToJSON(updateFirewallConfigRequestBody4) {
    return JSON.stringify(UpdateFirewallConfigRequestBody4$outboundSchema.parse(updateFirewallConfigRequestBody4));
}
export function updateFirewallConfigRequestBody4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody4' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequestAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequestAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequestAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequestAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequestAction$;
(function (UpdateFirewallConfigRequestBodySecurityRequestAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequestAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequestAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequestAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequestAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequestAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequestAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequestAction$ || (UpdateFirewallConfigRequestBodySecurityRequestAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityType$inboundSchema = z
    .nativeEnum(UpdateFirewallConfigRequestBodySecurityType);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityType$outboundSchema = UpdateFirewallConfigRequestBodySecurityType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityType$;
(function (UpdateFirewallConfigRequestBodySecurityType$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityType$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityType$.inboundSchema = UpdateFirewallConfigRequestBodySecurityType$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityType$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityType$.outboundSchema = UpdateFirewallConfigRequestBodySecurityType$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityType$ || (UpdateFirewallConfigRequestBodySecurityType$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodyOp$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodyOp);
/** @internal */
export const UpdateFirewallConfigRequestBodyOp$outboundSchema = UpdateFirewallConfigRequestBodyOp$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyOp$;
(function (UpdateFirewallConfigRequestBodyOp$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyOp$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyOp$.inboundSchema = UpdateFirewallConfigRequestBodyOp$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyOp$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyOp$.outboundSchema = UpdateFirewallConfigRequestBodyOp$outboundSchema;
})(UpdateFirewallConfigRequestBodyOp$ || (UpdateFirewallConfigRequestBodyOp$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest3Value$inboundSchema = z.union([z.string(), z.array(z.string()), z.number()]);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest3Value$outboundSchema = z.union([z.string(), z.array(z.string()), z.number()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest3Value$;
(function (UpdateFirewallConfigRequestBodySecurityRequest3Value$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest3Value$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest3Value$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest3Value$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest3Value$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest3Value$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest3Value$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest3Value$ || (UpdateFirewallConfigRequestBodySecurityRequest3Value$ = {}));
export function updateFirewallConfigRequestBodySecurityRequest3ValueToJSON(updateFirewallConfigRequestBodySecurityRequest3Value) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequest3Value$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequest3Value));
}
export function updateFirewallConfigRequestBodySecurityRequest3ValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequest3Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequest3Value' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyConditions$inboundSchema = z.object({
    type: UpdateFirewallConfigRequestBodySecurityType$inboundSchema,
    op: UpdateFirewallConfigRequestBodyOp$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBodyConditions$outboundSchema = z.object({
    type: UpdateFirewallConfigRequestBodySecurityType$outboundSchema,
    op: UpdateFirewallConfigRequestBodyOp$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyConditions$;
(function (UpdateFirewallConfigRequestBodyConditions$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyConditions$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyConditions$.inboundSchema = UpdateFirewallConfigRequestBodyConditions$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyConditions$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyConditions$.outboundSchema = UpdateFirewallConfigRequestBodyConditions$outboundSchema;
})(UpdateFirewallConfigRequestBodyConditions$ || (UpdateFirewallConfigRequestBodyConditions$ = {}));
export function updateFirewallConfigRequestBodyConditionsToJSON(updateFirewallConfigRequestBodyConditions) {
    return JSON.stringify(UpdateFirewallConfigRequestBodyConditions$outboundSchema.parse(updateFirewallConfigRequestBodyConditions));
}
export function updateFirewallConfigRequestBodyConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodyConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodyConditions' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => UpdateFirewallConfigRequestBodyConditions$inboundSchema)),
});
/** @internal */
export const UpdateFirewallConfigRequestBodyConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => UpdateFirewallConfigRequestBodyConditions$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyConditionGroup$;
(function (UpdateFirewallConfigRequestBodyConditionGroup$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyConditionGroup$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyConditionGroup$.inboundSchema = UpdateFirewallConfigRequestBodyConditionGroup$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyConditionGroup$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyConditionGroup$.outboundSchema = UpdateFirewallConfigRequestBodyConditionGroup$outboundSchema;
})(UpdateFirewallConfigRequestBodyConditionGroup$ || (UpdateFirewallConfigRequestBodyConditionGroup$ = {}));
export function updateFirewallConfigRequestBodyConditionGroupToJSON(updateFirewallConfigRequestBodyConditionGroup) {
    return JSON.stringify(UpdateFirewallConfigRequestBodyConditionGroup$outboundSchema.parse(updateFirewallConfigRequestBodyConditionGroup));
}
export function updateFirewallConfigRequestBodyConditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodyConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodyConditionGroup' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest3ValueAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$;
(function (UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$ || (UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRateLimitAlgo$inboundSchema = z.nativeEnum(UpdateFirewallConfigRateLimitAlgo);
/** @internal */
export const UpdateFirewallConfigRateLimitAlgo$outboundSchema = UpdateFirewallConfigRateLimitAlgo$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRateLimitAlgo$;
(function (UpdateFirewallConfigRateLimitAlgo$) {
    /** @deprecated use `UpdateFirewallConfigRateLimitAlgo$inboundSchema` instead. */
    UpdateFirewallConfigRateLimitAlgo$.inboundSchema = UpdateFirewallConfigRateLimitAlgo$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRateLimitAlgo$outboundSchema` instead. */
    UpdateFirewallConfigRateLimitAlgo$.outboundSchema = UpdateFirewallConfigRateLimitAlgo$outboundSchema;
})(UpdateFirewallConfigRateLimitAlgo$ || (UpdateFirewallConfigRateLimitAlgo$ = {}));
/** @internal */
export const UpdateFirewallConfigActionSecurity1$inboundSchema = z.nativeEnum(UpdateFirewallConfigActionSecurity1);
/** @internal */
export const UpdateFirewallConfigActionSecurity1$outboundSchema = UpdateFirewallConfigActionSecurity1$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigActionSecurity1$;
(function (UpdateFirewallConfigActionSecurity1$) {
    /** @deprecated use `UpdateFirewallConfigActionSecurity1$inboundSchema` instead. */
    UpdateFirewallConfigActionSecurity1$.inboundSchema = UpdateFirewallConfigActionSecurity1$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigActionSecurity1$outboundSchema` instead. */
    UpdateFirewallConfigActionSecurity1$.outboundSchema = UpdateFirewallConfigActionSecurity1$outboundSchema;
})(UpdateFirewallConfigActionSecurity1$ || (UpdateFirewallConfigActionSecurity1$ = {}));
/** @internal */
export const UpdateFirewallConfigRateLimitSecurityAction$inboundSchema = z.union([UpdateFirewallConfigActionSecurity1$inboundSchema, z.any()]);
/** @internal */
export const UpdateFirewallConfigRateLimitSecurityAction$outboundSchema = z.union([UpdateFirewallConfigActionSecurity1$outboundSchema, z.any()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRateLimitSecurityAction$;
(function (UpdateFirewallConfigRateLimitSecurityAction$) {
    /** @deprecated use `UpdateFirewallConfigRateLimitSecurityAction$inboundSchema` instead. */
    UpdateFirewallConfigRateLimitSecurityAction$.inboundSchema = UpdateFirewallConfigRateLimitSecurityAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRateLimitSecurityAction$outboundSchema` instead. */
    UpdateFirewallConfigRateLimitSecurityAction$.outboundSchema = UpdateFirewallConfigRateLimitSecurityAction$outboundSchema;
})(UpdateFirewallConfigRateLimitSecurityAction$ || (UpdateFirewallConfigRateLimitSecurityAction$ = {}));
export function updateFirewallConfigRateLimitSecurityActionToJSON(updateFirewallConfigRateLimitSecurityAction) {
    return JSON.stringify(UpdateFirewallConfigRateLimitSecurityAction$outboundSchema.parse(updateFirewallConfigRateLimitSecurityAction));
}
export function updateFirewallConfigRateLimitSecurityActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRateLimitSecurityAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRateLimitSecurityAction' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRateLimitSecurity1$inboundSchema = z.object({
    algo: UpdateFirewallConfigRateLimitAlgo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([UpdateFirewallConfigActionSecurity1$inboundSchema, z.any()])).optional(),
});
/** @internal */
export const UpdateFirewallConfigRateLimitSecurity1$outboundSchema = z.object({
    algo: UpdateFirewallConfigRateLimitAlgo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([UpdateFirewallConfigActionSecurity1$outboundSchema, z.any()])).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRateLimitSecurity1$;
(function (UpdateFirewallConfigRateLimitSecurity1$) {
    /** @deprecated use `UpdateFirewallConfigRateLimitSecurity1$inboundSchema` instead. */
    UpdateFirewallConfigRateLimitSecurity1$.inboundSchema = UpdateFirewallConfigRateLimitSecurity1$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRateLimitSecurity1$outboundSchema` instead. */
    UpdateFirewallConfigRateLimitSecurity1$.outboundSchema = UpdateFirewallConfigRateLimitSecurity1$outboundSchema;
})(UpdateFirewallConfigRateLimitSecurity1$ || (UpdateFirewallConfigRateLimitSecurity1$ = {}));
export function updateFirewallConfigRateLimitSecurity1ToJSON(updateFirewallConfigRateLimitSecurity1) {
    return JSON.stringify(UpdateFirewallConfigRateLimitSecurity1$outboundSchema.parse(updateFirewallConfigRateLimitSecurity1));
}
export function updateFirewallConfigRateLimitSecurity1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRateLimitSecurity1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRateLimitSecurity1' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyRateLimit$inboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRateLimitSecurity1$inboundSchema),
    z.any(),
]);
/** @internal */
export const UpdateFirewallConfigRequestBodyRateLimit$outboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRateLimitSecurity1$outboundSchema),
    z.any(),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyRateLimit$;
(function (UpdateFirewallConfigRequestBodyRateLimit$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyRateLimit$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyRateLimit$.inboundSchema = UpdateFirewallConfigRequestBodyRateLimit$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyRateLimit$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyRateLimit$.outboundSchema = UpdateFirewallConfigRequestBodyRateLimit$outboundSchema;
})(UpdateFirewallConfigRequestBodyRateLimit$ || (UpdateFirewallConfigRequestBodyRateLimit$ = {}));
export function updateFirewallConfigRequestBodyRateLimitToJSON(updateFirewallConfigRequestBodyRateLimit) {
    return JSON.stringify(UpdateFirewallConfigRequestBodyRateLimit$outboundSchema.parse(updateFirewallConfigRequestBodyRateLimit));
}
export function updateFirewallConfigRequestBodyRateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodyRateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodyRateLimit' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRedirectSecurity1$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const UpdateFirewallConfigRedirectSecurity1$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRedirectSecurity1$;
(function (UpdateFirewallConfigRedirectSecurity1$) {
    /** @deprecated use `UpdateFirewallConfigRedirectSecurity1$inboundSchema` instead. */
    UpdateFirewallConfigRedirectSecurity1$.inboundSchema = UpdateFirewallConfigRedirectSecurity1$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRedirectSecurity1$outboundSchema` instead. */
    UpdateFirewallConfigRedirectSecurity1$.outboundSchema = UpdateFirewallConfigRedirectSecurity1$outboundSchema;
})(UpdateFirewallConfigRedirectSecurity1$ || (UpdateFirewallConfigRedirectSecurity1$ = {}));
export function updateFirewallConfigRedirectSecurity1ToJSON(updateFirewallConfigRedirectSecurity1) {
    return JSON.stringify(UpdateFirewallConfigRedirectSecurity1$outboundSchema.parse(updateFirewallConfigRedirectSecurity1));
}
export function updateFirewallConfigRedirectSecurity1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRedirectSecurity1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRedirectSecurity1' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyRedirect$inboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRedirectSecurity1$inboundSchema),
    z.any(),
]);
/** @internal */
export const UpdateFirewallConfigRequestBodyRedirect$outboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRedirectSecurity1$outboundSchema),
    z.any(),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyRedirect$;
(function (UpdateFirewallConfigRequestBodyRedirect$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyRedirect$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyRedirect$.inboundSchema = UpdateFirewallConfigRequestBodyRedirect$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyRedirect$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyRedirect$.outboundSchema = UpdateFirewallConfigRequestBodyRedirect$outboundSchema;
})(UpdateFirewallConfigRequestBodyRedirect$ || (UpdateFirewallConfigRequestBodyRedirect$ = {}));
export function updateFirewallConfigRequestBodyRedirectToJSON(updateFirewallConfigRequestBodyRedirect) {
    return JSON.stringify(UpdateFirewallConfigRequestBodyRedirect$outboundSchema.parse(updateFirewallConfigRequestBodyRedirect));
}
export function updateFirewallConfigRequestBodyRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodyRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodyRedirect' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyMitigate$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$inboundSchema,
    rateLimit: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRateLimitSecurity1$inboundSchema),
        z.any(),
    ])).optional(),
    redirect: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRedirectSecurity1$inboundSchema),
        z.any(),
    ])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBodyMitigate$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest3ValueAction$outboundSchema,
    rateLimit: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRateLimitSecurity1$outboundSchema),
        z.any(),
    ])).optional(),
    redirect: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRedirectSecurity1$outboundSchema),
        z.any(),
    ])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyMitigate$;
(function (UpdateFirewallConfigRequestBodyMitigate$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyMitigate$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyMitigate$.inboundSchema = UpdateFirewallConfigRequestBodyMitigate$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyMitigate$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyMitigate$.outboundSchema = UpdateFirewallConfigRequestBodyMitigate$outboundSchema;
})(UpdateFirewallConfigRequestBodyMitigate$ || (UpdateFirewallConfigRequestBodyMitigate$ = {}));
export function updateFirewallConfigRequestBodyMitigateToJSON(updateFirewallConfigRequestBodyMitigate) {
    return JSON.stringify(UpdateFirewallConfigRequestBodyMitigate$outboundSchema.parse(updateFirewallConfigRequestBodyMitigate));
}
export function updateFirewallConfigRequestBodyMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodyMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodyMitigate' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest3Action$inboundSchema = z.object({
    mitigate: z.lazy(() => UpdateFirewallConfigRequestBodyMitigate$inboundSchema).optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest3Action$outboundSchema = z.object({
    mitigate: z.lazy(() => UpdateFirewallConfigRequestBodyMitigate$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest3Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest3Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest3Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest3Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest3Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest3Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest3Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest3Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest3Action$ || (UpdateFirewallConfigRequestBodySecurityRequest3Action$ = {}));
export function updateFirewallConfigRequestBodySecurityRequest3ActionToJSON(updateFirewallConfigRequestBodySecurityRequest3Action) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequest3Action$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequest3Action));
}
export function updateFirewallConfigRequestBodySecurityRequest3ActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequest3Action$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequest3Action' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyValue$inboundSchema = z.object({
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => UpdateFirewallConfigRequestBodyConditionGroup$inboundSchema)),
    action: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest3Action$inboundSchema),
});
/** @internal */
export const UpdateFirewallConfigRequestBodyValue$outboundSchema = z.object({
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => UpdateFirewallConfigRequestBodyConditionGroup$outboundSchema)),
    action: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest3Action$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyValue$;
(function (UpdateFirewallConfigRequestBodyValue$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyValue$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyValue$.inboundSchema = UpdateFirewallConfigRequestBodyValue$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyValue$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyValue$.outboundSchema = UpdateFirewallConfigRequestBodyValue$outboundSchema;
})(UpdateFirewallConfigRequestBodyValue$ || (UpdateFirewallConfigRequestBodyValue$ = {}));
export function updateFirewallConfigRequestBodyValueToJSON(updateFirewallConfigRequestBodyValue) {
    return JSON.stringify(UpdateFirewallConfigRequestBodyValue$outboundSchema.parse(updateFirewallConfigRequestBodyValue));
}
export function updateFirewallConfigRequestBodyValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodyValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodyValue' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBody3$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequestAction$inboundSchema,
    id: z.string(),
    value: z.lazy(() => UpdateFirewallConfigRequestBodyValue$inboundSchema),
});
/** @internal */
export const UpdateFirewallConfigRequestBody3$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequestAction$outboundSchema,
    id: z.string(),
    value: z.lazy(() => UpdateFirewallConfigRequestBodyValue$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody3$;
(function (UpdateFirewallConfigRequestBody3$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody3$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody3$.inboundSchema = UpdateFirewallConfigRequestBody3$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody3$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody3$.outboundSchema = UpdateFirewallConfigRequestBody3$outboundSchema;
})(UpdateFirewallConfigRequestBody3$ || (UpdateFirewallConfigRequestBody3$ = {}));
export function updateFirewallConfigRequestBody3ToJSON(updateFirewallConfigRequestBody3) {
    return JSON.stringify(UpdateFirewallConfigRequestBody3$outboundSchema.parse(updateFirewallConfigRequestBody3));
}
export function updateFirewallConfigRequestBody3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody3' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityAction$inboundSchema = z
    .nativeEnum(UpdateFirewallConfigRequestBodySecurityAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityAction$;
(function (UpdateFirewallConfigRequestBodySecurityAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityAction$ || (UpdateFirewallConfigRequestBodySecurityAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodyType$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodyType);
/** @internal */
export const UpdateFirewallConfigRequestBodyType$outboundSchema = UpdateFirewallConfigRequestBodyType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyType$;
(function (UpdateFirewallConfigRequestBodyType$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyType$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyType$.inboundSchema = UpdateFirewallConfigRequestBodyType$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyType$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyType$.outboundSchema = UpdateFirewallConfigRequestBodyType$outboundSchema;
})(UpdateFirewallConfigRequestBodyType$ || (UpdateFirewallConfigRequestBodyType$ = {}));
/** @internal */
export const RequestBodyOp$inboundSchema = z.nativeEnum(RequestBodyOp);
/** @internal */
export const RequestBodyOp$outboundSchema = RequestBodyOp$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyOp$;
(function (RequestBodyOp$) {
    /** @deprecated use `RequestBodyOp$inboundSchema` instead. */
    RequestBodyOp$.inboundSchema = RequestBodyOp$inboundSchema;
    /** @deprecated use `RequestBodyOp$outboundSchema` instead. */
    RequestBodyOp$.outboundSchema = RequestBodyOp$outboundSchema;
})(RequestBodyOp$ || (RequestBodyOp$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest2Value$inboundSchema = z.union([z.string(), z.array(z.string()), z.number()]);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest2Value$outboundSchema = z.union([z.string(), z.array(z.string()), z.number()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest2Value$;
(function (UpdateFirewallConfigRequestBodySecurityRequest2Value$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest2Value$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest2Value$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest2Value$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest2Value$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest2Value$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest2Value$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest2Value$ || (UpdateFirewallConfigRequestBodySecurityRequest2Value$ = {}));
export function updateFirewallConfigRequestBodySecurityRequest2ValueToJSON(updateFirewallConfigRequestBodySecurityRequest2Value) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequest2Value$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequest2Value));
}
export function updateFirewallConfigRequestBodySecurityRequest2ValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequest2Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequest2Value' from JSON`);
}
/** @internal */
export const RequestBodyConditions$inboundSchema = z.object({
    type: UpdateFirewallConfigRequestBodyType$inboundSchema,
    op: RequestBodyOp$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
/** @internal */
export const RequestBodyConditions$outboundSchema = z.object({
    type: UpdateFirewallConfigRequestBodyType$outboundSchema,
    op: RequestBodyOp$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyConditions$;
(function (RequestBodyConditions$) {
    /** @deprecated use `RequestBodyConditions$inboundSchema` instead. */
    RequestBodyConditions$.inboundSchema = RequestBodyConditions$inboundSchema;
    /** @deprecated use `RequestBodyConditions$outboundSchema` instead. */
    RequestBodyConditions$.outboundSchema = RequestBodyConditions$outboundSchema;
})(RequestBodyConditions$ || (RequestBodyConditions$ = {}));
export function requestBodyConditionsToJSON(requestBodyConditions) {
    return JSON.stringify(RequestBodyConditions$outboundSchema.parse(requestBodyConditions));
}
export function requestBodyConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBodyConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBodyConditions' from JSON`);
}
/** @internal */
export const RequestBodyConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => RequestBodyConditions$inboundSchema)),
});
/** @internal */
export const RequestBodyConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => RequestBodyConditions$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyConditionGroup$;
(function (RequestBodyConditionGroup$) {
    /** @deprecated use `RequestBodyConditionGroup$inboundSchema` instead. */
    RequestBodyConditionGroup$.inboundSchema = RequestBodyConditionGroup$inboundSchema;
    /** @deprecated use `RequestBodyConditionGroup$outboundSchema` instead. */
    RequestBodyConditionGroup$.outboundSchema = RequestBodyConditionGroup$outboundSchema;
})(RequestBodyConditionGroup$ || (RequestBodyConditionGroup$ = {}));
export function requestBodyConditionGroupToJSON(requestBodyConditionGroup) {
    return JSON.stringify(RequestBodyConditionGroup$outboundSchema.parse(requestBodyConditionGroup));
}
export function requestBodyConditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBodyConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBodyConditionGroup' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodySecurityRequest2ValueAction);
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$;
(function (UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$ || (UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$ = {}));
/** @internal */
export const RateLimitAlgo$inboundSchema = z.nativeEnum(RateLimitAlgo);
/** @internal */
export const RateLimitAlgo$outboundSchema = RateLimitAlgo$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RateLimitAlgo$;
(function (RateLimitAlgo$) {
    /** @deprecated use `RateLimitAlgo$inboundSchema` instead. */
    RateLimitAlgo$.inboundSchema = RateLimitAlgo$inboundSchema;
    /** @deprecated use `RateLimitAlgo$outboundSchema` instead. */
    RateLimitAlgo$.outboundSchema = RateLimitAlgo$outboundSchema;
})(RateLimitAlgo$ || (RateLimitAlgo$ = {}));
/** @internal */
export const UpdateFirewallConfigAction1$inboundSchema = z.nativeEnum(UpdateFirewallConfigAction1);
/** @internal */
export const UpdateFirewallConfigAction1$outboundSchema = UpdateFirewallConfigAction1$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigAction1$;
(function (UpdateFirewallConfigAction1$) {
    /** @deprecated use `UpdateFirewallConfigAction1$inboundSchema` instead. */
    UpdateFirewallConfigAction1$.inboundSchema = UpdateFirewallConfigAction1$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigAction1$outboundSchema` instead. */
    UpdateFirewallConfigAction1$.outboundSchema = UpdateFirewallConfigAction1$outboundSchema;
})(UpdateFirewallConfigAction1$ || (UpdateFirewallConfigAction1$ = {}));
/** @internal */
export const UpdateFirewallConfigRateLimitAction$inboundSchema = z.union([UpdateFirewallConfigAction1$inboundSchema, z.any()]);
/** @internal */
export const UpdateFirewallConfigRateLimitAction$outboundSchema = z.union([UpdateFirewallConfigAction1$outboundSchema, z.any()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRateLimitAction$;
(function (UpdateFirewallConfigRateLimitAction$) {
    /** @deprecated use `UpdateFirewallConfigRateLimitAction$inboundSchema` instead. */
    UpdateFirewallConfigRateLimitAction$.inboundSchema = UpdateFirewallConfigRateLimitAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRateLimitAction$outboundSchema` instead. */
    UpdateFirewallConfigRateLimitAction$.outboundSchema = UpdateFirewallConfigRateLimitAction$outboundSchema;
})(UpdateFirewallConfigRateLimitAction$ || (UpdateFirewallConfigRateLimitAction$ = {}));
export function updateFirewallConfigRateLimitActionToJSON(updateFirewallConfigRateLimitAction) {
    return JSON.stringify(UpdateFirewallConfigRateLimitAction$outboundSchema.parse(updateFirewallConfigRateLimitAction));
}
export function updateFirewallConfigRateLimitActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRateLimitAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRateLimitAction' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRateLimit1$inboundSchema = z.object({
    algo: RateLimitAlgo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([UpdateFirewallConfigAction1$inboundSchema, z.any()])).optional(),
});
/** @internal */
export const UpdateFirewallConfigRateLimit1$outboundSchema = z.object({
    algo: RateLimitAlgo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([UpdateFirewallConfigAction1$outboundSchema, z.any()])).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRateLimit1$;
(function (UpdateFirewallConfigRateLimit1$) {
    /** @deprecated use `UpdateFirewallConfigRateLimit1$inboundSchema` instead. */
    UpdateFirewallConfigRateLimit1$.inboundSchema = UpdateFirewallConfigRateLimit1$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRateLimit1$outboundSchema` instead. */
    UpdateFirewallConfigRateLimit1$.outboundSchema = UpdateFirewallConfigRateLimit1$outboundSchema;
})(UpdateFirewallConfigRateLimit1$ || (UpdateFirewallConfigRateLimit1$ = {}));
export function updateFirewallConfigRateLimit1ToJSON(updateFirewallConfigRateLimit1) {
    return JSON.stringify(UpdateFirewallConfigRateLimit1$outboundSchema.parse(updateFirewallConfigRateLimit1));
}
export function updateFirewallConfigRateLimit1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRateLimit1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRateLimit1' from JSON`);
}
/** @internal */
export const RequestBodyRateLimit$inboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRateLimit1$inboundSchema),
    z.any(),
]);
/** @internal */
export const RequestBodyRateLimit$outboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRateLimit1$outboundSchema),
    z.any(),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyRateLimit$;
(function (RequestBodyRateLimit$) {
    /** @deprecated use `RequestBodyRateLimit$inboundSchema` instead. */
    RequestBodyRateLimit$.inboundSchema = RequestBodyRateLimit$inboundSchema;
    /** @deprecated use `RequestBodyRateLimit$outboundSchema` instead. */
    RequestBodyRateLimit$.outboundSchema = RequestBodyRateLimit$outboundSchema;
})(RequestBodyRateLimit$ || (RequestBodyRateLimit$ = {}));
export function requestBodyRateLimitToJSON(requestBodyRateLimit) {
    return JSON.stringify(RequestBodyRateLimit$outboundSchema.parse(requestBodyRateLimit));
}
export function requestBodyRateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBodyRateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBodyRateLimit' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRedirect1$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const UpdateFirewallConfigRedirect1$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRedirect1$;
(function (UpdateFirewallConfigRedirect1$) {
    /** @deprecated use `UpdateFirewallConfigRedirect1$inboundSchema` instead. */
    UpdateFirewallConfigRedirect1$.inboundSchema = UpdateFirewallConfigRedirect1$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRedirect1$outboundSchema` instead. */
    UpdateFirewallConfigRedirect1$.outboundSchema = UpdateFirewallConfigRedirect1$outboundSchema;
})(UpdateFirewallConfigRedirect1$ || (UpdateFirewallConfigRedirect1$ = {}));
export function updateFirewallConfigRedirect1ToJSON(updateFirewallConfigRedirect1) {
    return JSON.stringify(UpdateFirewallConfigRedirect1$outboundSchema.parse(updateFirewallConfigRedirect1));
}
export function updateFirewallConfigRedirect1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRedirect1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRedirect1' from JSON`);
}
/** @internal */
export const RequestBodyRedirect$inboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRedirect1$inboundSchema),
    z.any(),
]);
/** @internal */
export const RequestBodyRedirect$outboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRedirect1$outboundSchema),
    z.any(),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyRedirect$;
(function (RequestBodyRedirect$) {
    /** @deprecated use `RequestBodyRedirect$inboundSchema` instead. */
    RequestBodyRedirect$.inboundSchema = RequestBodyRedirect$inboundSchema;
    /** @deprecated use `RequestBodyRedirect$outboundSchema` instead. */
    RequestBodyRedirect$.outboundSchema = RequestBodyRedirect$outboundSchema;
})(RequestBodyRedirect$ || (RequestBodyRedirect$ = {}));
export function requestBodyRedirectToJSON(requestBodyRedirect) {
    return JSON.stringify(RequestBodyRedirect$outboundSchema.parse(requestBodyRedirect));
}
export function requestBodyRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBodyRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBodyRedirect' from JSON`);
}
/** @internal */
export const RequestBodyMitigate$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$inboundSchema,
    rateLimit: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRateLimit1$inboundSchema),
        z.any(),
    ])).optional(),
    redirect: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRedirect1$inboundSchema),
        z.any(),
    ])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const RequestBodyMitigate$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityRequest2ValueAction$outboundSchema,
    rateLimit: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRateLimit1$outboundSchema),
        z.any(),
    ])).optional(),
    redirect: z.nullable(z.union([
        z.lazy(() => UpdateFirewallConfigRedirect1$outboundSchema),
        z.any(),
    ])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyMitigate$;
(function (RequestBodyMitigate$) {
    /** @deprecated use `RequestBodyMitigate$inboundSchema` instead. */
    RequestBodyMitigate$.inboundSchema = RequestBodyMitigate$inboundSchema;
    /** @deprecated use `RequestBodyMitigate$outboundSchema` instead. */
    RequestBodyMitigate$.outboundSchema = RequestBodyMitigate$outboundSchema;
})(RequestBodyMitigate$ || (RequestBodyMitigate$ = {}));
export function requestBodyMitigateToJSON(requestBodyMitigate) {
    return JSON.stringify(RequestBodyMitigate$outboundSchema.parse(requestBodyMitigate));
}
export function requestBodyMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBodyMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBodyMitigate' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest2Action$inboundSchema = z.object({
    mitigate: z.lazy(() => RequestBodyMitigate$inboundSchema).optional(),
});
/** @internal */
export const UpdateFirewallConfigRequestBodySecurityRequest2Action$outboundSchema = z.object({
    mitigate: z.lazy(() => RequestBodyMitigate$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodySecurityRequest2Action$;
(function (UpdateFirewallConfigRequestBodySecurityRequest2Action$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest2Action$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest2Action$.inboundSchema = UpdateFirewallConfigRequestBodySecurityRequest2Action$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodySecurityRequest2Action$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodySecurityRequest2Action$.outboundSchema = UpdateFirewallConfigRequestBodySecurityRequest2Action$outboundSchema;
})(UpdateFirewallConfigRequestBodySecurityRequest2Action$ || (UpdateFirewallConfigRequestBodySecurityRequest2Action$ = {}));
export function updateFirewallConfigRequestBodySecurityRequest2ActionToJSON(updateFirewallConfigRequestBodySecurityRequest2Action) {
    return JSON.stringify(UpdateFirewallConfigRequestBodySecurityRequest2Action$outboundSchema.parse(updateFirewallConfigRequestBodySecurityRequest2Action));
}
export function updateFirewallConfigRequestBodySecurityRequest2ActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBodySecurityRequest2Action$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBodySecurityRequest2Action' from JSON`);
}
/** @internal */
export const RequestBodyValue$inboundSchema = z.object({
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => RequestBodyConditionGroup$inboundSchema)),
    action: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest2Action$inboundSchema),
});
/** @internal */
export const RequestBodyValue$outboundSchema = z.object({
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => RequestBodyConditionGroup$outboundSchema)),
    action: z.lazy(() => UpdateFirewallConfigRequestBodySecurityRequest2Action$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RequestBodyValue$;
(function (RequestBodyValue$) {
    /** @deprecated use `RequestBodyValue$inboundSchema` instead. */
    RequestBodyValue$.inboundSchema = RequestBodyValue$inboundSchema;
    /** @deprecated use `RequestBodyValue$outboundSchema` instead. */
    RequestBodyValue$.outboundSchema = RequestBodyValue$outboundSchema;
})(RequestBodyValue$ || (RequestBodyValue$ = {}));
export function requestBodyValueToJSON(requestBodyValue) {
    return JSON.stringify(RequestBodyValue$outboundSchema.parse(requestBodyValue));
}
export function requestBodyValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBodyValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBodyValue' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBody2$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityAction$inboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.lazy(() => RequestBodyValue$inboundSchema),
});
/** @internal */
export const UpdateFirewallConfigRequestBody2$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodySecurityAction$outboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.lazy(() => RequestBodyValue$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody2$;
(function (UpdateFirewallConfigRequestBody2$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody2$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody2$.inboundSchema = UpdateFirewallConfigRequestBody2$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody2$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody2$.outboundSchema = UpdateFirewallConfigRequestBody2$outboundSchema;
})(UpdateFirewallConfigRequestBody2$ || (UpdateFirewallConfigRequestBody2$ = {}));
export function updateFirewallConfigRequestBody2ToJSON(updateFirewallConfigRequestBody2) {
    return JSON.stringify(UpdateFirewallConfigRequestBody2$outboundSchema.parse(updateFirewallConfigRequestBody2));
}
export function updateFirewallConfigRequestBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody2' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBodyAction$inboundSchema = z.nativeEnum(UpdateFirewallConfigRequestBodyAction);
/** @internal */
export const UpdateFirewallConfigRequestBodyAction$outboundSchema = UpdateFirewallConfigRequestBodyAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBodyAction$;
(function (UpdateFirewallConfigRequestBodyAction$) {
    /** @deprecated use `UpdateFirewallConfigRequestBodyAction$inboundSchema` instead. */
    UpdateFirewallConfigRequestBodyAction$.inboundSchema = UpdateFirewallConfigRequestBodyAction$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBodyAction$outboundSchema` instead. */
    UpdateFirewallConfigRequestBodyAction$.outboundSchema = UpdateFirewallConfigRequestBodyAction$outboundSchema;
})(UpdateFirewallConfigRequestBodyAction$ || (UpdateFirewallConfigRequestBodyAction$ = {}));
/** @internal */
export const UpdateFirewallConfigRequestBody1$inboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodyAction$inboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.boolean(),
});
/** @internal */
export const UpdateFirewallConfigRequestBody1$outboundSchema = z.object({
    action: UpdateFirewallConfigRequestBodyAction$outboundSchema,
    id: z.nullable(z.any()).optional(),
    value: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody1$;
(function (UpdateFirewallConfigRequestBody1$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody1$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody1$.inboundSchema = UpdateFirewallConfigRequestBody1$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody1$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody1$.outboundSchema = UpdateFirewallConfigRequestBody1$outboundSchema;
})(UpdateFirewallConfigRequestBody1$ || (UpdateFirewallConfigRequestBody1$ = {}));
export function updateFirewallConfigRequestBody1ToJSON(updateFirewallConfigRequestBody1) {
    return JSON.stringify(UpdateFirewallConfigRequestBody1$outboundSchema.parse(updateFirewallConfigRequestBody1));
}
export function updateFirewallConfigRequestBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody1' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequestBody$inboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRequestBody3$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody5$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody6$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody9$inboundSchema),
    z.lazy(() => RequestBody11$inboundSchema),
    z.lazy(() => RequestBody12$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody1$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody2$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody4$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody8$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody10$inboundSchema),
    z.lazy(() => RequestBody13$inboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody7$inboundSchema),
]);
/** @internal */
export const UpdateFirewallConfigRequestBody$outboundSchema = z.union([
    z.lazy(() => UpdateFirewallConfigRequestBody3$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody5$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody6$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody9$outboundSchema),
    z.lazy(() => RequestBody11$outboundSchema),
    z.lazy(() => RequestBody12$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody1$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody2$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody4$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody8$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody10$outboundSchema),
    z.lazy(() => RequestBody13$outboundSchema),
    z.lazy(() => UpdateFirewallConfigRequestBody7$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequestBody$;
(function (UpdateFirewallConfigRequestBody$) {
    /** @deprecated use `UpdateFirewallConfigRequestBody$inboundSchema` instead. */
    UpdateFirewallConfigRequestBody$.inboundSchema = UpdateFirewallConfigRequestBody$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequestBody$outboundSchema` instead. */
    UpdateFirewallConfigRequestBody$.outboundSchema = UpdateFirewallConfigRequestBody$outboundSchema;
})(UpdateFirewallConfigRequestBody$ || (UpdateFirewallConfigRequestBody$ = {}));
export function updateFirewallConfigRequestBodyToJSON(updateFirewallConfigRequestBody) {
    return JSON.stringify(UpdateFirewallConfigRequestBody$outboundSchema.parse(updateFirewallConfigRequestBody));
}
export function updateFirewallConfigRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequestBody' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.union([
        z.lazy(() => UpdateFirewallConfigRequestBody3$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody5$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody6$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody9$inboundSchema),
        z.lazy(() => RequestBody11$inboundSchema),
        z.lazy(() => RequestBody12$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody1$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody2$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody4$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody8$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody10$inboundSchema),
        z.lazy(() => RequestBody13$inboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody7$inboundSchema),
    ]),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateFirewallConfigRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.union([
        z.lazy(() => UpdateFirewallConfigRequestBody3$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody5$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody6$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody9$outboundSchema),
        z.lazy(() => RequestBody11$outboundSchema),
        z.lazy(() => RequestBody12$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody1$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody2$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody4$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody8$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody10$outboundSchema),
        z.lazy(() => RequestBody13$outboundSchema),
        z.lazy(() => UpdateFirewallConfigRequestBody7$outboundSchema),
    ]),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigRequest$;
(function (UpdateFirewallConfigRequest$) {
    /** @deprecated use `UpdateFirewallConfigRequest$inboundSchema` instead. */
    UpdateFirewallConfigRequest$.inboundSchema = UpdateFirewallConfigRequest$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigRequest$outboundSchema` instead. */
    UpdateFirewallConfigRequest$.outboundSchema = UpdateFirewallConfigRequest$outboundSchema;
})(UpdateFirewallConfigRequest$ || (UpdateFirewallConfigRequest$ = {}));
export function updateFirewallConfigRequestToJSON(updateFirewallConfigRequest) {
    return JSON.stringify(UpdateFirewallConfigRequest$outboundSchema.parse(updateFirewallConfigRequest));
}
export function updateFirewallConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigRequest' from JSON`);
}
/** @internal */
export const UpdateFirewallConfigResponseBody$inboundSchema = z.object({});
/** @internal */
export const UpdateFirewallConfigResponseBody$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateFirewallConfigResponseBody$;
(function (UpdateFirewallConfigResponseBody$) {
    /** @deprecated use `UpdateFirewallConfigResponseBody$inboundSchema` instead. */
    UpdateFirewallConfigResponseBody$.inboundSchema = UpdateFirewallConfigResponseBody$inboundSchema;
    /** @deprecated use `UpdateFirewallConfigResponseBody$outboundSchema` instead. */
    UpdateFirewallConfigResponseBody$.outboundSchema = UpdateFirewallConfigResponseBody$outboundSchema;
})(UpdateFirewallConfigResponseBody$ || (UpdateFirewallConfigResponseBody$ = {}));
export function updateFirewallConfigResponseBodyToJSON(updateFirewallConfigResponseBody) {
    return JSON.stringify(UpdateFirewallConfigResponseBody$outboundSchema.parse(updateFirewallConfigResponseBody));
}
export function updateFirewallConfigResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFirewallConfigResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFirewallConfigResponseBody' from JSON`);
}
