// @vercel/sdk/esm/models/vercelbadrequesterror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { VercelError } from "./vercelerror.js";
export class VercelBadRequestError extends VercelError {
    constructor(err, httpMeta) {
        const message = err.error?.message
            || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.name = "VercelBadRequestError";
    }
}
/** @internal */
export const ErrorT$inboundSchema = z
    .object({
    code: z.string(),
    message: z.string(),
});
/** @internal */
export const ErrorT$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ErrorT$;
(function (ErrorT$) {
    /** @deprecated use `ErrorT$inboundSchema` instead. */
    ErrorT$.inboundSchema = ErrorT$inboundSchema;
    /** @deprecated use `ErrorT$outboundSchema` instead. */
    ErrorT$.outboundSchema = ErrorT$outboundSchema;
})(ErrorT$ || (ErrorT$ = {}));
export function errorToJSON(errorT) {
    return JSON.stringify(ErrorT$outboundSchema.parse(errorT));
}
export function errorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ErrorT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ErrorT' from JSON`);
}
/** @internal */
export const VercelBadRequestError$inboundSchema = z.object({
    error: z.lazy(() => ErrorT$inboundSchema),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new VercelBadRequestError(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const VercelBadRequestError$outboundSchema = z.instanceof(VercelBadRequestError)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.lazy(() => ErrorT$outboundSchema),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var VercelBadRequestError$;
(function (VercelBadRequestError$) {
    /** @deprecated use `VercelBadRequestError$inboundSchema` instead. */
    VercelBadRequestError$.inboundSchema = VercelBadRequestError$inboundSchema;
    /** @deprecated use `VercelBadRequestError$outboundSchema` instead. */
    VercelBadRequestError$.outboundSchema = VercelBadRequestError$outboundSchema;
})(VercelBadRequestError$ || (VercelBadRequestError$ = {}));
