// @vercel/sdk/src/mcp-server/tools/accessGroupsListAccessGroups.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { accessGroupsListAccessGroups } from "../../funcs/accessGroupsListAccessGroups.js";
import { ListAccessGroupsRequest$inboundSchema } from "../../models/listaccessgroupsop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: ListAccessGroupsRequest$inboundSchema,
};

export const tool$accessGroupsListAccessGroups: ToolDefinition<typeof args> = {
  name: "access-groups-list-access-groups",
  description: `List access groups for a team, project or member

List access groups`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await accessGroupsListAccessGroups(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
