// @vercel/sdk/src/mcp-server/tools/artifactsDownloadArtifact.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { artifactsDownloadArtifact } from "../../funcs/artifactsDownloadArtifact.js";
import { DownloadArtifactRequest$inboundSchema } from "../../models/downloadartifactop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: DownloadArtifactRequest$inboundSchema,
};

export const tool$artifactsDownloadArtifact: ToolDefinition<typeof args> = {
  name: "artifacts-download-artifact",
  description: `Download a cache artifact

Downloads a cache artifact indentified by its \`hash\` specified on the request path. The artifact is downloaded as an octet-stream. The client should verify the content-length header and response body.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await artifactsDownloadArtifact(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
