// @vercel/sdk/src/mcp-server/tools/artifactsArtifactQuery.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { artifactsArtifactQuery } from "../../funcs/artifactsArtifactQuery.js";
import { ArtifactQueryRequest$inboundSchema } from "../../models/artifactqueryop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: ArtifactQueryRequest$inboundSchema,
};

export const tool$artifactsArtifactQuery: ToolDefinition<typeof args> = {
  name: "artifacts-artifact-query",
  description: `Query information about an artifact

Query information about an array of artifacts.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await artifactsArtifactQuery(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
