// @vercel/sdk/src/mcp-server/tools/authenticationDeleteAuthToken.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { authenticationDeleteAuthToken } from "../../funcs/authenticationDeleteAuthToken.js";
import { DeleteAuthTokenRequest$inboundSchema } from "../../models/deleteauthtokenop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: DeleteAuthTokenRequest$inboundSchema,
};

export const tool$authenticationDeleteAuthToken: ToolDefinition<typeof args> = {
  name: "authentication-delete-auth-token",
  description: `Delete an authentication token

Invalidate an authentication token, such that it will no longer be valid for future HTTP requests.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await authenticationDeleteAuthToken(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
