// @vercel/sdk/src/mcp-server/tools/domainsGetDomainTransfer.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { domainsGetDomainTransfer } from "../../funcs/domainsGetDomainTransfer.js";
import { GetDomainTransferRequest$inboundSchema } from "../../models/getdomaintransferop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetDomainTransferRequest$inboundSchema,
};

export const tool$domainsGetDomainTransfer: ToolDefinition<typeof args> = {
  name: "domains-get-domain-transfer",
  description: `Get domain transfer info.

Fetch domain transfer availability or transfer status if a transfer is in progress.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await domainsGetDomainTransfer(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
