// @vercel/sdk/src/mcp-server/tools/domainsCreateOrTransferDomain.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { domainsCreateOrTransferDomain } from "../../funcs/domainsCreateOrTransferDomain.js";
import { CreateOrTransferDomainRequest$inboundSchema } from "../../models/createortransferdomainop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: CreateOrTransferDomainRequest$inboundSchema,
};

export const tool$domainsCreateOrTransferDomain: ToolDefinition<typeof args> = {
  name: "domains-create-or-transfer-domain",
  description: `Register or transfer-in a new Domain

This endpoint is used for adding a new apex domain name with Vercel for the authenticating user. Can also be used for initiating a domain transfer request from an external Registrar to Vercel.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await domainsCreateOrTransferDomain(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
