// @vercel/sdk/src/mcp-server/tools/domainsCheckDomainStatus.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { domainsCheckDomainStatus } from "../../funcs/domainsCheckDomainStatus.js";
import { CheckDomainStatusRequest$inboundSchema } from "../../models/checkdomainstatusop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: CheckDomainStatusRequest$inboundSchema,
};

export const tool$domainsCheckDomainStatus: ToolDefinition<typeof args> = {
  name: "domains-check-domain-status",
  description: `Check a Domain Availability

Check if a domain name is available for purchase.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await domainsCheckDomainStatus(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
