// @vercel/sdk/src/mcp-server/tools/edgeConfigCreateEdgeConfig.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { edgeConfigCreateEdgeConfig } from "../../funcs/edgeConfigCreateEdgeConfig.js";
import { CreateEdgeConfigRequest$inboundSchema } from "../../models/createedgeconfigop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: CreateEdgeConfigRequest$inboundSchema,
};

export const tool$edgeConfigCreateEdgeConfig: ToolDefinition<typeof args> = {
  name: "edge-config-create-edge-config",
  description: `Create an Edge Config

Creates an Edge Config.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await edgeConfigCreateEdgeConfig(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
