// @vercel/sdk/src/mcp-server/tools/environmentRemoveCustomEnvironment.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { environmentRemoveCustomEnvironment } from "../../funcs/environmentRemoveCustomEnvironment.js";
import { RemoveCustomEnvironmentRequest$inboundSchema } from "../../models/removecustomenvironmentop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: RemoveCustomEnvironmentRequest$inboundSchema,
};

export const tool$environmentRemoveCustomEnvironment: ToolDefinition<
  typeof args
> = {
  name: "environment-remove-custom-environment",
  description: `Remove a custom environment

Remove a custom environment for the project. Must not be named 'Production' or 'Preview'.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await environmentRemoveCustomEnvironment(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
