// @vercel/sdk/src/mcp-server/tools/integrationsGetConfigurations.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { integrationsGetConfigurations } from "../../funcs/integrationsGetConfigurations.js";
import { GetConfigurationsRequest$inboundSchema } from "../../models/getconfigurationsop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetConfigurationsRequest$inboundSchema,
};

export const tool$integrationsGetConfigurations: ToolDefinition<typeof args> = {
  name: "integrations-get-configurations",
  description: `Get configurations for the authenticated user or team

Allows to retrieve all configurations for an authenticated integration. When the \`project\` view is used, configurations generated for the authorization flow will be filtered out of the results.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await integrationsGetConfigurations(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
