// @vercel/sdk/src/mcp-server/tools/marketplaceGetInvoice.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { marketplaceGetInvoice } from "../../funcs/marketplaceGetInvoice.js";
import { GetInvoiceRequest$inboundSchema } from "../../models/getinvoiceop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetInvoiceRequest$inboundSchema,
};

export const tool$marketplaceGetInvoice: ToolDefinition<typeof args> = {
  name: "marketplace-get-invoice",
  description: `Get Invoice

Get Invoice details and status for a given invoice ID.<br/> <br/> See Billing Events with Webhooks documentation on how to receive invoice events. This endpoint is used to retrieve the invoice details.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await marketplaceGetInvoice(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
