// @vercel/sdk/src/mcp-server/tools/marketplaceSubmitBillingData.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { marketplaceSubmitBillingData } from "../../funcs/marketplaceSubmitBillingData.js";
import { SubmitBillingDataRequest$inboundSchema } from "../../models/submitbillingdataop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: SubmitBillingDataRequest$inboundSchema,
};

export const tool$marketplaceSubmitBillingData: ToolDefinition<typeof args> = {
  name: "marketplace-submit-billing-data",
  description: `Submit Billing Data

Sends the billing and usage data. The partner should do this at least once a day and ideally once per hour. <br/> Use the \`credentials.access_token\` we provided in the [Upsert Installation](#upsert-installation) body to authorize this request.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await marketplaceSubmitBillingData(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    return formatResult(void 0, apiCall);
  },
};
