// @vercel/sdk/src/mcp-server/tools/marketplaceDeleteInstallationIntegrationConfiguration.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { marketplaceDeleteInstallationIntegrationConfiguration } from "../../funcs/marketplaceDeleteInstallationIntegrationConfiguration.js";
import {
  DeleteV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationItemsItemIdRequest$inboundSchema,
} from "../../models/deletev1installationsintegrationconfigurationidresourcesresourceidexperimentationitemsitemidop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request:
    DeleteV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationItemsItemIdRequest$inboundSchema,
};

export const tool$marketplaceDeleteInstallationIntegrationConfiguration:
  ToolDefinition<typeof args> = {
    name: "marketplace-delete-installation-integration-configuration",
    description: `Delete an existing experimentation item

Delete an existing experimentation item`,
    args,
    tool: async (client, args, ctx) => {
      const [result, apiCall] =
        await marketplaceDeleteInstallationIntegrationConfiguration(
          client,
          args.request,
          { fetchOptions: { signal: ctx.signal } },
        ).$inspect();

      if (!result.ok) {
        return {
          content: [{ type: "text", text: result.error.message }],
          isError: true,
        };
      }

      return formatResult(void 0, apiCall);
    },
  };
