// @vercel/sdk/src/mcp-server/tools/projectsAcceptProjectTransferRequest.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectsAcceptProjectTransferRequest } from "../../funcs/projectsAcceptProjectTransferRequest.js";
import { AcceptProjectTransferRequestRequest$inboundSchema } from "../../models/acceptprojecttransferrequestop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: AcceptProjectTransferRequestRequest$inboundSchema,
};

export const tool$projectsAcceptProjectTransferRequest: ToolDefinition<
  typeof args
> = {
  name: "projects-accept-project-transfer-request",
  description: `Accept project transfer request

Accept a project transfer request initated by another team. <br/> The \`code\` is generated using the \`POST /projects/:idOrName/transfer-request\` endpoint.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectsAcceptProjectTransferRequest(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
