// @vercel/sdk/src/mcp-server/tools/projectsRemoveProjectEnv.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectsRemoveProjectEnv } from "../../funcs/projectsRemoveProjectEnv.js";
import { RemoveProjectEnvRequest$inboundSchema } from "../../models/removeprojectenvop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: RemoveProjectEnvRequest$inboundSchema,
};

export const tool$projectsRemoveProjectEnv: ToolDefinition<typeof args> = {
  name: "projects-remove-project-env",
  description: `Remove an environment variable

Delete a specific environment variable for a given project by passing the environment variable identifier and either passing the project \`id\` or \`name\` in the URL.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectsRemoveProjectEnv(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
