// @vercel/sdk/src/mcp-server/tools/projectsMoveProjectDomain.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectsMoveProjectDomain } from "../../funcs/projectsMoveProjectDomain.js";
import { MoveProjectDomainRequest$inboundSchema } from "../../models/moveprojectdomainop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: MoveProjectDomainRequest$inboundSchema,
};

export const tool$projectsMoveProjectDomain: ToolDefinition<typeof args> = {
  name: "projects-move-project-domain",
  description: `Move a project domain

Move one project's domain to another project. Also allows the move of all redirects pointed to that domain in the same project.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectsMoveProjectDomain(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
