// @vercel/sdk/src/mcp-server/tools/projectMembersRemoveProjectMember.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectMembersRemoveProjectMember } from "../../funcs/projectMembersRemoveProjectMember.js";
import { RemoveProjectMemberRequest$inboundSchema } from "../../models/removeprojectmemberop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: RemoveProjectMemberRequest$inboundSchema,
};

export const tool$projectMembersRemoveProjectMember: ToolDefinition<
  typeof args
> = {
  name: "project-members-remove-project-member",
  description: `Remove a Project Member

Remove a member from a specific project`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectMembersRemoveProjectMember(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
