// @vercel/sdk/src/mcp-server/tools/projectsVerifyProjectDomain.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectsVerifyProjectDomain } from "../../funcs/projectsVerifyProjectDomain.js";
import { VerifyProjectDomainRequest$inboundSchema } from "../../models/verifyprojectdomainop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: VerifyProjectDomainRequest$inboundSchema,
};

export const tool$projectsVerifyProjectDomain: ToolDefinition<typeof args> = {
  name: "projects-verify-project-domain",
  description: `Verify project domain

Attempts to verify a project domain with \`verified = false\` by checking the correctness of the project domain's \`verification\` challenge.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectsVerifyProjectDomain(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
