// @vercel/sdk/src/mcp-server/tools/securityGetActiveAttackStatus.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { securityGetActiveAttackStatus } from "../../funcs/securityGetActiveAttackStatus.js";
import { GetActiveAttackStatusRequest$inboundSchema } from "../../models/getactiveattackstatusop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetActiveAttackStatusRequest$inboundSchema,
};

export const tool$securityGetActiveAttackStatus: ToolDefinition<typeof args> = {
  name: "security-get-active-attack-status",
  description: `Read active attack data

Retrieve active attack data within the last 24h window`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await securityGetActiveAttackStatus(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
