// @vercel/sdk/src/mcp-server/tools/teamsDeleteTeam.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { teamsDeleteTeam } from "../../funcs/teamsDeleteTeam.js";
import { DeleteTeamRequest$inboundSchema } from "../../models/deleteteamop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: DeleteTeamRequest$inboundSchema,
};

export const tool$teamsDeleteTeam: ToolDefinition<typeof args> = {
  name: "teams-delete-team",
  description: `Delete a Team

Delete a team under your account. You need to send a \`DELETE\` request with the desired team \`id\`. An optional array of reasons for deletion may also be sent.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await teamsDeleteTeam(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
