// @vercel/sdk/src/mcp-server/tools/teamsDeleteTeamInviteCode.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { teamsDeleteTeamInviteCode } from "../../funcs/teamsDeleteTeamInviteCode.js";
import { DeleteTeamInviteCodeRequest$inboundSchema } from "../../models/deleteteaminvitecodeop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: DeleteTeamInviteCodeRequest$inboundSchema,
};

export const tool$teamsDeleteTeamInviteCode: ToolDefinition<typeof args> = {
  name: "teams-delete-team-invite-code",
  description: `Delete a Team invite code

Delete an active Team invite code.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await teamsDeleteTeamInviteCode(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
