// @vercel/sdk/src/mcp-server/tools/userRequestDelete.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { userRequestDelete } from "../../funcs/userRequestDelete.js";
import { RequestDeleteRequestBody$inboundSchema } from "../../models/requestdeleteop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: RequestDeleteRequestBody$inboundSchema,
};

export const tool$userRequestDelete: ToolDefinition<typeof args> = {
  name: "user-request-delete",
  description: `Delete User Account

Initiates the deletion process for the currently authenticated User, by sending a deletion confirmation email. The email contains a link that the user needs to visit in order to proceed with the deletion process.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await userRequestDelete(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
