// @vercel/sdk/esm/mcp-server/tools/accessGroupsCreateAccessGroup.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { accessGroupsCreateAccessGroup } from "../../funcs/accessGroupsCreateAccessGroup.js";
import { CreateAccessGroupRequest$inboundSchema } from "../../models/createaccessgroupop.js";
import { formatResult } from "../tools.js";
const args = {
    request: CreateAccessGroupRequest$inboundSchema,
};
export const tool$accessGroupsCreateAccessGroup = {
    name: "access-groups-create-access-group",
    description: `Creates an access group

Allows to create an access group`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await accessGroupsCreateAccessGroup(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
