// @vercel/sdk/esm/mcp-server/tools/edgeConfigGetEdgeConfigSchema.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigGetEdgeConfigSchema } from "../../funcs/edgeConfigGetEdgeConfigSchema.js";
import { GetEdgeConfigSchemaRequest$inboundSchema } from "../../models/getedgeconfigschemaop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetEdgeConfigSchemaRequest$inboundSchema,
};
export const tool$edgeConfigGetEdgeConfigSchema = {
    name: "edge-config-get-edge-config-schema",
    description: `Get Edge Config schema

Returns the schema of an Edge Config.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigGetEdgeConfigSchema(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
