// @vercel/sdk/esm/mcp-server/tools/getV1IntegrationsIntegrationIntegrationIdOrSlugProductsProductIdOrSlugPlans.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { getV1IntegrationsIntegrationIntegrationIdOrSlugProductsProductIdOrSlugPlans } from "../../funcs/getV1IntegrationsIntegrationIntegrationIdOrSlugProductsProductIdOrSlugPlans.js";
import { GetV1IntegrationsIntegrationIntegrationIdOrSlugProductsProductIdOrSlugPlansRequest$inboundSchema, } from "../../models/getv1integrationsintegrationintegrationidorslugproductsproductidorslugplansop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetV1IntegrationsIntegrationIntegrationIdOrSlugProductsProductIdOrSlugPlansRequest$inboundSchema,
};
export const tool$getV1IntegrationsIntegrationIntegrationIdOrSlugProductsProductIdOrSlugPlans = {
    name: "get-v1-integrations-integration-integration-id-or-slug-products-product-id-or-slug-plans",
    description: ``,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await getV1IntegrationsIntegrationIntegrationIdOrSlugProductsProductIdOrSlugPlans(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
