// @vercel/sdk/esm/mcp-server/tools/marketplaceUpdateInstallationIntegrationConfiguration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { marketplaceUpdateInstallationIntegrationConfiguration } from "../../funcs/marketplaceUpdateInstallationIntegrationConfiguration.js";
import { PatchV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationItemsItemIdRequest$inboundSchema, } from "../../models/patchv1installationsintegrationconfigurationidresourcesresourceidexperimentationitemsitemidop.js";
import { formatResult } from "../tools.js";
const args = {
    request: PatchV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationItemsItemIdRequest$inboundSchema,
};
export const tool$marketplaceUpdateInstallationIntegrationConfiguration = {
    name: "marketplace-update-installation-integration-configuration",
    description: `Patch an existing experimentation item

Patch an existing experimentation item`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await marketplaceUpdateInstallationIntegrationConfiguration(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return formatResult(void 0, apiCall);
    },
};
