// @vercel/sdk/esm/mcp-server/tools/marketplaceDeleteIntegrationResource.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { marketplaceDeleteIntegrationResource } from "../../funcs/marketplaceDeleteIntegrationResource.js";
import { DeleteIntegrationResourceRequest$inboundSchema } from "../../models/deleteintegrationresourceop.js";
import { formatResult } from "../tools.js";
const args = {
    request: DeleteIntegrationResourceRequest$inboundSchema,
};
export const tool$marketplaceDeleteIntegrationResource = {
    name: "marketplace-delete-integration-resource",
    description: `Delete Integration Resource

Delete a resource owned by the selected installation ID.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await marketplaceDeleteIntegrationResource(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return formatResult(void 0, apiCall);
    },
};
