// @vercel/sdk/esm/mcp-server/tools/projectsUpdateProjectDataCache.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { projectsUpdateProjectDataCache } from "../../funcs/projectsUpdateProjectDataCache.js";
import { UpdateProjectDataCacheRequest$inboundSchema } from "../../models/updateprojectdatacacheop.js";
import { formatResult } from "../tools.js";
const args = {
    request: UpdateProjectDataCacheRequest$inboundSchema,
};
export const tool$projectsUpdateProjectDataCache = {
    name: "projects-update-project-data-cache",
    description: `Update the data cache feature

Update the data cache feature on a project.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await projectsUpdateProjectDataCache(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
