// @vercel/sdk/esm/mcp-server/tools/securityPutFirewallConfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { securityPutFirewallConfig } from "../../funcs/securityPutFirewallConfig.js";
import { PutFirewallConfigRequest$inboundSchema } from "../../models/putfirewallconfigop.js";
import { formatResult } from "../tools.js";
const args = {
    request: PutFirewallConfigRequest$inboundSchema,
};
export const tool$securityPutFirewallConfig = {
    name: "security-put-firewall-config",
    description: `Put Firewall Configuration

Set the firewall configuration to provided rules and settings. Creates or overwrite the existing firewall configuration.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await securityPutFirewallConfig(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
