// @vercel/sdk/esm/models/deleteedgeconfigschemaop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteEdgeConfigSchemaRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const DeleteEdgeConfigSchemaRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteEdgeConfigSchemaRequest$;
(function (DeleteEdgeConfigSchemaRequest$) {
    /** @deprecated use `DeleteEdgeConfigSchemaRequest$inboundSchema` instead. */
    DeleteEdgeConfigSchemaRequest$.inboundSchema = DeleteEdgeConfigSchemaRequest$inboundSchema;
    /** @deprecated use `DeleteEdgeConfigSchemaRequest$outboundSchema` instead. */
    DeleteEdgeConfigSchemaRequest$.outboundSchema = DeleteEdgeConfigSchemaRequest$outboundSchema;
})(DeleteEdgeConfigSchemaRequest$ || (DeleteEdgeConfigSchemaRequest$ = {}));
export function deleteEdgeConfigSchemaRequestToJSON(deleteEdgeConfigSchemaRequest) {
    return JSON.stringify(DeleteEdgeConfigSchemaRequest$outboundSchema.parse(deleteEdgeConfigSchemaRequest));
}
export function deleteEdgeConfigSchemaRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteEdgeConfigSchemaRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteEdgeConfigSchemaRequest' from JSON`);
}
