// @vercel/sdk/esm/models/deleteaccessgroupprojectop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteAccessGroupProjectRequest$inboundSchema = z.object({
    accessGroupIdOrName: z.string(),
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const DeleteAccessGroupProjectRequest$outboundSchema = z.object({
    accessGroupIdOrName: z.string(),
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteAccessGroupProjectRequest$;
(function (DeleteAccessGroupProjectRequest$) {
    /** @deprecated use `DeleteAccessGroupProjectRequest$inboundSchema` instead. */
    DeleteAccessGroupProjectRequest$.inboundSchema = DeleteAccessGroupProjectRequest$inboundSchema;
    /** @deprecated use `DeleteAccessGroupProjectRequest$outboundSchema` instead. */
    DeleteAccessGroupProjectRequest$.outboundSchema = DeleteAccessGroupProjectRequest$outboundSchema;
})(DeleteAccessGroupProjectRequest$ || (DeleteAccessGroupProjectRequest$ = {}));
export function deleteAccessGroupProjectRequestToJSON(deleteAccessGroupProjectRequest) {
    return JSON.stringify(DeleteAccessGroupProjectRequest$outboundSchema.parse(deleteAccessGroupProjectRequest));
}
export function deleteAccessGroupProjectRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteAccessGroupProjectRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteAccessGroupProjectRequest' from JSON`);
}
