// @vercel/sdk/esm/models/edgeconfigitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { EdgeConfigItemValue$inboundSchema, EdgeConfigItemValue$outboundSchema, } from "./edgeconfigitemvalue.js";
/** @internal */
export const EdgeConfigItem$inboundSchema = z.object({
    key: z.string(),
    value: z.nullable(EdgeConfigItemValue$inboundSchema),
    description: z.string().optional(),
    edgeConfigId: z.string(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const EdgeConfigItem$outboundSchema = z.object({
    key: z.string(),
    value: z.nullable(EdgeConfigItemValue$outboundSchema),
    description: z.string().optional(),
    edgeConfigId: z.string(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var EdgeConfigItem$;
(function (EdgeConfigItem$) {
    /** @deprecated use `EdgeConfigItem$inboundSchema` instead. */
    EdgeConfigItem$.inboundSchema = EdgeConfigItem$inboundSchema;
    /** @deprecated use `EdgeConfigItem$outboundSchema` instead. */
    EdgeConfigItem$.outboundSchema = EdgeConfigItem$outboundSchema;
})(EdgeConfigItem$ || (EdgeConfigItem$ = {}));
export function edgeConfigItemToJSON(edgeConfigItem) {
    return JSON.stringify(EdgeConfigItem$outboundSchema.parse(edgeConfigItem));
}
export function edgeConfigItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EdgeConfigItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EdgeConfigItem' from JSON`);
}
