// @vercel/sdk/esm/models/getedgeconfigitemsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetEdgeConfigItemsRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetEdgeConfigItemsRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetEdgeConfigItemsRequest$;
(function (GetEdgeConfigItemsRequest$) {
    /** @deprecated use `GetEdgeConfigItemsRequest$inboundSchema` instead. */
    GetEdgeConfigItemsRequest$.inboundSchema = GetEdgeConfigItemsRequest$inboundSchema;
    /** @deprecated use `GetEdgeConfigItemsRequest$outboundSchema` instead. */
    GetEdgeConfigItemsRequest$.outboundSchema = GetEdgeConfigItemsRequest$outboundSchema;
})(GetEdgeConfigItemsRequest$ || (GetEdgeConfigItemsRequest$ = {}));
export function getEdgeConfigItemsRequestToJSON(getEdgeConfigItemsRequest) {
    return JSON.stringify(GetEdgeConfigItemsRequest$outboundSchema.parse(getEdgeConfigItemsRequest));
}
export function getEdgeConfigItemsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigItemsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigItemsRequest' from JSON`);
}
