// @vercel/sdk/esm/models/getedgeconfigtokensop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetEdgeConfigTokensRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetEdgeConfigTokensRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetEdgeConfigTokensRequest$;
(function (GetEdgeConfigTokensRequest$) {
    /** @deprecated use `GetEdgeConfigTokensRequest$inboundSchema` instead. */
    GetEdgeConfigTokensRequest$.inboundSchema = GetEdgeConfigTokensRequest$inboundSchema;
    /** @deprecated use `GetEdgeConfigTokensRequest$outboundSchema` instead. */
    GetEdgeConfigTokensRequest$.outboundSchema = GetEdgeConfigTokensRequest$outboundSchema;
})(GetEdgeConfigTokensRequest$ || (GetEdgeConfigTokensRequest$ = {}));
export function getEdgeConfigTokensRequestToJSON(getEdgeConfigTokensRequest) {
    return JSON.stringify(GetEdgeConfigTokensRequest$outboundSchema.parse(getEdgeConfigTokensRequest));
}
export function getEdgeConfigTokensRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigTokensRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigTokensRequest' from JSON`);
}
