// @vercel/sdk/esm/models/inviteusertoteamop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The role of the user to invite
 */
export const InviteUserToTeamRole = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
/**
 * Sets the project roles for the invited user
 */
export const InviteUserToTeamTeamsRole = {
    Admin: "ADMIN",
    ProjectViewer: "PROJECT_VIEWER",
    ProjectDeveloper: "PROJECT_DEVELOPER",
};
/**
 * The role used for the invitation
 */
export const InviteUserToTeamTeamsResponseRole = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
/**
 * The team roles of the user
 */
export const InviteUserToTeamTeamRoles = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
/**
 * The team permissions of the user
 */
export const InviteUserToTeamTeamPermissions = {
    IntegrationManager: "IntegrationManager",
    CreateProject: "CreateProject",
    FullProductionDeployment: "FullProductionDeployment",
    UsageViewer: "UsageViewer",
    EnvVariableManager: "EnvVariableManager",
    EnvironmentManager: "EnvironmentManager",
    V0Builder: "V0Builder",
    V0Chatter: "V0Chatter",
    V0Viewer: "V0Viewer",
};
/** @internal */
export const InviteUserToTeamRole$inboundSchema = z.nativeEnum(InviteUserToTeamRole);
/** @internal */
export const InviteUserToTeamRole$outboundSchema = InviteUserToTeamRole$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamRole$;
(function (InviteUserToTeamRole$) {
    /** @deprecated use `InviteUserToTeamRole$inboundSchema` instead. */
    InviteUserToTeamRole$.inboundSchema = InviteUserToTeamRole$inboundSchema;
    /** @deprecated use `InviteUserToTeamRole$outboundSchema` instead. */
    InviteUserToTeamRole$.outboundSchema = InviteUserToTeamRole$outboundSchema;
})(InviteUserToTeamRole$ || (InviteUserToTeamRole$ = {}));
/** @internal */
export const InviteUserToTeamTeamsRole$inboundSchema = z.nativeEnum(InviteUserToTeamTeamsRole);
/** @internal */
export const InviteUserToTeamTeamsRole$outboundSchema = InviteUserToTeamTeamsRole$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamTeamsRole$;
(function (InviteUserToTeamTeamsRole$) {
    /** @deprecated use `InviteUserToTeamTeamsRole$inboundSchema` instead. */
    InviteUserToTeamTeamsRole$.inboundSchema = InviteUserToTeamTeamsRole$inboundSchema;
    /** @deprecated use `InviteUserToTeamTeamsRole$outboundSchema` instead. */
    InviteUserToTeamTeamsRole$.outboundSchema = InviteUserToTeamTeamsRole$outboundSchema;
})(InviteUserToTeamTeamsRole$ || (InviteUserToTeamTeamsRole$ = {}));
/** @internal */
export const InviteUserToTeamProjects$inboundSchema = z.object({
    projectId: z.string(),
    role: InviteUserToTeamTeamsRole$inboundSchema,
});
/** @internal */
export const InviteUserToTeamProjects$outboundSchema = z.object({
    projectId: z.string(),
    role: InviteUserToTeamTeamsRole$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamProjects$;
(function (InviteUserToTeamProjects$) {
    /** @deprecated use `InviteUserToTeamProjects$inboundSchema` instead. */
    InviteUserToTeamProjects$.inboundSchema = InviteUserToTeamProjects$inboundSchema;
    /** @deprecated use `InviteUserToTeamProjects$outboundSchema` instead. */
    InviteUserToTeamProjects$.outboundSchema = InviteUserToTeamProjects$outboundSchema;
})(InviteUserToTeamProjects$ || (InviteUserToTeamProjects$ = {}));
export function inviteUserToTeamProjectsToJSON(inviteUserToTeamProjects) {
    return JSON.stringify(InviteUserToTeamProjects$outboundSchema.parse(inviteUserToTeamProjects));
}
export function inviteUserToTeamProjectsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InviteUserToTeamProjects$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InviteUserToTeamProjects' from JSON`);
}
/** @internal */
export const InviteUserToTeamRequestBody$inboundSchema = z.object({
    email: z.string(),
    role: InviteUserToTeamRole$inboundSchema.default("MEMBER"),
    projects: z.array(z.lazy(() => InviteUserToTeamProjects$inboundSchema))
        .optional(),
});
/** @internal */
export const InviteUserToTeamRequestBody$outboundSchema = z.object({
    email: z.string(),
    role: InviteUserToTeamRole$outboundSchema.default("MEMBER"),
    projects: z.array(z.lazy(() => InviteUserToTeamProjects$outboundSchema))
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamRequestBody$;
(function (InviteUserToTeamRequestBody$) {
    /** @deprecated use `InviteUserToTeamRequestBody$inboundSchema` instead. */
    InviteUserToTeamRequestBody$.inboundSchema = InviteUserToTeamRequestBody$inboundSchema;
    /** @deprecated use `InviteUserToTeamRequestBody$outboundSchema` instead. */
    InviteUserToTeamRequestBody$.outboundSchema = InviteUserToTeamRequestBody$outboundSchema;
})(InviteUserToTeamRequestBody$ || (InviteUserToTeamRequestBody$ = {}));
export function inviteUserToTeamRequestBodyToJSON(inviteUserToTeamRequestBody) {
    return JSON.stringify(InviteUserToTeamRequestBody$outboundSchema.parse(inviteUserToTeamRequestBody));
}
export function inviteUserToTeamRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InviteUserToTeamRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InviteUserToTeamRequestBody' from JSON`);
}
/** @internal */
export const InviteUserToTeamRequest$inboundSchema = z.object({
    teamId: z.string(),
    RequestBody: z.lazy(() => InviteUserToTeamRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const InviteUserToTeamRequest$outboundSchema = z.object({
    teamId: z.string(),
    requestBody: z.lazy(() => InviteUserToTeamRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamRequest$;
(function (InviteUserToTeamRequest$) {
    /** @deprecated use `InviteUserToTeamRequest$inboundSchema` instead. */
    InviteUserToTeamRequest$.inboundSchema = InviteUserToTeamRequest$inboundSchema;
    /** @deprecated use `InviteUserToTeamRequest$outboundSchema` instead. */
    InviteUserToTeamRequest$.outboundSchema = InviteUserToTeamRequest$outboundSchema;
})(InviteUserToTeamRequest$ || (InviteUserToTeamRequest$ = {}));
export function inviteUserToTeamRequestToJSON(inviteUserToTeamRequest) {
    return JSON.stringify(InviteUserToTeamRequest$outboundSchema.parse(inviteUserToTeamRequest));
}
export function inviteUserToTeamRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InviteUserToTeamRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InviteUserToTeamRequest' from JSON`);
}
/** @internal */
export const InviteUserToTeamTeamsResponseRole$inboundSchema = z.nativeEnum(InviteUserToTeamTeamsResponseRole);
/** @internal */
export const InviteUserToTeamTeamsResponseRole$outboundSchema = InviteUserToTeamTeamsResponseRole$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamTeamsResponseRole$;
(function (InviteUserToTeamTeamsResponseRole$) {
    /** @deprecated use `InviteUserToTeamTeamsResponseRole$inboundSchema` instead. */
    InviteUserToTeamTeamsResponseRole$.inboundSchema = InviteUserToTeamTeamsResponseRole$inboundSchema;
    /** @deprecated use `InviteUserToTeamTeamsResponseRole$outboundSchema` instead. */
    InviteUserToTeamTeamsResponseRole$.outboundSchema = InviteUserToTeamTeamsResponseRole$outboundSchema;
})(InviteUserToTeamTeamsResponseRole$ || (InviteUserToTeamTeamsResponseRole$ = {}));
/** @internal */
export const InviteUserToTeamTeamRoles$inboundSchema = z.nativeEnum(InviteUserToTeamTeamRoles);
/** @internal */
export const InviteUserToTeamTeamRoles$outboundSchema = InviteUserToTeamTeamRoles$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamTeamRoles$;
(function (InviteUserToTeamTeamRoles$) {
    /** @deprecated use `InviteUserToTeamTeamRoles$inboundSchema` instead. */
    InviteUserToTeamTeamRoles$.inboundSchema = InviteUserToTeamTeamRoles$inboundSchema;
    /** @deprecated use `InviteUserToTeamTeamRoles$outboundSchema` instead. */
    InviteUserToTeamTeamRoles$.outboundSchema = InviteUserToTeamTeamRoles$outboundSchema;
})(InviteUserToTeamTeamRoles$ || (InviteUserToTeamTeamRoles$ = {}));
/** @internal */
export const InviteUserToTeamTeamPermissions$inboundSchema = z.nativeEnum(InviteUserToTeamTeamPermissions);
/** @internal */
export const InviteUserToTeamTeamPermissions$outboundSchema = InviteUserToTeamTeamPermissions$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamTeamPermissions$;
(function (InviteUserToTeamTeamPermissions$) {
    /** @deprecated use `InviteUserToTeamTeamPermissions$inboundSchema` instead. */
    InviteUserToTeamTeamPermissions$.inboundSchema = InviteUserToTeamTeamPermissions$inboundSchema;
    /** @deprecated use `InviteUserToTeamTeamPermissions$outboundSchema` instead. */
    InviteUserToTeamTeamPermissions$.outboundSchema = InviteUserToTeamTeamPermissions$outboundSchema;
})(InviteUserToTeamTeamPermissions$ || (InviteUserToTeamTeamPermissions$ = {}));
/** @internal */
export const InviteUserToTeamResponseBody$inboundSchema = z.object({
    uid: z.string(),
    username: z.string(),
    email: z.string(),
    role: InviteUserToTeamTeamsResponseRole$inboundSchema,
    teamRoles: z.array(InviteUserToTeamTeamRoles$inboundSchema).optional(),
    teamPermissions: z.array(InviteUserToTeamTeamPermissions$inboundSchema)
        .optional(),
});
/** @internal */
export const InviteUserToTeamResponseBody$outboundSchema = z.object({
    uid: z.string(),
    username: z.string(),
    email: z.string(),
    role: InviteUserToTeamTeamsResponseRole$outboundSchema,
    teamRoles: z.array(InviteUserToTeamTeamRoles$outboundSchema).optional(),
    teamPermissions: z.array(InviteUserToTeamTeamPermissions$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InviteUserToTeamResponseBody$;
(function (InviteUserToTeamResponseBody$) {
    /** @deprecated use `InviteUserToTeamResponseBody$inboundSchema` instead. */
    InviteUserToTeamResponseBody$.inboundSchema = InviteUserToTeamResponseBody$inboundSchema;
    /** @deprecated use `InviteUserToTeamResponseBody$outboundSchema` instead. */
    InviteUserToTeamResponseBody$.outboundSchema = InviteUserToTeamResponseBody$outboundSchema;
})(InviteUserToTeamResponseBody$ || (InviteUserToTeamResponseBody$ = {}));
export function inviteUserToTeamResponseBodyToJSON(inviteUserToTeamResponseBody) {
    return JSON.stringify(InviteUserToTeamResponseBody$outboundSchema.parse(inviteUserToTeamResponseBody));
}
export function inviteUserToTeamResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InviteUserToTeamResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InviteUserToTeamResponseBody' from JSON`);
}
