// @vercel/sdk/esm/models/removeprojectmemberop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const RemoveProjectMemberRequest$inboundSchema = z.object({
    idOrName: z.string(),
    uid: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const RemoveProjectMemberRequest$outboundSchema = z.object({
    idOrName: z.string(),
    uid: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveProjectMemberRequest$;
(function (RemoveProjectMemberRequest$) {
    /** @deprecated use `RemoveProjectMemberRequest$inboundSchema` instead. */
    RemoveProjectMemberRequest$.inboundSchema = RemoveProjectMemberRequest$inboundSchema;
    /** @deprecated use `RemoveProjectMemberRequest$outboundSchema` instead. */
    RemoveProjectMemberRequest$.outboundSchema = RemoveProjectMemberRequest$outboundSchema;
})(RemoveProjectMemberRequest$ || (RemoveProjectMemberRequest$ = {}));
export function removeProjectMemberRequestToJSON(removeProjectMemberRequest) {
    return JSON.stringify(RemoveProjectMemberRequest$outboundSchema.parse(removeProjectMemberRequest));
}
export function removeProjectMemberRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectMemberRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectMemberRequest' from JSON`);
}
/** @internal */
export const RemoveProjectMemberResponseBody$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const RemoveProjectMemberResponseBody$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveProjectMemberResponseBody$;
(function (RemoveProjectMemberResponseBody$) {
    /** @deprecated use `RemoveProjectMemberResponseBody$inboundSchema` instead. */
    RemoveProjectMemberResponseBody$.inboundSchema = RemoveProjectMemberResponseBody$inboundSchema;
    /** @deprecated use `RemoveProjectMemberResponseBody$outboundSchema` instead. */
    RemoveProjectMemberResponseBody$.outboundSchema = RemoveProjectMemberResponseBody$outboundSchema;
})(RemoveProjectMemberResponseBody$ || (RemoveProjectMemberResponseBody$ = {}));
export function removeProjectMemberResponseBodyToJSON(removeProjectMemberResponseBody) {
    return JSON.stringify(RemoveProjectMemberResponseBody$outboundSchema.parse(removeProjectMemberResponseBody));
}
export function removeProjectMemberResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectMemberResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectMemberResponseBody' from JSON`);
}
