// @vercel/sdk/esm/models/removeteammemberop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const RemoveTeamMemberRequest$inboundSchema = z.object({
    uid: z.string(),
    newDefaultTeamId: z.string().optional(),
    teamId: z.string(),
});
/** @internal */
export const RemoveTeamMemberRequest$outboundSchema = z.object({
    uid: z.string(),
    newDefaultTeamId: z.string().optional(),
    teamId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveTeamMemberRequest$;
(function (RemoveTeamMemberRequest$) {
    /** @deprecated use `RemoveTeamMemberRequest$inboundSchema` instead. */
    RemoveTeamMemberRequest$.inboundSchema = RemoveTeamMemberRequest$inboundSchema;
    /** @deprecated use `RemoveTeamMemberRequest$outboundSchema` instead. */
    RemoveTeamMemberRequest$.outboundSchema = RemoveTeamMemberRequest$outboundSchema;
})(RemoveTeamMemberRequest$ || (RemoveTeamMemberRequest$ = {}));
export function removeTeamMemberRequestToJSON(removeTeamMemberRequest) {
    return JSON.stringify(RemoveTeamMemberRequest$outboundSchema.parse(removeTeamMemberRequest));
}
export function removeTeamMemberRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveTeamMemberRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveTeamMemberRequest' from JSON`);
}
/** @internal */
export const RemoveTeamMemberResponseBody$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const RemoveTeamMemberResponseBody$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RemoveTeamMemberResponseBody$;
(function (RemoveTeamMemberResponseBody$) {
    /** @deprecated use `RemoveTeamMemberResponseBody$inboundSchema` instead. */
    RemoveTeamMemberResponseBody$.inboundSchema = RemoveTeamMemberResponseBody$inboundSchema;
    /** @deprecated use `RemoveTeamMemberResponseBody$outboundSchema` instead. */
    RemoveTeamMemberResponseBody$.outboundSchema = RemoveTeamMemberResponseBody$outboundSchema;
})(RemoveTeamMemberResponseBody$ || (RemoveTeamMemberResponseBody$ = {}));
export function removeTeamMemberResponseBodyToJSON(removeTeamMemberResponseBody) {
    return JSON.stringify(RemoveTeamMemberResponseBody$outboundSchema.parse(removeTeamMemberResponseBody));
}
export function removeTeamMemberResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveTeamMemberResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveTeamMemberResponseBody' from JSON`);
}
