// @vercel/sdk/esm/models/updatecustomenvironmentop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * Type of matcher. One of \"equals\", \"startsWith\", or \"endsWith\".
 */
export const UpdateCustomEnvironmentType = {
    Equals: "equals",
    StartsWith: "startsWith",
    EndsWith: "endsWith",
};
/**
 * The type of environment (production, preview, or development)
 */
export const UpdateCustomEnvironmentEnvironmentType = {
    Preview: "preview",
    Production: "production",
    Development: "development",
};
/**
 * The type of matching to perform
 */
export const UpdateCustomEnvironmentEnvironmentResponseType = {
    EndsWith: "endsWith",
    StartsWith: "startsWith",
    Equals: "equals",
};
/** @internal */
export const UpdateCustomEnvironmentType$inboundSchema = z.nativeEnum(UpdateCustomEnvironmentType);
/** @internal */
export const UpdateCustomEnvironmentType$outboundSchema = UpdateCustomEnvironmentType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentType$;
(function (UpdateCustomEnvironmentType$) {
    /** @deprecated use `UpdateCustomEnvironmentType$inboundSchema` instead. */
    UpdateCustomEnvironmentType$.inboundSchema = UpdateCustomEnvironmentType$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentType$outboundSchema` instead. */
    UpdateCustomEnvironmentType$.outboundSchema = UpdateCustomEnvironmentType$outboundSchema;
})(UpdateCustomEnvironmentType$ || (UpdateCustomEnvironmentType$ = {}));
/** @internal */
export const UpdateCustomEnvironmentBranchMatcher$inboundSchema = z.object({
    type: UpdateCustomEnvironmentType$inboundSchema,
    pattern: z.string(),
});
/** @internal */
export const UpdateCustomEnvironmentBranchMatcher$outboundSchema = z.object({
    type: UpdateCustomEnvironmentType$outboundSchema,
    pattern: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentBranchMatcher$;
(function (UpdateCustomEnvironmentBranchMatcher$) {
    /** @deprecated use `UpdateCustomEnvironmentBranchMatcher$inboundSchema` instead. */
    UpdateCustomEnvironmentBranchMatcher$.inboundSchema = UpdateCustomEnvironmentBranchMatcher$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentBranchMatcher$outboundSchema` instead. */
    UpdateCustomEnvironmentBranchMatcher$.outboundSchema = UpdateCustomEnvironmentBranchMatcher$outboundSchema;
})(UpdateCustomEnvironmentBranchMatcher$ || (UpdateCustomEnvironmentBranchMatcher$ = {}));
export function updateCustomEnvironmentBranchMatcherToJSON(updateCustomEnvironmentBranchMatcher) {
    return JSON.stringify(UpdateCustomEnvironmentBranchMatcher$outboundSchema.parse(updateCustomEnvironmentBranchMatcher));
}
export function updateCustomEnvironmentBranchMatcherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateCustomEnvironmentBranchMatcher$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateCustomEnvironmentBranchMatcher' from JSON`);
}
/** @internal */
export const UpdateCustomEnvironmentRequestBody$inboundSchema = z.object({
    slug: z.string().optional(),
    description: z.string().optional(),
    branchMatcher: z.nullable(z.lazy(() => UpdateCustomEnvironmentBranchMatcher$inboundSchema)).optional(),
});
/** @internal */
export const UpdateCustomEnvironmentRequestBody$outboundSchema = z.object({
    slug: z.string().optional(),
    description: z.string().optional(),
    branchMatcher: z.nullable(z.lazy(() => UpdateCustomEnvironmentBranchMatcher$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentRequestBody$;
(function (UpdateCustomEnvironmentRequestBody$) {
    /** @deprecated use `UpdateCustomEnvironmentRequestBody$inboundSchema` instead. */
    UpdateCustomEnvironmentRequestBody$.inboundSchema = UpdateCustomEnvironmentRequestBody$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentRequestBody$outboundSchema` instead. */
    UpdateCustomEnvironmentRequestBody$.outboundSchema = UpdateCustomEnvironmentRequestBody$outboundSchema;
})(UpdateCustomEnvironmentRequestBody$ || (UpdateCustomEnvironmentRequestBody$ = {}));
export function updateCustomEnvironmentRequestBodyToJSON(updateCustomEnvironmentRequestBody) {
    return JSON.stringify(UpdateCustomEnvironmentRequestBody$outboundSchema.parse(updateCustomEnvironmentRequestBody));
}
export function updateCustomEnvironmentRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateCustomEnvironmentRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateCustomEnvironmentRequestBody' from JSON`);
}
/** @internal */
export const UpdateCustomEnvironmentRequest$inboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateCustomEnvironmentRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateCustomEnvironmentRequest$outboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateCustomEnvironmentRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentRequest$;
(function (UpdateCustomEnvironmentRequest$) {
    /** @deprecated use `UpdateCustomEnvironmentRequest$inboundSchema` instead. */
    UpdateCustomEnvironmentRequest$.inboundSchema = UpdateCustomEnvironmentRequest$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentRequest$outboundSchema` instead. */
    UpdateCustomEnvironmentRequest$.outboundSchema = UpdateCustomEnvironmentRequest$outboundSchema;
})(UpdateCustomEnvironmentRequest$ || (UpdateCustomEnvironmentRequest$ = {}));
export function updateCustomEnvironmentRequestToJSON(updateCustomEnvironmentRequest) {
    return JSON.stringify(UpdateCustomEnvironmentRequest$outboundSchema.parse(updateCustomEnvironmentRequest));
}
export function updateCustomEnvironmentRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateCustomEnvironmentRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateCustomEnvironmentRequest' from JSON`);
}
/** @internal */
export const UpdateCustomEnvironmentEnvironmentType$inboundSchema = z.nativeEnum(UpdateCustomEnvironmentEnvironmentType);
/** @internal */
export const UpdateCustomEnvironmentEnvironmentType$outboundSchema = UpdateCustomEnvironmentEnvironmentType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentEnvironmentType$;
(function (UpdateCustomEnvironmentEnvironmentType$) {
    /** @deprecated use `UpdateCustomEnvironmentEnvironmentType$inboundSchema` instead. */
    UpdateCustomEnvironmentEnvironmentType$.inboundSchema = UpdateCustomEnvironmentEnvironmentType$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentEnvironmentType$outboundSchema` instead. */
    UpdateCustomEnvironmentEnvironmentType$.outboundSchema = UpdateCustomEnvironmentEnvironmentType$outboundSchema;
})(UpdateCustomEnvironmentEnvironmentType$ || (UpdateCustomEnvironmentEnvironmentType$ = {}));
/** @internal */
export const UpdateCustomEnvironmentEnvironmentResponseType$inboundSchema = z
    .nativeEnum(UpdateCustomEnvironmentEnvironmentResponseType);
/** @internal */
export const UpdateCustomEnvironmentEnvironmentResponseType$outboundSchema = UpdateCustomEnvironmentEnvironmentResponseType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentEnvironmentResponseType$;
(function (UpdateCustomEnvironmentEnvironmentResponseType$) {
    /** @deprecated use `UpdateCustomEnvironmentEnvironmentResponseType$inboundSchema` instead. */
    UpdateCustomEnvironmentEnvironmentResponseType$.inboundSchema = UpdateCustomEnvironmentEnvironmentResponseType$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentEnvironmentResponseType$outboundSchema` instead. */
    UpdateCustomEnvironmentEnvironmentResponseType$.outboundSchema = UpdateCustomEnvironmentEnvironmentResponseType$outboundSchema;
})(UpdateCustomEnvironmentEnvironmentResponseType$ || (UpdateCustomEnvironmentEnvironmentResponseType$ = {}));
/** @internal */
export const UpdateCustomEnvironmentEnvironmentBranchMatcher$inboundSchema = z.object({
    type: UpdateCustomEnvironmentEnvironmentResponseType$inboundSchema,
    pattern: z.string(),
});
/** @internal */
export const UpdateCustomEnvironmentEnvironmentBranchMatcher$outboundSchema = z.object({
    type: UpdateCustomEnvironmentEnvironmentResponseType$outboundSchema,
    pattern: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentEnvironmentBranchMatcher$;
(function (UpdateCustomEnvironmentEnvironmentBranchMatcher$) {
    /** @deprecated use `UpdateCustomEnvironmentEnvironmentBranchMatcher$inboundSchema` instead. */
    UpdateCustomEnvironmentEnvironmentBranchMatcher$.inboundSchema = UpdateCustomEnvironmentEnvironmentBranchMatcher$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentEnvironmentBranchMatcher$outboundSchema` instead. */
    UpdateCustomEnvironmentEnvironmentBranchMatcher$.outboundSchema = UpdateCustomEnvironmentEnvironmentBranchMatcher$outboundSchema;
})(UpdateCustomEnvironmentEnvironmentBranchMatcher$ || (UpdateCustomEnvironmentEnvironmentBranchMatcher$ = {}));
export function updateCustomEnvironmentEnvironmentBranchMatcherToJSON(updateCustomEnvironmentEnvironmentBranchMatcher) {
    return JSON.stringify(UpdateCustomEnvironmentEnvironmentBranchMatcher$outboundSchema.parse(updateCustomEnvironmentEnvironmentBranchMatcher));
}
export function updateCustomEnvironmentEnvironmentBranchMatcherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateCustomEnvironmentEnvironmentBranchMatcher$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateCustomEnvironmentEnvironmentBranchMatcher' from JSON`);
}
/** @internal */
export const UpdateCustomEnvironmentVerification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const UpdateCustomEnvironmentVerification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentVerification$;
(function (UpdateCustomEnvironmentVerification$) {
    /** @deprecated use `UpdateCustomEnvironmentVerification$inboundSchema` instead. */
    UpdateCustomEnvironmentVerification$.inboundSchema = UpdateCustomEnvironmentVerification$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentVerification$outboundSchema` instead. */
    UpdateCustomEnvironmentVerification$.outboundSchema = UpdateCustomEnvironmentVerification$outboundSchema;
})(UpdateCustomEnvironmentVerification$ || (UpdateCustomEnvironmentVerification$ = {}));
export function updateCustomEnvironmentVerificationToJSON(updateCustomEnvironmentVerification) {
    return JSON.stringify(UpdateCustomEnvironmentVerification$outboundSchema.parse(updateCustomEnvironmentVerification));
}
export function updateCustomEnvironmentVerificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateCustomEnvironmentVerification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateCustomEnvironmentVerification' from JSON`);
}
/** @internal */
export const UpdateCustomEnvironmentDomains$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => UpdateCustomEnvironmentVerification$inboundSchema)).optional(),
});
/** @internal */
export const UpdateCustomEnvironmentDomains$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => UpdateCustomEnvironmentVerification$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentDomains$;
(function (UpdateCustomEnvironmentDomains$) {
    /** @deprecated use `UpdateCustomEnvironmentDomains$inboundSchema` instead. */
    UpdateCustomEnvironmentDomains$.inboundSchema = UpdateCustomEnvironmentDomains$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentDomains$outboundSchema` instead. */
    UpdateCustomEnvironmentDomains$.outboundSchema = UpdateCustomEnvironmentDomains$outboundSchema;
})(UpdateCustomEnvironmentDomains$ || (UpdateCustomEnvironmentDomains$ = {}));
export function updateCustomEnvironmentDomainsToJSON(updateCustomEnvironmentDomains) {
    return JSON.stringify(UpdateCustomEnvironmentDomains$outboundSchema.parse(updateCustomEnvironmentDomains));
}
export function updateCustomEnvironmentDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateCustomEnvironmentDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateCustomEnvironmentDomains' from JSON`);
}
/** @internal */
export const UpdateCustomEnvironmentResponseBody$inboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: UpdateCustomEnvironmentEnvironmentType$inboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => UpdateCustomEnvironmentEnvironmentBranchMatcher$inboundSchema).optional(),
    domains: z.array(z.lazy(() => UpdateCustomEnvironmentDomains$inboundSchema))
        .optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const UpdateCustomEnvironmentResponseBody$outboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: UpdateCustomEnvironmentEnvironmentType$outboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => UpdateCustomEnvironmentEnvironmentBranchMatcher$outboundSchema).optional(),
    domains: z.array(z.lazy(() => UpdateCustomEnvironmentDomains$outboundSchema))
        .optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateCustomEnvironmentResponseBody$;
(function (UpdateCustomEnvironmentResponseBody$) {
    /** @deprecated use `UpdateCustomEnvironmentResponseBody$inboundSchema` instead. */
    UpdateCustomEnvironmentResponseBody$.inboundSchema = UpdateCustomEnvironmentResponseBody$inboundSchema;
    /** @deprecated use `UpdateCustomEnvironmentResponseBody$outboundSchema` instead. */
    UpdateCustomEnvironmentResponseBody$.outboundSchema = UpdateCustomEnvironmentResponseBody$outboundSchema;
})(UpdateCustomEnvironmentResponseBody$ || (UpdateCustomEnvironmentResponseBody$ = {}));
export function updateCustomEnvironmentResponseBodyToJSON(updateCustomEnvironmentResponseBody) {
    return JSON.stringify(UpdateCustomEnvironmentResponseBody$outboundSchema.parse(updateCustomEnvironmentResponseBody));
}
export function updateCustomEnvironmentResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateCustomEnvironmentResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateCustomEnvironmentResponseBody' from JSON`);
}
