// @vercel/sdk/src/models/deleteaccessgroupop.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { Result as SafeParseResult } from "../types/fp.js";
import { SDKValidationError } from "./sdkvalidationerror.js";

export type DeleteAccessGroupRequest = {
  idOrName: string;
  /**
   * The Team identifier to perform the request on behalf of.
   */
  teamId?: string | undefined;
  /**
   * The Team slug to perform the request on behalf of.
   */
  slug?: string | undefined;
};

/** @internal */
export const DeleteAccessGroupRequest$inboundSchema: z.ZodType<
  DeleteAccessGroupRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  idOrName: z.string(),
  teamId: z.string().optional(),
  slug: z.string().optional(),
});

/** @internal */
export type DeleteAccessGroupRequest$Outbound = {
  idOrName: string;
  teamId?: string | undefined;
  slug?: string | undefined;
};

/** @internal */
export const DeleteAccessGroupRequest$outboundSchema: z.ZodType<
  DeleteAccessGroupRequest$Outbound,
  z.ZodTypeDef,
  DeleteAccessGroupRequest
> = z.object({
  idOrName: z.string(),
  teamId: z.string().optional(),
  slug: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace DeleteAccessGroupRequest$ {
  /** @deprecated use `DeleteAccessGroupRequest$inboundSchema` instead. */
  export const inboundSchema = DeleteAccessGroupRequest$inboundSchema;
  /** @deprecated use `DeleteAccessGroupRequest$outboundSchema` instead. */
  export const outboundSchema = DeleteAccessGroupRequest$outboundSchema;
  /** @deprecated use `DeleteAccessGroupRequest$Outbound` instead. */
  export type Outbound = DeleteAccessGroupRequest$Outbound;
}

export function deleteAccessGroupRequestToJSON(
  deleteAccessGroupRequest: DeleteAccessGroupRequest,
): string {
  return JSON.stringify(
    DeleteAccessGroupRequest$outboundSchema.parse(deleteAccessGroupRequest),
  );
}

export function deleteAccessGroupRequestFromJSON(
  jsonString: string,
): SafeParseResult<DeleteAccessGroupRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => DeleteAccessGroupRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'DeleteAccessGroupRequest' from JSON`,
  );
}
