// @vercel/sdk/src/models/deleteconfigurationop.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { Result as SafeParseResult } from "../types/fp.js";
import { SDKValidationError } from "./sdkvalidationerror.js";

export type DeleteConfigurationRequest = {
  id: string;
  /**
   * The Team identifier to perform the request on behalf of.
   */
  teamId?: string | undefined;
  /**
   * The Team slug to perform the request on behalf of.
   */
  slug?: string | undefined;
};

/** @internal */
export const DeleteConfigurationRequest$inboundSchema: z.ZodType<
  DeleteConfigurationRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  id: z.string(),
  teamId: z.string().optional(),
  slug: z.string().optional(),
});

/** @internal */
export type DeleteConfigurationRequest$Outbound = {
  id: string;
  teamId?: string | undefined;
  slug?: string | undefined;
};

/** @internal */
export const DeleteConfigurationRequest$outboundSchema: z.ZodType<
  DeleteConfigurationRequest$Outbound,
  z.ZodTypeDef,
  DeleteConfigurationRequest
> = z.object({
  id: z.string(),
  teamId: z.string().optional(),
  slug: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace DeleteConfigurationRequest$ {
  /** @deprecated use `DeleteConfigurationRequest$inboundSchema` instead. */
  export const inboundSchema = DeleteConfigurationRequest$inboundSchema;
  /** @deprecated use `DeleteConfigurationRequest$outboundSchema` instead. */
  export const outboundSchema = DeleteConfigurationRequest$outboundSchema;
  /** @deprecated use `DeleteConfigurationRequest$Outbound` instead. */
  export type Outbound = DeleteConfigurationRequest$Outbound;
}

export function deleteConfigurationRequestToJSON(
  deleteConfigurationRequest: DeleteConfigurationRequest,
): string {
  return JSON.stringify(
    DeleteConfigurationRequest$outboundSchema.parse(deleteConfigurationRequest),
  );
}

export function deleteConfigurationRequestFromJSON(
  jsonString: string,
): SafeParseResult<DeleteConfigurationRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => DeleteConfigurationRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'DeleteConfigurationRequest' from JSON`,
  );
}
