// @vercel/sdk/src/models/removeprojectdomainop.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { Result as SafeParseResult } from "../types/fp.js";
import { SDKValidationError } from "./sdkvalidationerror.js";

export type RemoveProjectDomainRequestBody = {
  /**
   * Whether to remove all domains from this project that redirect to the domain being removed.
   */
  removeRedirects?: boolean | undefined;
};

export type RemoveProjectDomainRequest = {
  /**
   * The unique project identifier or the project name
   */
  idOrName: string;
  /**
   * The project domain name
   */
  domain: string;
  /**
   * The Team identifier to perform the request on behalf of.
   */
  teamId?: string | undefined;
  /**
   * The Team slug to perform the request on behalf of.
   */
  slug?: string | undefined;
  requestBody?: RemoveProjectDomainRequestBody | undefined;
};

/**
 * The domain was succesfully removed from the project
 */
export type RemoveProjectDomainResponseBody = {};

/** @internal */
export const RemoveProjectDomainRequestBody$inboundSchema: z.ZodType<
  RemoveProjectDomainRequestBody,
  z.ZodTypeDef,
  unknown
> = z.object({
  removeRedirects: z.boolean().optional(),
});

/** @internal */
export type RemoveProjectDomainRequestBody$Outbound = {
  removeRedirects?: boolean | undefined;
};

/** @internal */
export const RemoveProjectDomainRequestBody$outboundSchema: z.ZodType<
  RemoveProjectDomainRequestBody$Outbound,
  z.ZodTypeDef,
  RemoveProjectDomainRequestBody
> = z.object({
  removeRedirects: z.boolean().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RemoveProjectDomainRequestBody$ {
  /** @deprecated use `RemoveProjectDomainRequestBody$inboundSchema` instead. */
  export const inboundSchema = RemoveProjectDomainRequestBody$inboundSchema;
  /** @deprecated use `RemoveProjectDomainRequestBody$outboundSchema` instead. */
  export const outboundSchema = RemoveProjectDomainRequestBody$outboundSchema;
  /** @deprecated use `RemoveProjectDomainRequestBody$Outbound` instead. */
  export type Outbound = RemoveProjectDomainRequestBody$Outbound;
}

export function removeProjectDomainRequestBodyToJSON(
  removeProjectDomainRequestBody: RemoveProjectDomainRequestBody,
): string {
  return JSON.stringify(
    RemoveProjectDomainRequestBody$outboundSchema.parse(
      removeProjectDomainRequestBody,
    ),
  );
}

export function removeProjectDomainRequestBodyFromJSON(
  jsonString: string,
): SafeParseResult<RemoveProjectDomainRequestBody, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => RemoveProjectDomainRequestBody$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'RemoveProjectDomainRequestBody' from JSON`,
  );
}

/** @internal */
export const RemoveProjectDomainRequest$inboundSchema: z.ZodType<
  RemoveProjectDomainRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  idOrName: z.string(),
  domain: z.string(),
  teamId: z.string().optional(),
  slug: z.string().optional(),
  RequestBody: z.lazy(() => RemoveProjectDomainRequestBody$inboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "RequestBody": "requestBody",
  });
});

/** @internal */
export type RemoveProjectDomainRequest$Outbound = {
  idOrName: string;
  domain: string;
  teamId?: string | undefined;
  slug?: string | undefined;
  RequestBody?: RemoveProjectDomainRequestBody$Outbound | undefined;
};

/** @internal */
export const RemoveProjectDomainRequest$outboundSchema: z.ZodType<
  RemoveProjectDomainRequest$Outbound,
  z.ZodTypeDef,
  RemoveProjectDomainRequest
> = z.object({
  idOrName: z.string(),
  domain: z.string(),
  teamId: z.string().optional(),
  slug: z.string().optional(),
  requestBody: z.lazy(() => RemoveProjectDomainRequestBody$outboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    requestBody: "RequestBody",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RemoveProjectDomainRequest$ {
  /** @deprecated use `RemoveProjectDomainRequest$inboundSchema` instead. */
  export const inboundSchema = RemoveProjectDomainRequest$inboundSchema;
  /** @deprecated use `RemoveProjectDomainRequest$outboundSchema` instead. */
  export const outboundSchema = RemoveProjectDomainRequest$outboundSchema;
  /** @deprecated use `RemoveProjectDomainRequest$Outbound` instead. */
  export type Outbound = RemoveProjectDomainRequest$Outbound;
}

export function removeProjectDomainRequestToJSON(
  removeProjectDomainRequest: RemoveProjectDomainRequest,
): string {
  return JSON.stringify(
    RemoveProjectDomainRequest$outboundSchema.parse(removeProjectDomainRequest),
  );
}

export function removeProjectDomainRequestFromJSON(
  jsonString: string,
): SafeParseResult<RemoveProjectDomainRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => RemoveProjectDomainRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'RemoveProjectDomainRequest' from JSON`,
  );
}

/** @internal */
export const RemoveProjectDomainResponseBody$inboundSchema: z.ZodType<
  RemoveProjectDomainResponseBody,
  z.ZodTypeDef,
  unknown
> = z.object({});

/** @internal */
export type RemoveProjectDomainResponseBody$Outbound = {};

/** @internal */
export const RemoveProjectDomainResponseBody$outboundSchema: z.ZodType<
  RemoveProjectDomainResponseBody$Outbound,
  z.ZodTypeDef,
  RemoveProjectDomainResponseBody
> = z.object({});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RemoveProjectDomainResponseBody$ {
  /** @deprecated use `RemoveProjectDomainResponseBody$inboundSchema` instead. */
  export const inboundSchema = RemoveProjectDomainResponseBody$inboundSchema;
  /** @deprecated use `RemoveProjectDomainResponseBody$outboundSchema` instead. */
  export const outboundSchema = RemoveProjectDomainResponseBody$outboundSchema;
  /** @deprecated use `RemoveProjectDomainResponseBody$Outbound` instead. */
  export type Outbound = RemoveProjectDomainResponseBody$Outbound;
}

export function removeProjectDomainResponseBodyToJSON(
  removeProjectDomainResponseBody: RemoveProjectDomainResponseBody,
): string {
  return JSON.stringify(
    RemoveProjectDomainResponseBody$outboundSchema.parse(
      removeProjectDomainResponseBody,
    ),
  );
}

export function removeProjectDomainResponseBodyFromJSON(
  jsonString: string,
): SafeParseResult<RemoveProjectDomainResponseBody, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => RemoveProjectDomainResponseBody$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'RemoveProjectDomainResponseBody' from JSON`,
  );
}
