// @vercel/sdk/src/models/vercelnotfounderror.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { Result as SafeParseResult } from "../types/fp.js";
import { SDKValidationError } from "./sdkvalidationerror.js";
import { VercelError } from "./vercelerror.js";

export type VercelNotFoundErrorError = {
  code: string;
  message: string;
};

export type VercelNotFoundErrorData = {
  error: VercelNotFoundErrorError;
};

export class VercelNotFoundError extends VercelError {
  error: VercelNotFoundErrorError;

  /** The original data that was passed to this error instance. */
  data$: VercelNotFoundErrorData;

  constructor(
    err: VercelNotFoundErrorData,
    httpMeta: { response: Response; request: Request; body: string },
  ) {
    const message = err.error?.message
      || `API error occurred: ${JSON.stringify(err)}`;
    super(message, httpMeta);
    this.data$ = err;
    this.error = err.error;

    this.name = "VercelNotFoundError";
  }
}

/** @internal */
export const VercelNotFoundErrorError$inboundSchema: z.ZodType<
  VercelNotFoundErrorError,
  z.ZodTypeDef,
  unknown
> = z.object({
  code: z.string(),
  message: z.string(),
});

/** @internal */
export type VercelNotFoundErrorError$Outbound = {
  code: string;
  message: string;
};

/** @internal */
export const VercelNotFoundErrorError$outboundSchema: z.ZodType<
  VercelNotFoundErrorError$Outbound,
  z.ZodTypeDef,
  VercelNotFoundErrorError
> = z.object({
  code: z.string(),
  message: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace VercelNotFoundErrorError$ {
  /** @deprecated use `VercelNotFoundErrorError$inboundSchema` instead. */
  export const inboundSchema = VercelNotFoundErrorError$inboundSchema;
  /** @deprecated use `VercelNotFoundErrorError$outboundSchema` instead. */
  export const outboundSchema = VercelNotFoundErrorError$outboundSchema;
  /** @deprecated use `VercelNotFoundErrorError$Outbound` instead. */
  export type Outbound = VercelNotFoundErrorError$Outbound;
}

export function vercelNotFoundErrorErrorToJSON(
  vercelNotFoundErrorError: VercelNotFoundErrorError,
): string {
  return JSON.stringify(
    VercelNotFoundErrorError$outboundSchema.parse(vercelNotFoundErrorError),
  );
}

export function vercelNotFoundErrorErrorFromJSON(
  jsonString: string,
): SafeParseResult<VercelNotFoundErrorError, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => VercelNotFoundErrorError$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'VercelNotFoundErrorError' from JSON`,
  );
}

/** @internal */
export const VercelNotFoundError$inboundSchema: z.ZodType<
  VercelNotFoundError,
  z.ZodTypeDef,
  unknown
> = z.object({
  error: z.lazy(() => VercelNotFoundErrorError$inboundSchema),
  request$: z.instanceof(Request),
  response$: z.instanceof(Response),
  body$: z.string(),
})
  .transform((v) => {
    return new VercelNotFoundError(v, {
      request: v.request$,
      response: v.response$,
      body: v.body$,
    });
  });

/** @internal */
export type VercelNotFoundError$Outbound = {
  error: VercelNotFoundErrorError$Outbound;
};

/** @internal */
export const VercelNotFoundError$outboundSchema: z.ZodType<
  VercelNotFoundError$Outbound,
  z.ZodTypeDef,
  VercelNotFoundError
> = z.instanceof(VercelNotFoundError)
  .transform(v => v.data$)
  .pipe(z.object({
    error: z.lazy(() => VercelNotFoundErrorError$outboundSchema),
  }));

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace VercelNotFoundError$ {
  /** @deprecated use `VercelNotFoundError$inboundSchema` instead. */
  export const inboundSchema = VercelNotFoundError$inboundSchema;
  /** @deprecated use `VercelNotFoundError$outboundSchema` instead. */
  export const outboundSchema = VercelNotFoundError$outboundSchema;
  /** @deprecated use `VercelNotFoundError$Outbound` instead. */
  export type Outbound = VercelNotFoundError$Outbound;
}
