// @vercel/sdk/src/mcp-server/tools/accessGroupsReadAccessGroup.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { accessGroupsReadAccessGroup } from "../../funcs/accessGroupsReadAccessGroup.js";
import { ReadAccessGroupRequest$inboundSchema } from "../../models/readaccessgroupop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: ReadAccessGroupRequest$inboundSchema,
};

export const tool$accessGroupsReadAccessGroup: ToolDefinition<typeof args> = {
  name: "access-groups-read-access-group",
  description: `Reads an access group

Allows to read an access group`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await accessGroupsReadAccessGroup(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
