// @vercel/sdk/src/mcp-server/tools/edgeConfigGetEdgeConfigItems.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { edgeConfigGetEdgeConfigItems } from "../../funcs/edgeConfigGetEdgeConfigItems.js";
import { GetEdgeConfigItemsRequest$inboundSchema } from "../../models/getedgeconfigitemsop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetEdgeConfigItemsRequest$inboundSchema,
};

export const tool$edgeConfigGetEdgeConfigItems: ToolDefinition<typeof args> = {
  name: "edge-config-get-edge-config-items",
  description: `Get Edge Config items

Returns all items of an Edge Config.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await edgeConfigGetEdgeConfigItems(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
