// @vercel/sdk/src/mcp-server/tools/environmentCreateCustomEnvironment.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { environmentCreateCustomEnvironment } from "../../funcs/environmentCreateCustomEnvironment.js";
import { CreateCustomEnvironmentRequest$inboundSchema } from "../../models/createcustomenvironmentop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: CreateCustomEnvironmentRequest$inboundSchema,
};

export const tool$environmentCreateCustomEnvironment: ToolDefinition<
  typeof args
> = {
  name: "environment-create-custom-environment",
  description: `Create a custom environment for the current project.

Creates a custom environment for the current project. Cannot be named 'Production' or 'Preview'.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await environmentCreateCustomEnvironment(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
