// @vercel/sdk/src/mcp-server/tools/marketplaceGetMember.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { marketplaceGetMember } from "../../funcs/marketplaceGetMember.js";
import { GetMemberRequest$inboundSchema } from "../../models/getmemberop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetMemberRequest$inboundSchema,
};

export const tool$marketplaceGetMember: ToolDefinition<typeof args> = {
  name: "marketplace-get-member",
  description: `Get Member Information

Returns the member role and other information for a given member ID ("user_id" claim in the SSO OIDC token).`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await marketplaceGetMember(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
