// @vercel/sdk/src/mcp-server/tools/projectsGetProjectDomain.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectsGetProjectDomain } from "../../funcs/projectsGetProjectDomain.js";
import { GetProjectDomainRequest$inboundSchema } from "../../models/getprojectdomainop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetProjectDomainRequest$inboundSchema,
};

export const tool$projectsGetProjectDomain: ToolDefinition<typeof args> = {
  name: "projects-get-project-domain",
  description: `Get a project domain

Get project domain by project id/name and domain name.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectsGetProjectDomain(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
