// @vercel/sdk/src/mcp-server/tools/accessGroupsCreateAccessGroupProject.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { accessGroupsCreateAccessGroupProject } from "../../funcs/accessGroupsCreateAccessGroupProject.js";
import { CreateAccessGroupProjectRequest$inboundSchema } from "../../models/createaccessgroupprojectop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: CreateAccessGroupProjectRequest$inboundSchema,
};

export const tool$accessGroupsCreateAccessGroupProject: ToolDefinition<
  typeof args
> = {
  name: "access-groups-create-access-group-project",
  description: `Create an access group project

Allows creation of an access group project`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await accessGroupsCreateAccessGroupProject(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
