// @vercel/sdk/src/mcp-server/tools/accessGroupsReadAccessGroupProject.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { accessGroupsReadAccessGroupProject } from "../../funcs/accessGroupsReadAccessGroupProject.js";
import { ReadAccessGroupProjectRequest$inboundSchema } from "../../models/readaccessgroupprojectop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: ReadAccessGroupProjectRequest$inboundSchema,
};

export const tool$accessGroupsReadAccessGroupProject: ToolDefinition<
  typeof args
> = {
  name: "access-groups-read-access-group-project",
  description: `Reads an access group project

Allows reading an access group project`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await accessGroupsReadAccessGroupProject(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
