// @vercel/sdk/src/mcp-server/tools/deploymentsUpdateIntegrationDeploymentAction.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { deploymentsUpdateIntegrationDeploymentAction } from "../../funcs/deploymentsUpdateIntegrationDeploymentAction.js";
import { UpdateIntegrationDeploymentActionRequest$inboundSchema } from "../../models/updateintegrationdeploymentactionop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: UpdateIntegrationDeploymentActionRequest$inboundSchema,
};

export const tool$deploymentsUpdateIntegrationDeploymentAction: ToolDefinition<
  typeof args
> = {
  name: "deployments-update-integration-deployment-action",
  description: `Update deployment integration action

Updates the deployment integration action for the specified integration installation`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] =
      await deploymentsUpdateIntegrationDeploymentAction(
        client,
        args.request,
        { fetchOptions: { signal: ctx.signal } },
      ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    return formatResult(void 0, apiCall);
  },
};
