// @vercel/sdk/src/mcp-server/tools/marketplaceGetIntegrationResource.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { marketplaceGetIntegrationResource } from "../../funcs/marketplaceGetIntegrationResource.js";
import { GetIntegrationResourceRequest$inboundSchema } from "../../models/getintegrationresourceop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetIntegrationResourceRequest$inboundSchema,
};

export const tool$marketplaceGetIntegrationResource: ToolDefinition<
  typeof args
> = {
  name: "marketplace-get-integration-resource",
  description: `Get Integration Resource

Get a resource by its partner ID.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await marketplaceGetIntegrationResource(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
