// @vercel/sdk/src/mcp-server/tools/projectMembersGetProjectMembers.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectMembersGetProjectMembers } from "../../funcs/projectMembersGetProjectMembers.js";
import { GetProjectMembersRequest$inboundSchema } from "../../models/getprojectmembersop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetProjectMembersRequest$inboundSchema,
};

export const tool$projectMembersGetProjectMembers: ToolDefinition<typeof args> =
  {
    name: "project-members-get-project-members",
    description: `List project members

Lists all members of a project.`,
    args,
    tool: async (client, args, ctx) => {
      const [result, apiCall] = await projectMembersGetProjectMembers(
        client,
        args.request,
        { fetchOptions: { signal: ctx.signal } },
      ).$inspect();

      if (!result.ok) {
        return {
          content: [{ type: "text", text: result.error.message }],
          isError: true,
        };
      }

      const value = result.value;

      return formatResult(value, apiCall);
    },
  };
