// @vercel/sdk/src/mcp-server/tools/projectsRequestPromote.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectsRequestPromote } from "../../funcs/projectsRequestPromote.js";
import { RequestPromoteRequest$inboundSchema } from "../../models/requestpromoteop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: RequestPromoteRequest$inboundSchema,
};

export const tool$projectsRequestPromote: ToolDefinition<typeof args> = {
  name: "projects-request-promote",
  description: `Points all production domains for a project to the given deploy

Allows users to promote a deployment to production. Note: This does NOT rebuild the deployment. If you need that, then call create-deployments endpoint.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectsRequestPromote(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    return formatResult(void 0, apiCall);
  },
};
