// @vercel/sdk/esm/mcp-server/tools/aliasesListDeploymentAliases.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { aliasesListDeploymentAliases } from "../../funcs/aliasesListDeploymentAliases.js";
import { ListDeploymentAliasesRequest$inboundSchema } from "../../models/listdeploymentaliasesop.js";
import { formatResult } from "../tools.js";
const args = {
    request: ListDeploymentAliasesRequest$inboundSchema,
};
export const tool$aliasesListDeploymentAliases = {
    name: "aliases-list-deployment-aliases",
    description: `List Deployment Aliases

Retrieves all Aliases for the Deployment with the given ID. The authenticated user or team must own the deployment.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await aliasesListDeploymentAliases(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
