// @vercel/sdk/esm/mcp-server/tools/deploymentsDeleteDeployment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { deploymentsDeleteDeployment } from "../../funcs/deploymentsDeleteDeployment.js";
import { DeleteDeploymentRequest$inboundSchema } from "../../models/deletedeploymentop.js";
import { formatResult } from "../tools.js";
const args = {
    request: DeleteDeploymentRequest$inboundSchema,
};
export const tool$deploymentsDeleteDeployment = {
    name: "deployments-delete-deployment",
    description: `Delete a Deployment

This API allows you to delete a deployment, either by supplying its \`id\` in the URL or the \`url\` of the deployment as a query parameter. You can obtain the ID, for example, by listing all deployments.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await deploymentsDeleteDeployment(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
