// @vercel/sdk/esm/mcp-server/tools/edgeConfigGetEdgeConfigItem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigGetEdgeConfigItem } from "../../funcs/edgeConfigGetEdgeConfigItem.js";
import { GetEdgeConfigItemRequest$inboundSchema } from "../../models/getedgeconfigitemop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetEdgeConfigItemRequest$inboundSchema,
};
export const tool$edgeConfigGetEdgeConfigItem = {
    name: "edge-config-get-edge-config-item",
    description: `Get an Edge Config item

Returns a specific Edge Config Item.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigGetEdgeConfigItem(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
