// @vercel/sdk/esm/mcp-server/tools/edgeConfigGetEdgeConfigTokens.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigGetEdgeConfigTokens } from "../../funcs/edgeConfigGetEdgeConfigTokens.js";
import { GetEdgeConfigTokensRequest$inboundSchema } from "../../models/getedgeconfigtokensop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetEdgeConfigTokensRequest$inboundSchema,
};
export const tool$edgeConfigGetEdgeConfigTokens = {
    name: "edge-config-get-edge-config-tokens",
    description: `Get all tokens of an Edge Config

Returns all tokens of an Edge Config.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigGetEdgeConfigTokens(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
