// @vercel/sdk/esm/mcp-server/tools/edgeConfigDeleteEdgeConfigSchema.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { edgeConfigDeleteEdgeConfigSchema } from "../../funcs/edgeConfigDeleteEdgeConfigSchema.js";
import { DeleteEdgeConfigSchemaRequest$inboundSchema } from "../../models/deleteedgeconfigschemaop.js";
import { formatResult } from "../tools.js";
const args = {
    request: DeleteEdgeConfigSchemaRequest$inboundSchema,
};
export const tool$edgeConfigDeleteEdgeConfigSchema = {
    name: "edge-config-delete-edge-config-schema",
    description: `Delete an Edge Config's schema

Deletes the schema of existing Edge Config.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await edgeConfigDeleteEdgeConfigSchema(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return formatResult(void 0, apiCall);
    },
};
