// @vercel/sdk/esm/mcp-server/tools/marketplaceGetIntegrationResources.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { marketplaceGetIntegrationResources } from "../../funcs/marketplaceGetIntegrationResources.js";
import { GetIntegrationResourcesRequest$inboundSchema } from "../../models/getintegrationresourcesop.js";
import { formatResult } from "../tools.js";
const args = {
    request: GetIntegrationResourcesRequest$inboundSchema,
};
export const tool$marketplaceGetIntegrationResources = {
    name: "marketplace-get-integration-resources",
    description: `Get Integration Resources

Get all resources for a given installation ID.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await marketplaceGetIntegrationResources(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
