// @vercel/sdk/esm/mcp-server/tools/projectsRemoveProjectDomain.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { projectsRemoveProjectDomain } from "../../funcs/projectsRemoveProjectDomain.js";
import { RemoveProjectDomainRequest$inboundSchema } from "../../models/removeprojectdomainop.js";
import { formatResult } from "../tools.js";
const args = {
    request: RemoveProjectDomainRequest$inboundSchema,
};
export const tool$projectsRemoveProjectDomain = {
    name: "projects-remove-project-domain",
    description: `Remove a domain from a project

Remove a domain from a project by passing the domain name and by specifying the project by either passing the project \`id\` or \`name\` in the URL.`,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await projectsRemoveProjectDomain(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return formatResult(value, apiCall);
    },
};
