// @vercel/sdk/src/mcp-server/tools/accessGroupsListAccessGroupMembers.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { accessGroupsListAccessGroupMembers } from "../../funcs/accessGroupsListAccessGroupMembers.js";
import { ListAccessGroupMembersRequest$inboundSchema } from "../../models/listaccessgroupmembersop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: ListAccessGroupMembersRequest$inboundSchema,
};

export const tool$accessGroupsListAccessGroupMembers: ToolDefinition<
  typeof args
> = {
  name: "access-groups-list-access-group-members",
  description: `List members of an access group

List members of an access group`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await accessGroupsListAccessGroupMembers(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
