// @vercel/sdk/src/mcp-server/tools/checksCreateCheck.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { checksCreateCheck } from "../../funcs/checksCreateCheck.js";
import { CreateCheckRequest$inboundSchema } from "../../models/createcheckop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: CreateCheckRequest$inboundSchema,
};

export const tool$checksCreateCheck: ToolDefinition<typeof args> = {
  name: "checks-create-check",
  description: `Creates a new Check

Creates a new check. This endpoint must be called with an OAuth2 or it will produce a 400 error.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await checksCreateCheck(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
