// @vercel/sdk/src/mcp-server/tools/edgeConfigUpdateEdgeConfig.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { edgeConfigUpdateEdgeConfig } from "../../funcs/edgeConfigUpdateEdgeConfig.js";
import { UpdateEdgeConfigRequest$inboundSchema } from "../../models/updateedgeconfigop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: UpdateEdgeConfigRequest$inboundSchema,
};

export const tool$edgeConfigUpdateEdgeConfig: ToolDefinition<typeof args> = {
  name: "edge-config-update-edge-config",
  description: `Update an Edge Config

Updates an Edge Config.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await edgeConfigUpdateEdgeConfig(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
