// @vercel/sdk/src/mcp-server/tools/marketplaceCreateInstallationIntegrationConfiguration.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { marketplaceCreateInstallationIntegrationConfiguration } from "../../funcs/marketplaceCreateInstallationIntegrationConfiguration.js";
import {
  PostV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationItemsRequest$inboundSchema,
} from "../../models/postv1installationsintegrationconfigurationidresourcesresourceidexperimentationitemsop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request:
    PostV1InstallationsIntegrationConfigurationIdResourcesResourceIdExperimentationItemsRequest$inboundSchema,
};

export const tool$marketplaceCreateInstallationIntegrationConfiguration:
  ToolDefinition<typeof args> = {
    name: "marketplace-create-installation-integration-configuration",
    description: `Create one or multiple experimentation items

Create one or multiple experimentation items`,
    args,
    tool: async (client, args, ctx) => {
      const [result, apiCall] =
        await marketplaceCreateInstallationIntegrationConfiguration(
          client,
          args.request,
          { fetchOptions: { signal: ctx.signal } },
        ).$inspect();

      if (!result.ok) {
        return {
          content: [{ type: "text", text: result.error.message }],
          isError: true,
        };
      }

      return formatResult(void 0, apiCall);
    },
  };
