// @vercel/sdk/src/mcp-server/tools/projectMembersAddProjectMember.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectMembersAddProjectMember } from "../../funcs/projectMembersAddProjectMember.js";
import { AddProjectMemberRequest$inboundSchema } from "../../models/addprojectmemberop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: AddProjectMemberRequest$inboundSchema,
};

export const tool$projectMembersAddProjectMember: ToolDefinition<typeof args> =
  {
    name: "project-members-add-project-member",
    description: `Adds a new member to a project.

Adds a new member to the project.`,
    args,
    tool: async (client, args, ctx) => {
      const [result, apiCall] = await projectMembersAddProjectMember(
        client,
        args.request,
        { fetchOptions: { signal: ctx.signal } },
      ).$inspect();

      if (!result.ok) {
        return {
          content: [{ type: "text", text: result.error.message }],
          isError: true,
        };
      }

      const value = result.value;

      return formatResult(value, apiCall);
    },
  };
