// @vercel/sdk/src/mcp-server/tools/projectsGetProjectEnv.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { projectsGetProjectEnv } from "../../funcs/projectsGetProjectEnv.js";
import { GetProjectEnvRequest$inboundSchema } from "../../models/getprojectenvop.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: GetProjectEnvRequest$inboundSchema,
};

export const tool$projectsGetProjectEnv: ToolDefinition<typeof args> = {
  name: "projects-get-project-env",
  description:
    `Retrieve the decrypted value of an environment variable of a project by id

Retrieve the environment variable for a given project.`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await projectsGetProjectEnv(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
