// @vercel/sdk/esm/models/getcertbyidop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetCertByIdRequest$inboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetCertByIdRequest$outboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetCertByIdRequest$;
(function (GetCertByIdRequest$) {
    /** @deprecated use `GetCertByIdRequest$inboundSchema` instead. */
    GetCertByIdRequest$.inboundSchema = GetCertByIdRequest$inboundSchema;
    /** @deprecated use `GetCertByIdRequest$outboundSchema` instead. */
    GetCertByIdRequest$.outboundSchema = GetCertByIdRequest$outboundSchema;
})(GetCertByIdRequest$ || (GetCertByIdRequest$ = {}));
export function getCertByIdRequestToJSON(getCertByIdRequest) {
    return JSON.stringify(GetCertByIdRequest$outboundSchema.parse(getCertByIdRequest));
}
export function getCertByIdRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCertByIdRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCertByIdRequest' from JSON`);
}
/** @internal */
export const GetCertByIdResponseBody$inboundSchema = z.object({
    id: z.string(),
    createdAt: z.number(),
    expiresAt: z.number(),
    autoRenew: z.boolean(),
    cns: z.array(z.string()),
});
/** @internal */
export const GetCertByIdResponseBody$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.number(),
    expiresAt: z.number(),
    autoRenew: z.boolean(),
    cns: z.array(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetCertByIdResponseBody$;
(function (GetCertByIdResponseBody$) {
    /** @deprecated use `GetCertByIdResponseBody$inboundSchema` instead. */
    GetCertByIdResponseBody$.inboundSchema = GetCertByIdResponseBody$inboundSchema;
    /** @deprecated use `GetCertByIdResponseBody$outboundSchema` instead. */
    GetCertByIdResponseBody$.outboundSchema = GetCertByIdResponseBody$outboundSchema;
})(GetCertByIdResponseBody$ || (GetCertByIdResponseBody$ = {}));
export function getCertByIdResponseBodyToJSON(getCertByIdResponseBody) {
    return JSON.stringify(GetCertByIdResponseBody$outboundSchema.parse(getCertByIdResponseBody));
}
export function getCertByIdResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCertByIdResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCertByIdResponseBody' from JSON`);
}
