// @vercel/sdk/esm/models/getedgeconfigitemop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetEdgeConfigItemRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    edgeConfigItemKey: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetEdgeConfigItemRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    edgeConfigItemKey: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetEdgeConfigItemRequest$;
(function (GetEdgeConfigItemRequest$) {
    /** @deprecated use `GetEdgeConfigItemRequest$inboundSchema` instead. */
    GetEdgeConfigItemRequest$.inboundSchema = GetEdgeConfigItemRequest$inboundSchema;
    /** @deprecated use `GetEdgeConfigItemRequest$outboundSchema` instead. */
    GetEdgeConfigItemRequest$.outboundSchema = GetEdgeConfigItemRequest$outboundSchema;
})(GetEdgeConfigItemRequest$ || (GetEdgeConfigItemRequest$ = {}));
export function getEdgeConfigItemRequestToJSON(getEdgeConfigItemRequest) {
    return JSON.stringify(GetEdgeConfigItemRequest$outboundSchema.parse(getEdgeConfigItemRequest));
}
export function getEdgeConfigItemRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigItemRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigItemRequest' from JSON`);
}
