// @vercel/sdk/esm/models/getedgeconfigtokenop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetEdgeConfigTokenRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    token: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetEdgeConfigTokenRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    token: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetEdgeConfigTokenRequest$;
(function (GetEdgeConfigTokenRequest$) {
    /** @deprecated use `GetEdgeConfigTokenRequest$inboundSchema` instead. */
    GetEdgeConfigTokenRequest$.inboundSchema = GetEdgeConfigTokenRequest$inboundSchema;
    /** @deprecated use `GetEdgeConfigTokenRequest$outboundSchema` instead. */
    GetEdgeConfigTokenRequest$.outboundSchema = GetEdgeConfigTokenRequest$outboundSchema;
})(GetEdgeConfigTokenRequest$ || (GetEdgeConfigTokenRequest$ = {}));
export function getEdgeConfigTokenRequestToJSON(getEdgeConfigTokenRequest) {
    return JSON.stringify(GetEdgeConfigTokenRequest$outboundSchema.parse(getEdgeConfigTokenRequest));
}
export function getEdgeConfigTokenRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigTokenRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigTokenRequest' from JSON`);
}
