// @vercel/sdk/esm/models/vercelnotfounderror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { VercelError } from "./vercelerror.js";
export class VercelNotFoundError extends VercelError {
    constructor(err, httpMeta) {
        const message = err.error?.message
            || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.name = "VercelNotFoundError";
    }
}
/** @internal */
export const VercelNotFoundErrorError$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
});
/** @internal */
export const VercelNotFoundErrorError$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var VercelNotFoundErrorError$;
(function (VercelNotFoundErrorError$) {
    /** @deprecated use `VercelNotFoundErrorError$inboundSchema` instead. */
    VercelNotFoundErrorError$.inboundSchema = VercelNotFoundErrorError$inboundSchema;
    /** @deprecated use `VercelNotFoundErrorError$outboundSchema` instead. */
    VercelNotFoundErrorError$.outboundSchema = VercelNotFoundErrorError$outboundSchema;
})(VercelNotFoundErrorError$ || (VercelNotFoundErrorError$ = {}));
export function vercelNotFoundErrorErrorToJSON(vercelNotFoundErrorError) {
    return JSON.stringify(VercelNotFoundErrorError$outboundSchema.parse(vercelNotFoundErrorError));
}
export function vercelNotFoundErrorErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => VercelNotFoundErrorError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VercelNotFoundErrorError' from JSON`);
}
/** @internal */
export const VercelNotFoundError$inboundSchema = z.object({
    error: z.lazy(() => VercelNotFoundErrorError$inboundSchema),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new VercelNotFoundError(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const VercelNotFoundError$outboundSchema = z.instanceof(VercelNotFoundError)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.lazy(() => VercelNotFoundErrorError$outboundSchema),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var VercelNotFoundError$;
(function (VercelNotFoundError$) {
    /** @deprecated use `VercelNotFoundError$inboundSchema` instead. */
    VercelNotFoundError$.inboundSchema = VercelNotFoundError$inboundSchema;
    /** @deprecated use `VercelNotFoundError$outboundSchema` instead. */
    VercelNotFoundError$.outboundSchema = VercelNotFoundError$outboundSchema;
})(VercelNotFoundError$ || (VercelNotFoundError$ = {}));
